---
title: "13.Diversity_calculation_comparison_among_pops"
output: html_document
---

# Create dataframes

## Load library & wd
```{r}
library(viridis)
library(ggplot2); theme_set(theme_minimal())
library(tidyr)
library(plyr)
library(dplyr)
library(data.table)
library(arules)
library(psych)
library(ggpubr)
library(boot)
library(broom)
library(ggpubr)
library(ggrepel)
library(ggpmisc)
library(gridExtra)
library(mgcv)
library(MuMIn)

wd <- "/Users/marialucenaperez/Documents/WG_lynx_diversity_per_unit/"
wd_output <- "/Users/marialucenaperez/Documents/WG_lynx_diversity_per_unit/comparison_among_pops/"
wd_input_stats <- "/Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/per_pop/"
  
```

## Load pre-data

```{r}
c_ll_ki_n013_diversity <- read.csv(paste(wd, "c_ll_ki_n013_diversity.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") %>% mutate (., color =  viridis_pal()(5)[1])

c_ll_no_n008_diversity <- read.csv(paste(wd, "c_ll_no_n008_diversity.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") %>% mutate (., color = viridis_pal()(5)[2])

c_ll_po_n008_diversity <- read.csv(paste(wd, "c_ll_po_n008_diversity.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") %>% mutate (., color =  viridis_pal()(5)[3])

c_lp_sm_n019_diversity <- read.csv(paste(wd, "c_lp_sm_n019_diversity.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") %>% mutate (., color =  "#5DC863FF")

c_lp_do_n012_diversity <- read.csv(paste(wd, "c_lp_do_n012_diversity.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") %>% mutate (., color =  "#FDE725FF")


density_min_values_pairwise <- read.table(paste(wd, "density_min_values_pairwise.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") 
density_min_values_watterson <- read.table(paste(wd, "density_min_values_watterson.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") 
stats_df_pop <- read.table(paste(wd_input_stats,"stats_diversity_per_pop.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T)
stats_df_pop_feature <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_region <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_recomb.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_feature_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_region_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_recomb_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_recomb_chr.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_feature_region <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_region.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_feature_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_recomb.tsv",sep=""),row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_region_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region_recomb.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
```

## 1. Create df per unit pop comparison

```{r}
# Function:
# Cuidado porque aquí estoy asumiendo que la region es igual, y esto parece que es asi, pero no estoy 100% segura de que no haya podido pasar que en una población se haya definido como telomérico y en otra no por falta de bases. 
create_data_diversity_bootleneck_vs_non_bootleneck <- function(POP1, POP2, name_POP1, name_POP2){
    POPULATION1=POP1 %>% select(Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1] 
    POPULATION2=POP2 %>% select(Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1] 
  data_diversity_POP1_POP2 <- dplyr::inner_join (POP1, POP2, by = c("scaffold", "start_cero_based", "end", "length", "feature", "strandness", "frame", "id_gene", "id", "region", "chr", "unique_id", "recombination_rate", "recomb", "type_chr", "divergence", "informative_sites_substitutions", "substitutions", "GC_content", "rvis_score", "cat_position_start", "cat_position_end")) %>%  
      # Calculo el delta 
    dplyr::mutate (delta_pairwise_per_unit = (pairwise_ave.x -pairwise_ave.y)) %>%
    dplyr::mutate (delta_watterson_per_unit = (watterson_ave.x - watterson_ave.y)) %>% 
    dplyr::mutate (delta_pairwise_per_unit_zero = (pairwise_zero.x - pairwise_zero.y)) %>% 
    dplyr::mutate (delta_watterson_per_unit_zero = (watterson_zero.x - watterson_zero.y)) %>% 
    # Calculo el delta normalizado para la suma de los valores de diversidad  
    dplyr::mutate (delta_pairwise_per_unit_normalized = ((pairwise_ave.x - pairwise_ave.y) / (pairwise_ave.x + pairwise_ave.y))) %>%
    dplyr::mutate (delta_watterson_per_unit_normalized = ((watterson_ave.x - watterson_ave.y) / (watterson_ave.x + watterson_ave.y))) %>% 
    dplyr::mutate (delta_pairwise_per_unit_normalized_zero = ((pairwise_zero.x - pairwise_zero.y) / (pairwise_zero.x + pairwise_zero.y))) %>% 
    dplyr::mutate (delta_watterson_per_unit_normalized_zero = ((watterson_zero.x - watterson_zero.y) / (watterson_zero.x + watterson_zero.y))) %>% 
    # Add comparison title  
    dplyr::mutate(comparison=paste(POPULATION1, "-", POPULATION2, sep=""))
  dataframename <- paste (name_POP1, name_POP2, sep="_")
  assign (dataframename, data_diversity_POP1_POP2,.GlobalEnv)
}
# Create data frame
# Dataframe Kirov-Norway
create_data_diversity_bootleneck_vs_non_bootleneck(c_ll_ki_n013_diversity,c_ll_no_n008_diversity, deparse(substitute(c_ll_ki_n013_diversity)),deparse(substitute(c_ll_no_n008_diversity)))
# Dataframe Kirov-Poland
create_data_diversity_bootleneck_vs_non_bootleneck(c_ll_ki_n013_diversity,c_ll_po_n008_diversity, deparse(substitute(c_ll_ki_n013_diversity)),deparse(substitute(c_ll_po_n008_diversity)))
# Dataframe Sierra_Morena-Doñana
create_data_diversity_bootleneck_vs_non_bootleneck(c_lp_sm_n019_diversity,c_lp_do_n012_diversity, deparse(substitute(c_lp_sm_n019_diversity)),deparse(substitute(c_lp_do_n012_diversity)))
# Combine comparisons
all_comparisons_diversity_bootleneck_vs_non_bootleneck <- rbind (c_ll_ki_n013_diversity_c_ll_no_n008_diversity, c_ll_ki_n013_diversity_c_ll_po_n008_diversity, c_lp_sm_n019_diversity_c_lp_do_n012_diversity)

# Write table
write.table (c_ll_ki_n013_diversity_c_ll_no_n008_diversity, paste0 (wd_output, "c_ll_ki_n013_diversity_c_ll_no_n008_diversity.txt"), quote = F,  row.names = F, sep =";", dec="," )
write.table (c_ll_ki_n013_diversity_c_ll_po_n008_diversity, paste0 (wd_output, "c_ll_ki_n013_diversity_c_ll_po_n008_diversity.txt"), quote = F,  row.names = F, sep =";", dec="," )
write.table (c_lp_sm_n019_diversity_c_lp_do_n012_diversity, paste0 (wd_output, "c_lp_sm_n019_diversity_c_lp_do_n012_diversity.txt"), quote = F,  row.names = F, sep =";", dec="," )
write.table(all_comparisons_diversity_bootleneck_vs_non_bootleneck, paste0 (wd_output, "all_comparisons_diversity_bootleneck_vs_non_bootleneck.txt"), quote = F,  row.names = F, sep =";", dec="," )
```

## 2. Create stats dataframes

Primero creo una formula para poder calcular la stdev basada en bootstrapping.
```{r}
sample.wtd.mean <- function(x, w, d) {
    return(weighted.mean(x = x[d], w = w[d], na.rm=T ))}
```

### Summary stats per pop comparison 
Now I group the data based on population.

```{r}
diversity_stats_delta <- function(DATAFRAME)
{ diversity_stats_df <- 
             DATAFRAME %>% 
             dplyr::group_by_("comparison") %>%
             dplyr::summarise(
                       total_count=n(), 
                       # Distribución sin corregir
                       wmean_delta_watterson_per_unit_normalized_zero=weighted.mean(delta_watterson_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_watterson_per_unit_normalized_zero_boot = sd(boot(delta_watterson_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t),
                       wmean_delta_pairwise_per_unit_normalized_zero=weighted.mean(delta_pairwise_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_pairwise_per_unit_normalized_zero_boot = sd(boot(delta_pairwise_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t)
                       ) }


stats_df_pop_comparison_autosomes <- diversity_stats_delta (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes")) %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX <- diversity_stats_delta (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX")) %>%  mutate (type_chr="chrX")

stats_df_pop_comparison <- rbind(stats_df_pop_comparison_autosomes, stats_df_pop_comparison_chrX)
write.table(stats_df_pop_comparison, paste(wd_output,"stats_df_pop_comparison.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
```

### Summary stats per pop & different group (feature, chr, region, recom_category) 

```{r}

diversity_stats_delta_group <- function(DATAFRAME, GROUP1)
{ diversity_stats_df_grouped <- 
             DATAFRAME %>% 
             dplyr::group_by_("comparison", GROUP1) %>%
             dplyr::summarise(
                       total_count=n(), 
                       # Distribución sin corregir
                       wmean_delta_watterson_per_unit_normalized_zero=weighted.mean(delta_watterson_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_watterson_per_unit_normalized_zero_boot = sd(boot(delta_watterson_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t),
                       wmean_delta_pairwise_per_unit_normalized_zero=weighted.mean(delta_pairwise_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_pairwise_per_unit_normalized_zero_boot = sd(boot(delta_pairwise_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t)
                       ) }


stats_df_pop_comparison_autosomes_feature <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "feature") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX_feature <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "feature") %>%  mutate (type_chr="chrX")

stats_df_pop_comparison_autosomes_region <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "region") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX_region <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "region") %>%  mutate (type_chr="chrX")

stats_df_pop_comparison_autosomes_recomb <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "recomb") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX_recomb <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "recomb") %>%  mutate (type_chr="chrX")

stats_df_pop_comparison_autosomes_chr <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "chr") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX_chr <- diversity_stats_delta_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "chr") %>%  mutate (type_chr="chrX")

# Fusiono y guardo

stats_df_pop_comparison_feature <- rbind (stats_df_pop_comparison_autosomes_feature, stats_df_pop_comparison_chrX_feature)
stats_df_pop_comparison_region <- rbind (stats_df_pop_comparison_autosomes_region, stats_df_pop_comparison_chrX_region)
stats_df_pop_comparison_recomb <- rbind (stats_df_pop_comparison_autosomes_recomb, stats_df_pop_comparison_chrX_recomb)
stats_df_pop_comparison_chr <- rbind (stats_df_pop_comparison_autosomes_chr, stats_df_pop_comparison_chrX_chr)

write.table(stats_df_pop_comparison_feature, paste(wd_output,"stats_df_pop_comparison_feature.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_region, paste(wd_output,"stats_df_pop_comparison_region.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_chr, paste(wd_output,"stats_df_pop_comparison_chr.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_recomb, paste(wd_output,"stats_df_pop_comparison_recomb.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )

```

###  Summary stats per pop & different group - group (feature-chr, feature-region, chr-region) 

```{r}

diversity_stats_delta_group_group <- function(DATAFRAME, GROUP1, GROUP2)
{ diversity_stats_df_grouped <- 
             DATAFRAME %>% 
             dplyr::group_by_("comparison", GROUP1, GROUP2) %>%
             dplyr::summarise(
                       total_count=n(), 
                       # Distribución sin corregir
                       wmean_delta_watterson_per_unit_normalized_zero=weighted.mean(delta_watterson_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_watterson_per_unit_normalized_zero_boot = sd(boot(delta_watterson_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t),
                       wmean_delta_pairwise_per_unit_normalized_zero=weighted.mean(delta_pairwise_per_unit_normalized_zero,w=(informative_sites.x + informative_sites.y), na.rm = TRUE),
                       se_wmean_delta_pairwise_per_unit_normalized_zero_boot = sd(boot(delta_pairwise_per_unit_normalized_zero, sample.wtd.mean, R = 100, w=(as.numeric(informative_sites.x) + as.numeric(informative_sites.y)))$t)
                       ) }

stats_df_pop_comparison_autosomes_feature_region <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "feature", "region") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_autosomes_feature_recomb <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "feature", "recomb") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_autosomes_region_recomb <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "region", "recomb") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_autosomes_feature_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "feature", "chr") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_autosomes_region_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "region", "chr") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_autosomes_recomb_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="autosomes"), "recomb", "chr") %>%  mutate (type_chr="autosomes")
stats_df_pop_comparison_chrX_feature_region <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "feature", "region") %>%  mutate (type_chr="chrX")
stats_df_pop_comparison_chrX_feature_recomb <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "feature", "recomb") %>%  mutate (type_chr="chrX")
stats_df_pop_comparison_chrX_region_recomb <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "region", "recomb") %>%  mutate (type_chr="chrX")
stats_df_pop_comparison_chrX_feature_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "feature", "chr") %>%  mutate (type_chr="chrX")
stats_df_pop_comparison_chrX_region_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "region", "chr") %>%  mutate (type_chr="chrX")
stats_df_pop_comparison_chrX_recomb_chr <- diversity_stats_delta_group_group (all_comparisons_diversity_bootleneck_vs_non_bootleneck %>% filter (type_chr=="chrX"), "recomb", "chr") %>%  mutate (type_chr="chrX")

# Fusiono y guardo
stats_df_pop_comparison_feature_region <- rbind (stats_df_pop_comparison_autosomes_feature_region, stats_df_pop_comparison_chrX_feature_region)
stats_df_pop_comparison_feature_recomb <- rbind (stats_df_pop_comparison_autosomes_feature_recomb, stats_df_pop_comparison_chrX_feature_recomb)
stats_df_pop_comparison_region_recomb <- rbind (stats_df_pop_comparison_autosomes_region_recomb, stats_df_pop_comparison_chrX_region_recomb)
stats_df_pop_comparison_feature_chr <- rbind (stats_df_pop_comparison_autosomes_feature_chr, stats_df_pop_comparison_chrX_feature_chr)
stats_df_pop_comparison_region_chr <- rbind (stats_df_pop_comparison_autosomes_region_chr, stats_df_pop_comparison_chrX_region_chr)
stats_df_pop_comparison_recomb_chr <- rbind (stats_df_pop_comparison_autosomes_recomb_chr, stats_df_pop_comparison_chrX_recomb_chr)

write.table(stats_df_pop_comparison_feature_region, paste(wd_output,"stats_df_pop_comparison_feature_region.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_feature_recomb, paste(wd_output,"stats_df_pop_comparison_feature_recomb.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_region_recomb, paste(wd_output,"stats_df_pop_comparison_region_recomb.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_feature_chr, paste(wd_output,"stats_df_pop_comparison_feature_chr.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_region_chr, paste(wd_output,"stats_df_pop_comparison_region_chr.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_df_pop_comparison_recomb_chr, paste(wd_output,"stats_df_pop_comparison_recomb_chr.tsv",sep=""),quote = F,  row.names = F, sep =";", dec="," )

```


## 3. Modify individal per pop stat file

We have loaded per pop stats. No we have to change the configuration of those so that we can compare two populations.

```{r}
POPS=c("c_ll_ki_n013","c_ll_po_n008","c_ll_no_n008", "c_lp_sm_n019", "c_lp_do_n012")

# Feature
for (POP in POPS)
  {print (POP)
    data.frame <-  stats_df_pop_feature  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_feature", sep=""), data.frame)
  rm (data.frame)}

# Chr
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_chr  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_chr", sep=""), data.frame)
  rm (data.frame)}

# Region
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_region  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_region", sep=""), data.frame)
  rm (data.frame)}

# Recomb
for (POP in POPS)
  {print (POP)
    data.frame <-  stats_df_pop_recomb  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_recomb", sep=""), data.frame)
  rm (data.frame)}

## INTERACCION
# Feature - chr
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_feature_chr  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_feature_chr", sep=""), data.frame)
  rm (data.frame)}

# Feature - region
for (POP in POPS)
{ print (POP)
    data.frame <-  stats_df_pop_feature_region  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_feature_region", sep=""), data.frame)
  rm (data.frame)}

# Feature - recomb
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_feature_recomb  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_feature_recomb", sep=""), data.frame)
  rm (data.frame)}

# Region - chr
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_region_chr  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_region_chr", sep=""), data.frame)
  rm (data.frame)}

# Region - recomb
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_region_recomb  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_region_recomb", sep=""), data.frame)
  rm (data.frame)}

# Chr - recomb
for (POP in POPS)
{print (POP)
    data.frame <-  stats_df_pop_recomb_chr  %>%  filter (pop==as.character(POP))
    assign(paste(POP,"_stats_chr_recomb", sep=""), data.frame)
  rm (data.frame)}

```

#### $ NO GROUP


#### One group

En estas tablas estoy uniendo toda la info de cada una de las poblaciones junto con la de la comparación, i.e. diversidad de una población, diversidad de la otra, y luego la delta, tanto global como 

Function

Con esta funcion calculamos la delta global para pares de poblaciones. 
```{r}
bootleneck_vs_non_bootleneck_interaction_stats_group <- function (dataframe_POP1, dataframe_POP2, name_POP1, name_POP2, GROUP){
  # Nombre de las poblaciones
  POP1=as.character(as.data.frame(dataframe_POP1) %>% select (pop) %>% mutate(pop=as.character(pop)) %>% .[1,1]) 
  POP2=as.character(as.data.frame(dataframe_POP2) %>% select (pop) %>% mutate(pop=as.character(pop)) %>% .[1,1]) 
  complete_pop_name1=as.character(as.data.frame(dataframe_POP1) %>% select (Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1]) 
  complete_pop_name2=as.character(as.data.frame(dataframe_POP2) %>% select (Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1]) 
  # Calculo de los ratios globales.
  POP1_POP2 <- full_join ( dataframe_POP1, dataframe_POP2, by=c(GROUP, "type_chr", "Species")) %>% 
    mutate (comparison=paste(complete_pop_name1,"-", complete_pop_name2, sep="")) %>% 
  # Calculate global delta for a pair of populations
    mutate (global_delta_pairwise = wmean_pairwise_ave.y/wmean_pairwise_ave.x) %>% 
    mutate (global_delta_watterson = wmean_watterson_ave.y/wmean_watterson_ave.x)
  dataframename <- paste ("stats_df_pop", POP1, POP2, GROUP, sep="_")
  assign (dataframename, POP1_POP2,.GlobalEnv)
}
```

Create data frame

```{r}
# Feature
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_feature,c_ll_no_n008_stats_feature, deparse(substitute(c_ll_ki_n013_stats_feature)),deparse(substitute(c_ll_no_n008_stats_feature)), "feature")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_feature,c_ll_po_n008_stats_feature, deparse(substitute(c_ll_ki_n013_stats_feature)),deparse(substitute(c_ll_po_n008_stats_feature)), "feature")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_group(c_lp_sm_n019_stats_feature, c_lp_do_n012_stats_feature, deparse(substitute(c_lp_sm_n019_stats_feature)),deparse(substitute(c_lp_do_n012_stats_feature)), "feature")
# Region
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_region,c_ll_no_n008_stats_region, deparse(substitute(c_ll_ki_n013_stats_region)),deparse(substitute(c_ll_no_n008_stats_region)), "region")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_region,c_ll_po_n008_stats_region, deparse(substitute(c_ll_ki_n013_stats_region)),deparse(substitute(c_ll_po_n008_stats_region)), "region")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_group(c_lp_sm_n019_stats_region, c_lp_do_n012_stats_region, deparse(substitute(c_lp_sm_n019_stats_region)),deparse(substitute(c_lp_do_n012_stats_region)), "region")
# Recomb
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_recomb,c_ll_no_n008_stats_recomb, deparse(substitute(c_ll_ki_n013_stats_recomb)),deparse(substitute(c_ll_no_n008_stats_recomb)), "recomb")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_recomb,c_ll_po_n008_stats_recomb, deparse(substitute(c_ll_ki_n013_stats_recomb)),deparse(substitute(c_ll_po_n008_stats_recomb)), "recomb")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_group(c_lp_sm_n019_stats_recomb, c_lp_do_n012_stats_recomb, deparse(substitute(c_lp_sm_n019_stats_recomb)),deparse(substitute(c_lp_do_n012_stats_recomb)), "recomb")
# Chr
# Dataframe Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_chr,c_ll_no_n008_stats_chr, deparse(substitute(c_ll_ki_n013_stats_chr)),deparse(substitute(c_ll_no_n008_stats_chr)), "chr")
# Dataframe Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_group(c_ll_ki_n013_stats_chr,c_ll_po_n008_stats_chr, deparse(substitute(c_ll_ki_n013_stats_chr)),deparse(substitute(c_ll_po_n008_stats_chr)), "chr")
# Dataframe Sierra_Morena-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_group(c_lp_sm_n019_stats_chr, c_lp_do_n012_stats_chr, deparse(substitute(c_lp_sm_n019_stats_chr)),deparse(substitute(c_lp_do_n012_stats_chr)), "chr")

stats_all_pops_feature <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_feature, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_feature, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_feature) %>% 
  full_join (stats_df_pop_comparison_feature, by=c("comparison", "type_chr", "feature"))

stats_all_pops_region <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_region, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_region, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_region) %>% 
  full_join (stats_df_pop_comparison_region, by=c("comparison", "type_chr", "region"))

stats_df_pop_comparison_recomb$recomb <- as.factor(stats_df_pop_comparison_recomb$recomb)
stats_all_pops_recomb <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_recomb, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_recomb, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_recomb) %>% 
  dplyr::full_join (stats_df_pop_comparison_recomb, by=c("comparison", "type_chr", "recomb"))

stats_all_pops_chr <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_chr, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_chr, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_chr) %>% 
  dplyr::full_join (stats_df_pop_comparison_chr, by=c("comparison", "type_chr", "chr"))

write.table(stats_all_pops_feature, paste(wd_output,"stats_all_pops_feature.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_region, paste(wd_output,"stats_all_pops_region.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_recomb, paste(wd_output,"stats_all_pops_recomb.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_chr, paste(wd_output,"stats_all_pops_chr.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )

```

#### Two groups

```{r}
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP <- function(dataframe_POP1, dataframe_POP2, name_POP1, name_POP2, GROUP1, GROUP2){
  POP1=as.character(as.data.frame(dataframe_POP1) %>% select (pop) %>% mutate(pop=as.character(pop)) %>% .[1,1]) 
  POP2=as.character(as.data.frame(dataframe_POP2) %>% select (pop) %>% mutate(pop=as.character(pop)) %>% .[1,1]) 
  complete_pop_name1=as.character(as.data.frame(dataframe_POP1) %>% select (Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1]) 
  complete_pop_name2=as.character(as.data.frame(dataframe_POP2) %>% select (Populations) %>% mutate(Populations=as.character(Populations)) %>% .[1,1]) 
  # Calculo de los ratios globales.
  POP1_POP2 <- full_join ( dataframe_POP1, dataframe_POP2, by=c(GROUP1, GROUP2, "type_chr", "Species")) %>% 
    mutate (comparison=paste(complete_pop_name1,"-", complete_pop_name2, sep="")) %>% 
  # Calculate global delta for a pair of populations
    mutate (global_delta_pairwise = wmean_pairwise_ave.y/wmean_pairwise_ave.x) %>% 
    mutate (global_delta_watterson = wmean_watterson_ave.y/wmean_watterson_ave.x)
  dataframename <- paste ("stats_df_pop", POP1, POP2, GROUP1, GROUP2, sep="_")
  assign (dataframename, POP1_POP2,.GlobalEnv)
  }
```

```{r}
# Feature-chr
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_chr,c_ll_no_n008_stats_feature_chr, deparse(substitute(c_ll_ki_n013_stats_feature_chr)),deparse(substitute(c_ll_no_n008_stats_feature_chr)), "feature", "chr")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_chr,c_ll_po_n008_stats_feature_chr, deparse(substitute(c_ll_ki_n013_stats_feature_chr)),deparse(substitute(c_ll_po_n008_stats_feature_chr)), "feature", "chr")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_feature_chr, c_lp_do_n012_stats_feature_chr, deparse(substitute(c_lp_sm_n019_stats_feature_chr)),deparse(substitute(c_lp_do_n012_stats_feature_chr)), "feature", "chr")
# Feature-region
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_region,c_ll_no_n008_stats_feature_region, deparse(substitute(c_ll_ki_n013_stats_feature_region)),deparse(substitute(c_ll_no_n008_stats_feature_region)), "feature", "region")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_region,c_ll_po_n008_stats_feature_region, deparse(substitute(c_ll_ki_n013_stats_feature_region)),deparse(substitute(c_ll_po_n008_stats_feature_region)), "feature", "region")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_feature_region,c_lp_do_n012_stats_feature_region, deparse(substitute(c_lp_sm_n019_stats_feature_region)),deparse(substitute(c_lp_do_n012_stats_feature_region)), "feature", "region")
# Feature-recomb
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_recomb,c_ll_no_n008_stats_feature_recomb, deparse(substitute(c_ll_ki_n013_stats_feature_recomb)),deparse(substitute(c_ll_no_n008_stats_feature_recomb)), "feature", "recomb")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_feature_recomb,c_ll_po_n008_stats_feature_recomb, deparse(substitute(c_ll_ki_n013_stats_feature_recomb)),deparse(substitute(c_ll_po_n008_stats_feature_recomb)), "feature", "recomb")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_feature_recomb, c_lp_do_n012_stats_feature_recomb, deparse(substitute(c_lp_sm_n019_stats_feature_recomb)),deparse(substitute(c_lp_do_n012_stats_feature_recomb)), "feature", "recomb")
# Region-Chr
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_region_chr,c_ll_no_n008_stats_region_chr, deparse(substitute(c_ll_ki_n013_stats_region_chr)),deparse(substitute(c_ll_no_n008_stats_region_chr)), "region", "chr")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_region_chr,c_ll_po_n008_stats_region_chr, deparse(substitute(c_ll_ki_n013_stats_region_chr)),deparse(substitute(c_ll_po_n008_stats_region_chr)), "region", "chr")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_region_chr, c_lp_do_n012_stats_region_chr, deparse(substitute(c_lp_sm_n019_stats_region_chr)),deparse(substitute(c_lp_do_n012_stats_region_chr)), "region", "chr")
# Region-Recomb
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_region_recomb,c_ll_no_n008_stats_region_recomb, deparse(substitute(c_ll_ki_n013_stats_region_recomb)),deparse(substitute(c_ll_no_n008_stats_region_recomb)), "region", "recomb")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_region_recomb,c_ll_po_n008_stats_region_recomb, deparse(substitute(c_ll_ki_n013_stats_region_recomb)),deparse(substitute(c_ll_po_n008_stats_region_recomb)), "region", "recomb")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_region_recomb, c_lp_do_n012_stats_region_recomb, deparse(substitute(c_lp_sm_n019_stats_region_recomb)),deparse(substitute(c_lp_do_n012_stats_region_recomb)), "region", "recomb")
# Recomb-chr
## Kirov-Norway
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_chr_recomb,c_ll_no_n008_stats_chr_recomb, deparse(substitute(c_ll_ki_n013_stats_chr_recomb)),deparse(substitute(c_ll_no_n008_stats_chr_recomb)), "recomb", "chr")
## Kirov-Poland
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_ll_ki_n013_stats_chr_recomb,c_ll_po_n008_stats_chr_recomb, deparse(substitute(c_ll_ki_n013_stats_chr_recomb)),deparse(substitute(c_ll_po_n008_stats_chr_recomb)), "recomb", "chr")
## Andujar-Doñana
bootleneck_vs_non_bootleneck_interaction_stats_feature_GROUP_GROUP(c_lp_sm_n019_stats_chr_recomb, c_lp_do_n012_stats_chr_recomb, deparse(substitute(c_lp_sm_n019_stats_chr_recomb)),deparse(substitute(c_lp_do_n012_stats_chr_recomb)), "recomb", "chr")



stats_all_pops_feature_chr <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_feature_chr, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_feature_chr, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_feature_chr) %>% 
  full_join (stats_df_pop_comparison_feature_chr, by=c("comparison", "type_chr", "chr", "feature"))

stats_all_pops_feature_region <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_feature_region, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_feature_region, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_feature_region) %>% 
  full_join (stats_df_pop_comparison_feature_region, by=c("comparison", "type_chr", "region", "feature"))

stats_df_pop_comparison_feature_recomb$recomb <- as.factor(stats_df_pop_comparison_feature_recomb$recomb)
stats_all_pops_feature_recomb <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_feature_recomb, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_feature_recomb, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_feature_recomb) %>% 
  full_join (stats_df_pop_comparison_feature_recomb, by=c("comparison", "type_chr", "recomb", "feature"))

stats_all_pops_region_chr <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_region_chr, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_region_chr, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_region_chr) %>% 
  full_join (stats_df_pop_comparison_region_chr, by=c("comparison", "type_chr", "region", "chr"))

stats_df_pop_comparison_region_recomb$recomb <- as.factor(stats_df_pop_comparison_region_recomb$recomb)
stats_all_pops_region_recomb <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_region_recomb, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_region_recomb, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_region_recomb) %>% 
  full_join (stats_df_pop_comparison_region_recomb, by=c("comparison", "type_chr", "region", "recomb"))

stats_df_pop_comparison_recomb_chr$recomb <- as.factor(stats_df_pop_comparison_recomb_chr$recomb)
stats_all_pops_recomb_chr <- rbind (stats_df_pop_c_ll_ki_n013_c_ll_no_n008_recomb_chr, stats_df_pop_c_ll_ki_n013_c_ll_po_n008_recomb_chr, stats_df_pop_c_lp_sm_n019_c_lp_do_n012_recomb_chr) %>% 
  full_join (stats_df_pop_comparison_recomb_chr, by=c("comparison", "type_chr", "recomb", "chr"))

write.table(stats_all_pops_feature_chr, paste(wd_output,"stats_all_pops_feature_chr.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_feature_region, paste(wd_output,"stats_all_pops_feature_region.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_feature_recomb, paste(wd_output,"stats_all_pops_feature_recomb.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_region_chr, paste(wd_output,"stats_all_pops_region_chr.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_region_recomb, paste(wd_output,"stats_all_pops_region_recomb.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
write.table(stats_all_pops_recomb_chr, paste(wd_output,"stats_all_pops_recomb_chr.txt",sep=""),quote = F,  row.names = F, sep =";", dec="," )
```

# -----------------------------------------------------------------
# -----------------------------------------------------------------
# START THE ANALYSIS
# -----------------------------------------------------------------

# Diversity per pop
## Load library & wd
```{r}
library(viridis)
library(ggplot2); theme_set(theme_minimal())
library(tidyr)
library(plyr)
library(dplyr)
library(data.table)
library(arules)
library(boot)
library(broom)
library(Hmisc)
library(GGally)
library(ggpmisc)
library(mgcv)
library(MuMIn)
library(mgcViz)
library(ggrepel)
library(ggsignif)
library(ggpubr)
library(cowplot)
library(ggsignif)
library(tidyverse)

wd <- "/Users/marialucenaperez/Documents/WG_lynx_diversity_per_unit/"
wd_output <- "/Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/"
wd_input <- "/Users/marialucenaperez/Documents/WG_lynx_diversity_per_unit/comparison_among_pops/"
wd_input_stats <- "/Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/per_pop/"
```

## Load functions
Esto sirve para despues poder girar los ejes de los factores, para tener intergénico arriba del todo. 
```{r}
ScaleDiscretePositionFunc <- ggproto(
  "ScaleDiscretePositionReversed", ScaleDiscretePosition,
  get_limits = function(self) {
    if (self$is_empty()) {
      c(0, 1)
    } else if (is.null(self$limits)) {
      self$range$range
    } else if (is.function(self$limits)) {
      self$limits(self$range$range)
    } else {
      integer(0)
    }
  }
)

scale_x_discrete2 <- function(..., expand = waiver(), position = "bottom") {
  sc <- discrete_scale(c("x", "xmin", "xmax", "xend"), "position_d", identity, ..., expand = expand, guide = "none", position = position, super = ScaleDiscretePositionFunc)
  sc$range_c <- ggplot2:::continuous_range()
  sc
}

```

## Load data 

¡Depende de lo que vaya a correr no tengo que cargar todos los datos!

## A) Empiracal data

```{r}

c_ll_ki_n013_diversity_c_ll_no_n008_diversity <- read.table (paste0 (wd_input, "c_ll_ki_n013_diversity_c_ll_no_n008_diversity.txt"), row.names = NULL, sep =";", dec=",", header=T, stringsAsFactors = F, comment.char="")
c_ll_ki_n013_diversity_c_ll_po_n008_diversity <- read.table (paste0 (wd_input, "c_ll_ki_n013_diversity_c_ll_po_n008_diversity.txt"), row.names = NULL, sep =";", dec=",", header=T, stringsAsFactors = F, comment.char="" )
c_lp_sm_n019_diversity_c_lp_do_n012_diversity <- read.table (paste0 (wd_input, "c_lp_sm_n019_diversity_c_lp_do_n012_diversity.txt"), row.names = NULL, sep =";", dec=",", header=T, stringsAsFactors = F, comment.char="" )
all_comparisons_diversity_bootleneck_vs_non_bootleneck <- read.table (paste0 (wd_input, "all_comparisons_diversity_bootleneck_vs_non_bootleneck.txt"), row.names = NULL, sep =";", dec=",", header=T, stringsAsFactors = F, comment.char="" )

# Sort factors
c_ll_ki_n013_diversity_c_ll_no_n008_diversity$feature <- factor (c_ll_ki_n013_diversity_c_ll_no_n008_diversity$feature, levels=c("lncRNA_promoter", "UCNE", "ncRNA", "lncRNA_intron", "lncRNA_exons", "3UTR", "Intron", "CDS","5UTR", "Gen_promoter", "Intergenic"))
c_ll_ki_n013_diversity_c_ll_po_n008_diversity$feature <- factor (c_ll_ki_n013_diversity_c_ll_po_n008_diversity$feature, levels=c("lncRNA_promoter", "UCNE", "ncRNA", "lncRNA_intron", "lncRNA_exons", "3UTR", "Intron", "CDS","5UTR", "Gen_promoter", "Intergenic"))
c_lp_sm_n019_diversity_c_lp_do_n012_diversity$feature <- factor (c_lp_sm_n019_diversity_c_lp_do_n012_diversity$feature, levels=c("lncRNA_promoter", "UCNE", "ncRNA", "lncRNA_intron", "lncRNA_exons", "3UTR", "Intron", "CDS","5UTR", "Gen_promoter", "Intergenic"))

all_comparisons_diversity_bootleneck_vs_non_bootleneck$feature <- factor (all_comparisons_diversity_bootleneck_vs_non_bootleneck$feature, levels=c("lncRNA_promoter", "UCNE", "ncRNA", "lncRNA_intron", "lncRNA_exons", "3UTR", "Intron", "CDS","5UTR", "Gen_promoter", "Intergenic"))


density_min_values_pairwise <- read.table(paste(wd, "density_min_values_pairwise.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") 
density_min_values_watterson <- read.table(paste(wd, "density_min_values_watterson.csv", sep=""), header=T, na.strings=c("NA", "na"), stringsAsFactors=F, row.names=NULL, sep=";", dec=".") 

```

## B) Summarized data

```{r}
# Stats per pop
stats_all_pops <- read.table(paste(wd_input,"stats_df_pop_comparison.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )

# Stats interaction
stats_all_pops_feature<- read.table(paste(wd_input,"stats_all_pops_feature.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_region <- read.table(paste(wd_input,"stats_all_pops_region.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_recomb <- read.table(paste(wd_input,"stats_all_pops_recomb.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_chr <- read.table(paste(wd_input,"stats_all_pops_chr.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_feature_chr <- read.table(paste(wd_input,"stats_all_pops_feature_chr.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_feature_region <- read.table(paste(wd_input,"stats_all_pops_feature_region.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_feature_recomb <- read.table(paste(wd_input,"stats_all_pops_feature_recomb.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_region_chr <- read.table(paste(wd_input,"stats_all_pops_region_chr.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_region_recomb <- read.table(paste(wd_input,"stats_all_pops_region_recomb.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_all_pops_recomb_chr <- read.table(paste(wd_input,"stats_all_pops_recomb_chr.txt",sep=""), row.names = NULL, sep =";", dec=",", header=T )

#interaction_stats_all_pops_region_chr_cat_info <-  rbind (interaction_stats_c_ll_ki_n013_stats_region_chr_c_ll_po_n008_stats_region_chr ,interaction_stats_c_ll_ki_n013_stats_region_chr_c_ll_no_n008_stats_region_chr, interaction_stats_c_lp_sm_n019_stats_region_chr_c_lp_do_n012_stats_region_chr) %>% dplyr::full_join(., cat_chr_info_divergence, by="chr") 

# Stats per pop
stats_df_pop <- read.table(paste(wd_input_stats,"stats_diversity_per_pop.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T)
stats_df_pop_feature <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_region <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_recomb.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_feature_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_region_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region_chr.tsv",sep=""), row.names = NULL, sep =";", dec=",", header=T )
stats_df_pop_recomb_chr <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_recomb_chr.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_feature_region <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_region.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_feature_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_feature_recomb.tsv",sep=""),row.names = NULL, sep =";", dec="," , header=T)
stats_df_pop_region_recomb <- read.table(paste(wd_input_stats,"stats_diversity_per_pop_per_region_recomb.tsv",sep=""), row.names = NULL, sep =";", dec="," , header=T)
```

#### Order

```{r}
stats_all_pops_chr$comparison <- factor (stats_all_pops_chr$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))

stats_all_pops_region_chr$comparison <- factor (stats_all_pops_region_chr$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))

stats_all_pops_region_chr$region <- factor (stats_all_pops_region_chr$region, levels=c("Interstitial", "Centromere", "Telomere_2m"))

stats_all_pops_feature$comparison <- factor (stats_all_pops_feature$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))

stats_all_pops_feature$feature <- factor (stats_all_pops_feature$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))

stats_all_pops_chr$comparison <- factor (stats_all_pops_chr$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))

stats_all_pops_region$comparison <- factor (stats_all_pops_region$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))


```


## ----------
# A) Empirical data


## *Dataset categories prep
```{r}
DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity")

for (DATAFRAME in DATAFRAMES) 
{
  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
  
  # Pi
  
  min_non_bootleneck_pop_value_pairwise <- as.numeric(density_min_values_pairwise %>% filter(pop==POP1) %>% .[1,2] )
  min_bootleneck_pop_value_pairwise <- as.numeric(density_min_values_pairwise %>% filter(pop==POP2) %>% .[1,2] )

  data_diversity_categories_pairwise <- get(DATAFRAME) %>% 
  mutate (evolution_category=
            ifelse (log10(pairwise_ave.x)>min_non_bootleneck_pop_value_pairwise & log10(pairwise_ave.y)>min_bootleneck_pop_value_pairwise, "D_D_pairwise",
            ifelse (log10(pairwise_ave.x)>min_non_bootleneck_pop_value_pairwise & log10(pairwise_ave.y)<min_bootleneck_pop_value_pairwise, "D_ND_pairwise",
            ifelse (log10(pairwise_ave.x)<min_non_bootleneck_pop_value_pairwise & log10(pairwise_ave.y)<min_bootleneck_pop_value_pairwise, "ND_ND_pairwise",
            ifelse (log10(pairwise_ave.x)<min_non_bootleneck_pop_value_pairwise & log10(pairwise_ave.y)>min_bootleneck_pop_value_pairwise, "ND_D_pairwise",NA))))) %>% 
  mutate (diversity_category=
            ifelse (log10(pairwise_ave.x)>min_non_bootleneck_pop_value_pairwise, "HD_pairwise",
            ifelse (log10(pairwise_ave.x)<min_non_bootleneck_pop_value_pairwise, "LD_pairwise",NA)))

data_diversity_categories_pairwise$evolution_category <- factor (data_diversity_categories_pairwise$evolution_category, levels=c("D_ND_pairwise", "D_D_pairwise", "ND_ND_pairwise", "ND_D_pairwise"))
data_diversity_categories_pairwise$diversity_category <- factor (data_diversity_categories_pairwise$diversity_category, levels=c("HD_pairwise", "LD_pairwise"))

  assign(paste0(POP1, "_diversity_", POP2, "_diversity_categories_pairwise"), data_diversity_categories_pairwise)

   # Watterson
  
  min_non_bootleneck_pop_value_watterson <- as.numeric(density_min_values_watterson %>% filter(pop==POP1) %>% .[1,2] )
  min_bootleneck_pop_value_watterson <- as.numeric(density_min_values_watterson %>% filter(pop==POP2) %>% .[1,2] )

  data_diversity_categories_watterson <- get(DATAFRAME) %>% 
  dplyr::mutate (evolution_category=
            ifelse ( (log10 (watterson_ave.x) > min_non_bootleneck_pop_value_watterson & log10(watterson_ave.y) > min_bootleneck_pop_value_watterson), "D_D_watterson",
            ifelse ( (log10 (watterson_ave.x) > min_non_bootleneck_pop_value_watterson & log10(watterson_ave.y) < min_bootleneck_pop_value_watterson), "D_ND_watterson",
            ifelse ( (log10 (watterson_ave.x) < min_non_bootleneck_pop_value_watterson & log10(watterson_ave.y) < min_bootleneck_pop_value_watterson), "ND_ND_watterson",
            ifelse ( (log10 (watterson_ave.x) < min_non_bootleneck_pop_value_watterson & log10(watterson_ave.y) > min_bootleneck_pop_value_watterson), "ND_D_watterson",NA))))) %>% 
    
  dplyr::mutate (diversity_category=
            ifelse (log10 (watterson_ave.x) > min_non_bootleneck_pop_value_watterson, "HD_watterson",
            ifelse (log10 (watterson_ave.x) < min_non_bootleneck_pop_value_watterson, "LD_watterson",NA)))

data_diversity_categories_watterson$evolution_category <- factor (data_diversity_categories_watterson$evolution_category, levels=c("D_ND_watterson", "D_D_watterson", "ND_ND_watterson", "ND_D_watterson"))
data_diversity_categories_watterson$diversity_category <- factor (data_diversity_categories_watterson$diversity_category, levels=c("HD_watterson", "LD_watterson"))

  assign(paste0(POP1, "_diversity_", POP2, "_diversity_categories_watterson"), data_diversity_categories_watterson)

   div_recom_GC_by_categ_summary_watterson <- 
             data_diversity_categories_watterson %>% dplyr::filter (type_chr=="autosomes") %>% 
             dplyr::group_by(diversity_category, evolution_category) %>%
             dplyr::summarise(
                       total_count=n()) 
  
    }

###########################################################################
## sort factors

# Pi
c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise$feature <- factor (c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))
c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise$feature  <- factor (c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))
c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise$feature  <- factor (c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))

#Watterson
c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson$feature <- factor (c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))
c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson$feature  <- factor (c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))
c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson$feature  <- factor (c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))

```

# Statistical analysis
Empleo el wilcox.test (por defecto tiene paired=F). Es un test para medidas no parametricas para comparación de rank. 

Explicacion: https://en.wikipedia.org/wiki/Mann–Whitney_U_test
https://data.library.virginia.edu/the-wilcoxon-rank-sum-test/
Check also: https://stats.stackexchange.com/questions/250468/wilcoxon-mann-whitney-test-giving-surprising-results

In statistics, the Mann–Whitney U test (also called the Mann–Whitney–Wilcoxon (MWW), Wilcoxon rank-sum test, or Wilcoxon–Mann–Whitney test) is a nonparametric test of the null hypothesis that it is equally likely that a randomly selected value from one sample will be less than or greater than a randomly selected value from a second sample.

Unlike the t-test it does not require the assumption of normal distributions. It is nearly as efficient as the t-test on normal distributions.

This test can be used to determine whether two independent samples were selected from populations having the same distribution; a similar nonparametric test used on dependent samples is the Wilcoxon signed-rank test.

-->  “true location shift is not equal to 0”. That’s another way of saying “the distribution of one population is shifted to the left or right of the other,” which implies different medians.
--> The Wilcoxon statistic is returned as W. This is NOT an estimate of the difference in medians.

--> wilcox.test function provides this information when we set conf.int = TRUE.

wilcox.test(weight ~ company, data = dat, conf.int = TRUE)

	Wilcoxon rank sum test

data:  weight by company
W = 13, p-value = 0.04988
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -8.5 -0.1
sample estimates:
difference in location 
                 -4.65 

This returns a “difference in location” measure of -4.65. The documentation for the wilcox.test function states this “does not estimate the difference in medians (a common misconception) but rather the median of the difference between a sample from x and a sample from y.”

Nota: https://stackoverflow.com/questions/54842012/different-p-value-in-ggplot2-stat-compare-means-and-wilcox-test
According to help(wilcox.test), if the samples contain less than 50 values (as in your case), the exact p-value is computed (unless you specify otherwise).

stat_compare_means has a method.args argument but it doesn't seem to pass the exact = TRUE specification correctly. Instead you can compute the p-value exactly how you want it first and then add it to the plot
Por tanto, los p-values son una aproximacion normal (In one case the p-value is exact and in the other it is a normal approximation.)



### U-MW diversity lost (HD)

First we test for normality. I already did that, no need to do it again. 

```{r}

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson")

for (DATAFRAME in DATAFRAMES) {
  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

  DATAFRAME_filtered <- get(DATAFRAME) %>% filter(., diversity_category==paste0("HD_",DIVERSITY_MEASURE_CONSIDERED))
  
  ggplot(DATAFRAME_filtered, aes_string(x=paste0(DIVERSITY_MEASURE_CONSIDERED,"_ave.x", sep="" ))) + 
  geom_density() +
  ggsave(paste0(wd_output,paste0("diversity_lost_UMW/",POP1,"_",POP2,"_",DIVERSITY_MEASURE_CONSIDERED,"_HD_density.pdf")))
  
  if (DIVERSITY_MEASURE_CONSIDERED=="pairwise") {
results_pairwise <- shapiro.test(DATAFRAME_filtered$pairwise_ave.x[0:5000])
a <- results_pairwise$p.value
write.csv(a,paste0(wd_output, "diversity_lost_UMW/",POP1,"_",POP2,"_",DIVERSITY_MEASURE_CONSIDERED,"_HD_normality_p_value.csv"))
  }
  
  if (DIVERSITY_MEASURE_CONSIDERED=="watterson") {
results_watteson <- shapiro.test(DATAFRAME_filtered$watterson_ave.x[0:5000])
b <- results_watteson$p.value
write.csv(b,paste0(wd_output, "diversity_lost_UMW/",POP1,"_",POP2,"_",DIVERSITY_MEASURE_CONSIDERED,"_HD_normality_p_value.csv"))
  }

  # Añadimos una categoría que sería, evaluación de perdida de watterson, pero basandonos en los cuadrantes de pi
  
  if (DIVERSITY_MEASURE_CONSIDERED=="watterson") {
results_watteson2 <- shapiro.test(DATAFRAME_filtered$pairwise_ave.x[0:5000])
b <- results_watteson2$p.value
write.csv(b,paste0(wd_output, "diversity_lost_UMW/",POP1,"_",POP2,"_",DIVERSITY_MEASURE_CONSIDERED,"_pairwise_evaluated_HD_normality_p_value.csv"))
  }
  
    
}


```

```{r}

ruta_UMW_test_diversity <- "diversity_lost_UMW/"


DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson")

# Factor order: D_ND_pairwise D_D_pairwise ND_ND_pairwise ND_D_pairwise
# HD --> D_ND_pairwise D_D_pairwise
# LD --> ND_ND_pairwise ND_D_pairwise

for (DATAFRAME in DATAFRAMES) 
  {

  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

DATAFRAME_filtered <- get(DATAFRAME) %>% filter(., diversity_category==paste0("HD_",DIVERSITY_MEASURE_CONSIDERED))

# Diversity:
  if (DIVERSITY_MEASURE_CONSIDERED=="pairwise") 
        {
    
diversity.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(pairwise_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy)  

diversity.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(pairwise_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 
}


  if (DIVERSITY_MEASURE_CONSIDERED=="watterson") 
        {
diversity.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(watterson_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy)  

diversity.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(watterson_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# Definimos cuadrantes para watterson pero los evaluamos en función de pi. Es decir, iro si los cuadrantes que han perdido o ganado diversidad para watterson, están influidos por la pi inicial. 

diversity.greater.PI <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(pairwise_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy)  

diversity.less.PI <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(pairwise_ave.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

lista_results_diversity_melted.PI <- melt(list(diversity.greater.PI , diversity.less.PI)) %>% 
  mutate(genomic_variable="pairwise_evaluation_watterson_quadrants") %>% 
  spread(., variable, value) %>% 
  mutate (SIG=ifelse(p.value<0.05, "SIGN", "no")) %>% 
  mutate (comparison=paste0(POP1,"_", POP2)) %>% 
  mutate (method="wilcox.test") %>% 
  mutate (pop1_pop2 = ifelse (comparison == "c_ll_ki_n013_c_ll_po_n008", "Kirov_Poland",
          ifelse (comparison == "c_ll_ki_n013_c_ll_no_n008", "Kirov_Norway",
          ifelse (comparison == "c_lp_sm_n019_c_lp_do_n012", "Andujar_Donana", "NA")))) %>% 
  select (-comparison)

write.table(format(lista_results_diversity_melted.PI, digits=8, scientific=T), paste(wd_output,ruta_UMW_test_diversity,POP1,"_",POP2,"_features_UMW_test_pairwise_evaluation_watterson_quadrants_HD.txt",sep=""), row.names = F, quote=F)

}

lista_results_diversity_melted <- melt(list(diversity.greater , diversity.less)) %>% mutate(genomic_variable=DIVERSITY_MEASURE_CONSIDERED) %>% spread(., variable, value) %>% 
  mutate (SIG=ifelse(p.value<0.05, "SIGN", "no")) %>% 
  mutate (comparison=paste0(POP1,"_", POP2)) %>% 
  mutate (method="wilcox.test") %>% 
  mutate (pop1_pop2 = ifelse (comparison == "c_ll_ki_n013_c_ll_po_n008", "Kirov_Poland",
          ifelse (comparison == "c_ll_ki_n013_c_ll_no_n008", "Kirov_Norway",
          ifelse (comparison == "c_lp_sm_n019_c_lp_do_n012", "Andujar_Donana", "NA")))) %>% 
  select (-comparison)

write.table(format(lista_results_diversity_melted, digits=8, scientific=T), paste(wd_output,ruta_UMW_test_diversity,POP1,"_",POP2,"_features_UMW_test_",DIVERSITY_MEASURE_CONSIDERED,"_HD.txt",sep=""), row.names = F, quote=F)

rm(diversity.greater, diversity.less, lista_results_diversity_melted)

}

```

En mi local


```{bash}
cat /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/diversity_lost_UMW/c_ll_ki_n013_c_ll_no_n008_features_UMW_test_pairwise_evaluation_watterson_quadrants_HD.txt <(tail -n +2 /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/diversity_lost_UMW/c_ll_ki_n013_c_ll_po_n008_features_UMW_test_pairwise_evaluation_watterson_quadrants_HD.txt ) <(tail -n +2 /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/diversity_lost_UMW/c_lp_sm_n019_c_lp_do_n012_features_UMW_test_pairwise_evaluation_watterson_quadrants_HD.txt )  > /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/diversity_lost_UMW/all_pops_features_UMW_test_pairwise_evaluation_watterson_quadrants_HD.txt 
```


### U-MW test evolutionary category vs genomic variables

Aquí sí hago el test de U-MW por categoría y feature. 
```{r}
ruta_UMW_test <- "genomic_variable_diversity_category_UMT_boxplot_mean/"

DATAFRAMES=c( "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson")

DIVERSITY_CATEGORIES=c("LD")
# Factor order: D_ND_pairwise D_D_pairwise ND_ND_pairwise ND_D_pairwise
# HD --> D_ND_pairwise D_D_pairwise
# LD --> ND_ND_pairwise ND_D_pairwise

for (DATAFRAME in DATAFRAMES) 
  {
  for (DIVERSITY_CATEGORY in DIVERSITY_CATEGORIES)
  {
  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

DATAFRAME_filtered=get(DATAFRAME) %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")) %>% filter (type_chr=="autosomes") 

# Divergence:
divergence.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(divergence ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
divergence.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(divergence ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# Recombination
recombination.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(recombination_rate ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
recombination.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(recombination_rate ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# GC content
GC_content.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(GC_content ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
GC_content.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(GC_content ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# Size
informative_sites.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(informative_sites.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
informative_sites.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(informative_sites.x ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# rvis_score
rvis_score.greater <- wilcox.test(rvis_score ~ evolution_category, data = DATAFRAME_filtered %>% filter(feature=="CDS"), conf.int = TRUE, alternative = "greater")  %>% tidy(.)

rvis_score.less <- wilcox.test(rvis_score ~ evolution_category, data = DATAFRAME_filtered %>% filter(feature=="CDS"), conf.int = TRUE, alternative = "less")  %>% tidy(.)

# RVIS (Residual Variation Intolerance Score) is a gene-based score intended to help in the interpretation of human sequence data. The intolerance score in its current form is based upon allele frequency data as represented in whole exome sequence data from the NHLBI-ESP6500 data set. The score is designed to rank genes in terms of whether they have more or less common functional genetic variation relative to the genome wide expectation given the amount of apparently neutral variation the gene has. A gene with a positive score has more common functional variation, and a gene with a negative score has less and is referred to as "intolerant". By convention we rank all genes in order from most intolerant to least. As an example, a gene such as ATP1A3 has a RVIS score of -1.53 and a percentile of 3.37%, meaning it is amongst the 3.37% most intolerant of human genes. Depending on what disease area you are a studying, you may way to consider either intolerant genes (neurodevelomental disease) or tolerant genes (some immunological diseases) as better candidates. Data sourced from: EVS-v.0.0.14, (June 20, 2012)


lista_results_divergence_melted <- melt(list(divergence.greater , divergence.less)) %>% mutate(genomic_variable="divergence") %>% spread(., variable, value)
lista_results_recombination_melted <- melt(list(recombination.greater, recombination.less)) %>% mutate(genomic_variable="recombination") %>% spread(., variable, value)
lista_results_GC_content_melted <- melt(list(GC_content.greater, GC_content.less)) %>% mutate(genomic_variable="GC_content") %>% spread(., variable, value)
lista_results_size <- melt (list (informative_sites.greater, informative_sites.less)) %>% mutate(genomic_variable="informative_sites") %>% spread(., variable, value)
lista_results_rvis <- melt (list (rvis_score.greater, rvis_score.less)) %>% mutate(genomic_variable="rvis_score") %>% spread(., variable, value) %>% mutate (L2="CDS")



resultados_U_Man_Whitney <- rbind(lista_results_divergence_melted, lista_results_recombination_melted, lista_results_GC_content_melted, lista_results_size, lista_results_rvis) %>% mutate (SIG=ifelse(p.value<0.05, "SIGN", "no")) %>% mutate (comparison=paste0(POP1,"_", POP2)) %>% mutate (method="wilcox.test")

write.table(format(resultados_U_Man_Whitney, digits=8, scientific=T), paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_features_UMW_test_",DIVERSITY_MEASURE_CONSIDERED,"_",DIVERSITY_CATEGORY, ".txt",sep=""), row.names = F, quote=F, dec=",")

rm(divergence.greater, divergence.less, recombination.greater, recombination.less , GC_content.greater, GC_content.less, lista_results_divergence_melted,  lista_results_recombination_melted,  lista_results_GC_content_melted, resultados_U_Man_Whitney)

}
}

```

En mi local hago lo siguiente:

```{bash}
cat /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/genomic_variable_diversity_category_UMT_boxplot_mean/c_lp_sm_n019_c_lp_do_n012_features_UMW_test_watterson_LD.txt <(tail -n +2 /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/genomic_variable_diversity_category_UMT_boxplot_mean/c_ll_ki_n013_c_ll_po_n008_features_UMW_test_watterson_LD.txt ) <(tail -n +2 /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/genomic_variable_diversity_category_UMT_boxplot_mean/c_ll_ki_n013_c_ll_no_n008_features_UMW_test_watterson_LD.txt )  > /Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/pop_comparison/genomic_variable_diversity_category_UMT_boxplot_mean/all_comparisons_features_UMW_test_watterson_LD.txt 
```



### *Boxplot categories vs. genomic category

```{r}
ruta_UMW_test <- "genomic_variable_diversity_category_UMT_boxplot_mean/"

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson" )

DIVERSITY_CATEGORIES=c("HD","LD")
# Factor order: D_ND_pairwise D_D_pairwise ND_ND_pairwise ND_D_pairwise
# HD --> D_ND_pairwise D_D_pairwise
# LD --> ND_ND_pairwise ND_D_pairwise
  
GENOMIC_VARIABLES=c("divergence", "recombination_rate", "GC_content", "informative_sites.x", "rvis_score")
TEST_TYPES=c("less", "greater")
#fun_mean <- function(x){return(round(data.frame(y=mean(x),label=mean(x,na.rm=T)),digit=5))}

for (DATAFRAME in DATAFRAMES) 
{
  for (DIVERSITY_CATEGORY in DIVERSITY_CATEGORIES)
  {
    for (GENOMIC_VARIABLE in GENOMIC_VARIABLES)
    {
      for (TEST_TYPE in TEST_TYPES)
      {
        POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
        POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
        
        DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
          dplyr::select(evolution_category) %>% 
          dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
          head(.,1) %>% 
          dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                                            ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
          select(diversity) %>% 
          .[1,1]
        
        # Filtro mi DF para las categorías con las que estamos trabajando
        DATAFRAME_filtered=get(DATAFRAME) %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))
        
        # Selecciono cuales van a ser mis comparaciones (eg. ND-ND vs ND-D)
        my_comparisons <- unique(DATAFRAME_filtered$evolution_category) %>% as.vector() %>% list()
        
        # Selecciono los límites del plot
        if (GENOMIC_VARIABLE=="divergence") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$divergence)$stats  # Compute lower and upper whisker limits
        }
        
        if (GENOMIC_VARIABLE=="recombination_rate") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$recombination_rate)$stats  # Compute lower and upper whisker limits
        }
        
        if (GENOMIC_VARIABLE=="GC_content") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$GC_content)$stats  # Compute lower and upper whisker limits
        }
        
          if (GENOMIC_VARIABLE=="informative_sites.x") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$informative_sites.x)$stats  # Compute lower and upper whisker limits
        }
         
              if (GENOMIC_VARIABLE=="rvis_score") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$rvis_score)$stats  # Compute lower and upper whisker limits
        }  
           
        # Mean general ##

           ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
           #coord_cartesian(ylim = c(0,sts[5])) +
           stat_summary(aes(colour=evolution_category),fun.data = "mean_cl_boot", size = 0.5) +
           scale_colour_viridis_d(option="cividis") +
           #geom_signif(comparisons = my_comparisons, test="wilcox.test", y_position=sts[4], tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE)) + 
           theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
           ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_mean_se_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        # 
        # ## Mean feature ##
        # 
           ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
           #coord_cartesian(ylim = c(0,sts[5])) +
           facet_grid(col = vars(feature) ) +
           stat_summary(aes(colour=evolution_category),fun.data = "mean_cl_boot", size = 0.5) +
           scale_colour_viridis_d(option="cividis") +
           #geom_signif(comparisons = my_comparisons, test="wilcox.test", y_position=sts[4], tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE)) + 
           theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
           ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_mean_se_features_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
        ## Boxplot general ##
        
        ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
          geom_boxplot(aes(colour=evolution_category),outlier.shape = NA) +
          scale_colour_viridis_d(option="cividis") +
          coord_cartesian(ylim = c(0,max(sts)*2.3)) +
          geom_signif(comparisons = my_comparisons, test="wilcox.test",  y_position=max(sts)*2.1, tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE), map_signif_level=TRUE)  + 
          theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
          ggplot2::ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_boxplot_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
        
        ## Boxplot por feature ##
        
        ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
          facet_grid(col = vars(feature) ) +
          geom_boxplot(aes(colour=evolution_category),outlier.shape = NA) +
          scale_colour_viridis_d(option="cividis") +
          coord_cartesian(ylim = c(0,max(sts)*2.3)) +
          geom_signif(comparisons = my_comparisons, test="wilcox.test",  y_position=max(sts)*2.1, tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE), map_signif_level=TRUE)  + 
          theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
          ggplot2::ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_boxplot_features_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
      }
    }
  }
}


```

### *Genomic variables against informative sites.

Para ver si como pasa con la diversidad, el cambio en cualquiera de las variables genómicas está mediado por el tamaño de la unidad. 
```{r}
DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson" )
GENOMIC_VARIABLES=c("divergence", "recombination_rate", "GC_content", "rvis_score")

for (GENOMIC_VARIABLE in GENOMIC_VARIABLES)
{
for (DATAFRAME in DATAFRAMES)
{
  
     POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
     POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
     
ggplot (get(DATAFRAME) %>% filter(feature %in% c("Intergenic","CDS")) , aes_string("informative_sites.x", GENOMIC_VARIABLE)) +
  geom_point(alpha = 1/10) +
  geom_smooth() +
  facet_wrap (~feature, scales = "free")
ggsave (paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_vs_informative_sites.pdf",sep=""))

}
}
```

### Contingency tables LD

#### Low recombination & region

Tengo que calcular cuantas unidades relativas al total teng con recombination menos que 0.5 en ND-ND y en ND-D. 
```{r}
DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson" )

for (DATAFRAME in DATAFRAMES)
{

   POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
   POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
 

cases_ND_ND <- get(DATAFRAME) %>% 
  filter (diversity_category=="LD_watterson") %>% 
  mutate (recombination_category=ifelse(recombination_rate <= 0.5,"below_0.5", "above_0.5")) %>% 
  dplyr::select (., "evolution_category", "recombination_category", "feature", "region") %>% 
  droplevels(.$evolutionary_category) %>% 
  group_by(feature) 

cases_ND_ND_recomb_contingency <-  xtabs(~ evolution_category+recombination_category+feature, cases_ND_ND)
cases_ND_ND_region_contingency <-  xtabs(~ evolution_category+region+feature, cases_ND_ND)

write.table(cases_ND_ND_recomb_contingency, paste(wd_output,POP1,"_",POP2,"_low_recombination_in_ND_ND_vs_ND_D_watterson.txt",sep=""), row.names = F, quote=F, dec=",") 

write.table(cases_ND_ND_region_contingency, paste(wd_output,POP1,"_",POP2,"_regions_in_ND_ND_vs_ND_D_watterson.txt",sep=""), row.names = F, quote=F, dec=",") 


chi_results_recomb <- cases_ND_ND %>%
  group_by(feature) %>%
  select (-region) %>% 
  nest() %>% 
  mutate(
    chi_sq_results = map(data, ~ chisq.test(.x$evolution_category, .x$recombination_category)),
    tidied = map(chi_sq_results, tidy)
  ) %>%
  unnest(tidied, .drop = TRUE)


write.table(chi_results_recomb, paste(wd_output,POP1,"_",POP2,"_low_recombination_in_ND_ND_vs_ND_D_watterson_chi_square_results.txt",sep=""), row.names = F, quote=F, dec=",") 


chi_results_region <- cases_ND_ND %>%
  mutate (region=as.factor(region)) %>% 
  #filter (feature == "CDS" | feature == "Intergenic" | ) %>% 
  select (-recombination_category) %>% 
  group_by(feature) %>%
  nest() %>% 
  mutate(
    chi_sq_results = map(data, ~ chisq.test(.x$evolution_category, .x$region)),
    tidied = map(chi_sq_results, tidy)
  ) 


u <-chi_results_region$chi_sq_results
e <-lapply(u,function(x)
corrplot(u$residuals, is.cor = FALSE)
 )
lapply(M, function(x){
corrplot(x$r, type="upper",tl.col="black", tl.cex=0.7,tl.srt=45, col=col,
         p.mat = x$P, insig = "blank", sig.level = 0.01)
}


chi_results_region %>%
  unnest(tidied, .drop = TRUE)


write.table(chi_results_region, paste(wd_output,POP1,"_",POP2,"_regions_in_ND_ND_vs_ND_D_watterson_chi_square_results.txt",sep=""), row.names = F, quote=F, dec=",") 

library("gplots")

my.list[[FEATURE_NUMBER]] <- merge(FEATURE, chisq_results$p.value)
  #chisq_results$expected
  if (chisq_results$p.value > 0.05)
  {jpeg(paste0(wd_output_contingency,"/no_sig/", POPS,"_",DIVERSITY_MEASURE,"_",FEATURE ,".jpg"))
  corrplot(chisq_results$residuals, is.cor = FALSE)
  dev.off()} 
  else {
  jpeg(paste0(wd_output_contingency,"/sig/", POPS,"_",DIVERSITY_MEASURE,"_",FEATURE ,".jpg"))
  corrplot(chisq_results$residuals, is.cor = FALSE)
  dev.off()

test <-summary(cases_ND_ND_region_contingency)
chisq.test(cases_ND_ND_region_contingency)

chi.result(cases_ND_ND_region_contingency)
kk <- filter(cases_ND_ND, feature=="UCNE")
test <-chisq.test(droplevels(kk$evolution_category), kk$region)
test <-chisq.test(cases_ND_ND_region_contingency$region)

corrplot(test$residuals, is.cor = FALSE)
}
```



### Recombination per feature

```{r}
library(reshape2)
dat.m <- melt(c_ll_ki_n013_diversity_c_ll_no_n008_diversity,id.vars='unique_id', measure.vars=c('Freq','Freq.1','Freq.2'))
library(ggplot2)
p <- ggplot(dat.m) +
      geom_boxplot(aes(x=ID, y=value, color=variable))
```


## ----------

# B) Summarized data

## *Comparison NB & B
### *Chr
```{r}
DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")
  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
# Chr  
    ## Lynx lynx
ggplot(data=stats_all_pops_chr %>% filter(chr!="chrX") , mapping = aes_string(paste(DIVERSITY_MEASURE,".x",sep=""), paste(DIVERSITY_MEASURE,".y",sep="")))+
  facet_wrap (~comparison) +
  scale_fill_viridis(option="magma") +
  scale_colour_viridis(option="magma") +
  geom_abline(colour="grey", linetype = "dashed") +
  geom_point (colour="grey80", fill="grey80") +
  geom_smooth (method = "lm", colour="grey30", fill="grey60", size=0.5) +
  theme(axis.title.x = element_blank(), legend.position = "none") +
  coord_equal(ratio=1) +
  stat_poly_eq(aes(label =  paste("atop(",stat(adj.rr.label),",",stat(eq.label), ")", sep = "")), 
               formula = formula, parse = TRUE,size = 2.5, label.x.npc = "right",
                  label.y.npc = "bottom") +
ggsave (paste(wd_output,"panel_chr_", DIVERSITY_MEASURE,".pdf", sep=""), width = 169, height = 160, units = "mm")
  }

#With X chr

DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")
  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
# Chr  
    ## Lynx lynx
ggplot(data=stats_all_pops_chr, mapping = aes_string(paste(DIVERSITY_MEASURE,".x",sep=""), paste(DIVERSITY_MEASURE,".y",sep=""), color="type_chr", fill="type_chr"))+
  facet_wrap (~comparison) +
   scale_fill_viridis_d(option = "plasma")+
    scale_colour_viridis_d(option = "plasma")+
  geom_abline(colour="grey", linetype = "dashed") +
  geom_point () +
  geom_smooth (method = "lm", size=0.5) +
  theme(axis.title.x = element_blank(), legend.position = "none") +
  coord_equal(ratio=1) +
  
ggsave (paste(wd_output,"panel_chr_autosomes_vs_x", DIVERSITY_MEASURE,".pdf", sep=""), width = 169, height = 160, units = "mm")
}
```

### *Feature

```{r}
DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")
  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
ggplot (stats_all_pops_feature %>% filter(type_chr!="chrX"), 
          mapping=aes_string(paste(DIVERSITY_MEASURE, ".x", sep=""), paste(DIVERSITY_MEASURE, ".y", sep=""))) +
    facet_wrap (~comparison) +
    scale_fill_viridis_d(option = "plasma")+
    scale_colour_viridis_d(option = "plasma")+
    geom_abline (colour="grey", linetype = "dashed") +
    geom_smooth (method = "lm", colour="grey80", fill="grey80", size=0.5) +
    geom_point (aes(fill=feature, colour=feature)) +
    coord_equal(ratio=1) +
    theme(legend.position = "none") +
    geom_text_repel(aes(label = feature),
                  box.padding   = 1, 
                  point.padding = 0.1,
                  size          = 2.5,
                  force         = 80,
                  segment.size  = 0.2,
                  segment.color = 'grey50') +
  ggsave(paste(wd_output, "panel_feature_", DIVERSITY_MEASURE,".pdf", sep=""), width = 167, height = 80, units = "mm")
  }

## X chr
DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")
  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
ggplot (stats_all_pops_feature %>% filter(type_chr=="chrX"), 
          mapping=aes_string(paste(DIVERSITY_MEASURE, ".x", sep=""), paste(DIVERSITY_MEASURE, ".y", sep=""))) +
    facet_wrap (~comparison) +
    scale_fill_viridis_d(option = "plasma")+
    scale_colour_viridis_d(option = "plasma")+
    geom_abline (colour="grey", linetype = "dashed") +
    geom_smooth (method = "lm", colour="grey80", fill="grey80", size=0.5) +
    geom_point (aes(fill=feature, colour=feature)) +
    #coord_equal(ratio=1) +
    theme(legend.position = "none") +
    scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
    scale_y_continuous(labels = function(x) format(x, scientific = TRUE)) +
    geom_text_repel(aes(label = feature),
                  box.padding   = 1, 
                  point.padding = 0.1,
                  size          = 2.5,
                  force         = 80,
                  segment.size  = 0.2,
                  segment.color = 'grey50') +
    theme(axis.text.x = element_text(angle = 45)) +
  ggsave(paste(wd_output, "panel_feature_chrX_", DIVERSITY_MEASURE,".pdf", sep=""), width = 167, height = 80, units = "mm")
}
```


### *Region & chr

```{r}

DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")

  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
# Region and chr 
    ## Lynx lynx
ggplot(data=stats_all_pops_region_chr %>% filter(chr!="chrX") , mapping = aes_string(paste(DIVERSITY_MEASURE,".x",sep=""), paste(DIVERSITY_MEASURE,".y",sep="")))+
  facet_grid (region~comparison) +
  geom_abline(colour="grey", linetype = "dashed") +
  geom_point (colour="grey80", fill="grey80") +
  geom_smooth (method = "lm", colour="grey30", fill="grey60", size=0.5) +
  coord_equal(ratio=1) +
  theme (axis.text.x = element_text(angle = 45)) +
  stat_poly_eq(formula = my.formula,coef.digits=2 ,
                aes(label = paste(..eq.label..)), 
                parse = TRUE,label.y = 0) +  
  stat_poly_eq(formula = my.formula, 
                aes(label = paste(..rr.label..)), 
                parse = TRUE, label.y = 0.1) +
ggsave (paste(wd_output,"panel_region_", DIVERSITY_MEASURE,".pdf", sep=""), width = 180, height = 190, units = "mm")


# # Prueba en la que coloreamos por porcentage de genes. 
# region_chr_colour_recomb2 <- ggplot(data=interaction_stats_all_pops_region_chr_cat_info %>% filter(chr!="chrX") , mapping = aes_string(paste(DIVERSITY_MEASURE,".y",sep=""),                                                                                                                        paste(DIVERSITY_MEASURE,".x",sep=""), fill="percentage_genes", colour="percentage_genes"))+
#   facet_grid (region~comparison) +
#   scale_fill_viridis(option="magma") +
#   scale_colour_viridis(option="magma") +
#   geom_abline(colour="grey", linetype = "dashed") +
#   geom_point () + #colour="grey80", fill="grey80"
#   geom_smooth (method = "lm", colour="grey30", fill="grey60", size=0.5) +
#   theme(axis.title.x = element_blank() ) + # legend.position = "none"
#   coord_equal(ratio=1) +
#   stat_poly_eq(aes(label =  paste("atop(",stat(adj.rr.label),",",stat(eq.label), ")", sep = "")),
#                formula = formula, parse = TRUE,size = 2.5, label.x.npc = "right",
#                   label.y.npc = "bottom")
# Esta guardado como: comparisons_region_vs_chr_wmean_pairwise_ave_colour_gene_percentage


 

# figure <- ggarrange(region_chr_colour_recomb, feature_colour_feature, 
#          labels = c("A", "B"),
#          ncol = 1, nrow = 2,   heights = c(2, 1))
#ggsave(paste(wd_output, "panel_regions_feature_", DIVERSITY_MEASURE,".pdf", sep=""), figure, width = 20, height = 30, units = "cm")

}

```




## Recombination

```{r}
ggplot() + 
  geom_point (data=subset(stats_all_pops_recomb %>% filter (type_chr=="autosomes"), !is.na(recomb)), 
              aes(x=recomb, y=wmean_delta_watterson_per_unit_normalized_zero, fill=comparison, colour=comparison)) +
  geom_segment (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="autosomes"), !is.na(recomb)), 
                aes(x = recomb, xend=recomb, y = wmean_delta_watterson_per_unit_normalized_zero-se_wmean_delta_watterson_per_unit_normalized_zero_boot, yend = wmean_delta_watterson_per_unit_normalized_zero+se_wmean_delta_watterson_per_unit_normalized_zero_boot, colour=comparison)) +
  facet_wrap(~comparison, ncol=1, scales="free_y") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  ggsave(paste(wd_output,"recombination_categories_vs_delta_watterson_autosomes.pdf",sep=""))

ggplot() + 
  geom_point (data=subset(stats_all_pops_recomb %>% filter (type_chr=="chrX"), !is.na(recomb)), 
              aes(x=recomb, y=wmean_delta_watterson_per_unit_normalized_zero, fill=comparison, colour=comparison)) +
  geom_segment (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="chrX"), !is.na(recomb)), 
                aes(x = recomb, xend=recomb, y = wmean_delta_watterson_per_unit_normalized_zero-se_wmean_delta_watterson_per_unit_normalized_zero_boot, yend = wmean_delta_watterson_per_unit_normalized_zero+se_wmean_delta_watterson_per_unit_normalized_zero_boot, colour=comparison)) +
  facet_wrap(~comparison, ncol=1, scales="free_y") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  ggsave(paste(wd_output,"recombination_categories_vs_delta_watterson_chrX.pdf",sep=""))


ggplot() + 
  geom_point (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="autosomes"), !is.na(recomb)), 
              aes(x=recomb, y=wmean_delta_pairwise_per_unit_normalized_zero, fill=comparison, colour=comparison)) +
  geom_segment (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="autosomes"), !is.na(recomb)), 
                aes(x = recomb, xend=recomb, y = wmean_delta_pairwise_per_unit_normalized_zero-se_wmean_delta_pairwise_per_unit_normalized_zero_boot, yend = wmean_delta_pairwise_per_unit_normalized_zero+se_wmean_delta_pairwise_per_unit_normalized_zero_boot, colour=comparison)) +
   facet_wrap(~comparison, ncol=1, scales="free_y") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  ggsave(paste(wd_output,"recombination_categories_vs_delta_pairwise_autosomes.pdf",sep=""))


ggplot() + 
  geom_point (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="chrX"), !is.na(recomb)), 
              aes(x=recomb, y=wmean_delta_pairwise_per_unit_normalized_zero, fill=comparison, colour=comparison)) +
  geom_segment (data=subset(stats_df_pop_comparison_recomb %>% filter (type_chr=="chrX"), !is.na(recomb)), 
                aes(x = recomb, xend=recomb, y = wmean_delta_pairwise_per_unit_normalized_zero-se_wmean_delta_pairwise_per_unit_normalized_zero_boot, yend = wmean_delta_pairwise_per_unit_normalized_zero+se_wmean_delta_pairwise_per_unit_normalized_zero_boot, colour=comparison)) +
   facet_wrap(~comparison, ncol=1, scales="free_y") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  ggsave(paste(wd_output,"recombination_categories_vs_delta_pairwise_chrX.pdf",sep=""))


```


##---------



## Contingency test

```{r}
library(corrplot)

wd <- "/Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/"
wd_output_contingency <- "/Users/marialucenaperez/Owncloud/publico/PhD/WG_diversity/ANGSD/sfs/contingency_test_ND_D_distribution_regions"

# c_ll_ki_n013_c_ll_no_n008_pairwise_percentage_of_ND_D_in_different_regions.tsv
# c_ll_ki_n013_c_ll_po_n008_pairwise_percentage_of_ND_D_in_different_regions.tsv
# c_lp_sm_n019_c_lp_do_n012_pairwise_percentage_of_ND_D_in_different_regions.tsv
# c_ll_ki_n013_c_ll_po_n008_watterson_percentage_of_ND_D_in_different_regions.tsv
# c_ll_ki_n013_c_ll_no_n008_watterson_percentage_of_ND_D_in_different_regions.tsv
# c_lp_sm_n019_c_lp_do_n012_watterson_percentage_of_ND_D_in_different_regions.tsv

DATA="c_lp_sm_n019_c_lp_do_n012_pairwise_percentage_of_ND_D_in_different_regions.tsv"


DATAFRAME <- read.table(paste0(wd, DATA),stringsAsFactors = F) %>% select (-total_count)
POPS=substr(DATA, 1, 25)
DIVERSITY_MEASURE=substr(DATA, 27, 34)
  
DATAFRAME_SPLITTED <- split(DATAFRAME, DATAFRAME$feature) 
number_of_data_frames <- unique(DATAFRAME$feature) %>% length()
my.list <- vector("list", number_of_data_frames)


for(FEATURE_NUMBER in 1:number_of_data_frames)
{
  print (FEATURE_NUMBER)
  temp_data <- DATAFRAME_SPLITTED[[FEATURE_NUMBER]] 
  temp_data[is.na(temp_data)] <- 0
  FEATURE <- as.character(temp_data %>% select(feature) %>% .[1,])
  temp_data_matrix <- temp_data %>% select (-feature, -D_D, -D_ND)
  rownames(temp_data_matrix) <- temp_data_matrix[,1] 
  temp_data_matrix_ready <- temp_data_matrix %>%  select (-1) 
  chisq_results <- chisq.test( temp_data_matrix_ready) 
  my.list[[FEATURE_NUMBER]] <- merge(FEATURE, chisq_results$p.value)
  #chisq_results$expected
  if (chisq_results$p.value > 0.05)
  {jpeg(paste0(wd_output_contingency,"/no_sig/", POPS,"_",DIVERSITY_MEASURE,"_",FEATURE ,".jpg"))
  corrplot(chisq_results$residuals, is.cor = FALSE)
  dev.off()} 
  else {
  jpeg(paste0(wd_output_contingency,"/sig/", POPS,"_",DIVERSITY_MEASURE,"_",FEATURE ,".jpg"))
  corrplot(chisq_results$residuals, is.cor = FALSE)
  dev.off()
  }
  
  rm(temp_data, temp_data_matrix, chisq_results, FEATURE)
 }


df_results <- do.call("rbind", my.list) 
write.table( df_results, paste0(wd_output_contingency,"/", POPS,"_",DIVERSITY_MEASURE,".tsv"))
rm (df_results)
```


#------------







### U-MW test diversity category vs genomic variables

```{r}

ruta_UMW_test <- "genomic_variable_diversity_category_UMT_boxplot_mean/"

library(broom)

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson")

# Divergence
#Con este código lo que hacemos es que por cada feature (by) corremos el wilcox test. El resultado se guarda en una tabla que despues se va a limpiar usando la función tidy, y luego haciendo list juntamos todos los resultados, hasta que finalmente escribimos una tabla usando melt. 

for (DATAFRAME in DATAFRAMES) {
  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

# Divergence:
divergence.greater <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(divergence ~ diversity_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
divergence.less <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(divergence ~ diversity_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# Recombination
recombination.greater <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(recombination_rate ~ diversity_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
recombination.less <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(recombination_rate ~ diversity_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# GC content
GC_content.greater <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(GC_content ~ diversity_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
GC_content.less <- with(get(DATAFRAME),
       by(get(DATAFRAME), feature,
          function(x) wilcox.test(GC_content ~ diversity_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

lista_results_divergence_melted <- melt(list(divergence.greater , divergence.less)) %>% mutate(genomic_variable="divergence") %>% spread(., variable, value)
lista_results_recombination_melted <- melt(list(recombination.greater, recombination.less)) %>% mutate(genomic_variable="recombination") %>% spread(., variable, value)
lista_results_GC_content_melted <- melt(list(GC_content.greater, GC_content.less)) %>% mutate(genomic_variable="GC_content") %>% spread(., variable, value)

resultados_U_Man_Whitney <- rbind(lista_results_divergence_melted, lista_results_recombination_melted, lista_results_GC_content_melted)

write.table(format(resultados_U_Man_Whitney, digits=8, scientific=T), paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_features_UMW_test_",DIVERSITY_MEASURE_CONSIDERED,".txt",sep=""), row.names = F)

rm(divergence.greater, divergence.less, recombination.greater, recombination.less , GC_content.greater, GC_content.less, lista_results_divergence_melted,  lista_results_recombination_melted,  lista_results_GC_content_melted, resultados_U_Man_Whitney)

}


for (DATAFRAME in DATAFRAMES) 
  {

  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

# Divergence
divergence.greater <- wilcox.test(divergence ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "greater") %>% tidy(.) 
divergence.less <- wilcox.test(divergence ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "less") %>%  tidy(.) 
lista_divergence <- melt(list(divergence.greater, divergence.less)) %>% mutate(genomic_variable="divergence") %>% spread(., variable, value)

# Recombination
recombination.greater <- wilcox.test(recombination_rate ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "greater") %>%  tidy(.)
recombination.less <- wilcox.test(recombination_rate ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "less") %>%  tidy(.)
lista_recombination <- melt(list(recombination.greater, recombination.less)) %>% mutate(genomic_variable="recombination") %>% spread(., variable, value)

# GC content
GC_content.greater <- wilcox.test(GC_content ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "greater") %>%  tidy(.)
GC_content.less <- wilcox.test(GC_content ~ diversity_category, data = get(DATAFRAME), conf.int = TRUE, alternative = "less") %>%  tidy(.)
lista_GC_content <- melt(list(GC_content.greater, GC_content.less)) %>% mutate(genomic_variable="GC_content") %>% spread(., variable, value)

resultados_U_Man_Whitney <- rbind(lista_divergence, lista_recombination, lista_GC_content)

write.table(format(resultados_U_Man_Whitney, digits=8, scientific=T), paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_UMW_test_",DIVERSITY_MEASURE_CONSIDERED,".txt",sep=""), row.names = F)

rm(divergence.greater, divergence.less, recombination.greater, recombination.less , GC_content.greater, GC_content.less, lista_divergence,  lista_recombination,  lista_GC_content, resultados_U_Man_Whitney)

}


```

### U-MW test evolutionary category vs genomic variables


```{r}
ruta_UMW_test <- "genomic_variable_diversity_category_UMT_boxplot_mean/"


DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson")

DIVERSITY_CATEGORIES=c("HD","LD")
# Factor order: D_ND_pairwise D_D_pairwise ND_ND_pairwise ND_D_pairwise
# HD --> D_ND_pairwise D_D_pairwise
# LD --> ND_ND_pairwise ND_D_pairwise

for (DATAFRAME in DATAFRAMES) 
  {
  for (DIVERSITY_CATEGORY in DIVERSITY_CATEGORIES)
  {
  
  POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

  DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
    dplyr::select(evolution_category) %>% 
    dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
    head(.,1) %>% 
    dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                              ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
    select(diversity) %>% 
    .[1,1]

DATAFRAME_filtered=get(DATAFRAME) %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))

# Divergence:
divergence.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(divergence ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
divergence.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(divergence ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# Recombination
recombination.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(recombination_rate ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
recombination.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(recombination_rate ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

# GC content
GC_content.greater <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(GC_content ~ evolution_category, data = x, conf.int = TRUE, alternative = "greater"))) %>% lapply(., tidy) 
GC_content.less <- with(DATAFRAME_filtered,
       by(DATAFRAME_filtered, feature,
          function(x) wilcox.test(GC_content ~ evolution_category, data = x, conf.int = TRUE, alternative = "less"))) %>% lapply(., tidy) 

lista_results_divergence_melted <- melt(list(divergence.greater , divergence.less)) %>% mutate(genomic_variable="divergence") %>% spread(., variable, value)
lista_results_recombination_melted <- melt(list(recombination.greater, recombination.less)) %>% mutate(genomic_variable="recombination") %>% spread(., variable, value)
lista_results_GC_content_melted <- melt(list(GC_content.greater, GC_content.less)) %>% mutate(genomic_variable="GC_content") %>% spread(., variable, value)

resultados_U_Man_Whitney <- rbind(lista_results_divergence_melted, lista_results_recombination_melted, lista_results_GC_content_melted)

write.table(format(resultados_U_Man_Whitney, digits=8, scientific=T), paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_features_UMW_test_",DIVERSITY_MEASURE_CONSIDERED,"_",DIVERSITY_CATEGORY, ".txt",sep=""), row.names = F)

rm(divergence.greater, divergence.less, recombination.greater, recombination.less , GC_content.greater, GC_content.less, lista_results_divergence_melted,  lista_results_recombination_melted,  lista_results_GC_content_melted, resultados_U_Man_Whitney)

}
}

```


#### Boxplot categories vs. divergence

```{r}

library("ggsignif")
library("ggpubr")
library("cowplot")
library(ggplot2); theme_set(theme_minimal())

ruta_UMW_test <- "genomic_variable_diversity_category_UMT_boxplot_mean/"

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_pairwise", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_pairwise", "c_ll_ki_n013_diversity_c_ll_no_n008_diversity_categories_watterson", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity_categories_watterson", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity_categories_watterson" )

DIVERSITY_CATEGORIES=c("HD","LD")
# Factor order: D_ND_pairwise D_D_pairwise ND_ND_pairwise ND_D_pairwise
# HD --> D_ND_pairwise D_D_pairwise
# LD --> ND_ND_pairwise ND_D_pairwise

GENOMIC_VARIABLES=c("divergence", "recombination_rate", "GC_content")
TEST_TYPES=c("less", "greater")
#fun_mean <- function(x){return(round(data.frame(y=mean(x),label=mean(x,na.rm=T)),digit=5))}

for (DATAFRAME in DATAFRAMES) 
{
  for (DIVERSITY_CATEGORY in DIVERSITY_CATEGORIES)
  {
    for (GENOMIC_VARIABLE in GENOMIC_VARIABLES)
    {
      for (TEST_TYPE in TEST_TYPES)
      {
        POP1=get(DATAFRAME) %>% dplyr::select(pop.x) %>% dplyr::mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
        POP2=get(DATAFRAME) %>% dplyr::select(pop.y) %>% dplyr::mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
        
        DIVERSITY_MEASURE_CONSIDERED = get(DATAFRAME) %>% 
          dplyr::select(evolution_category) %>% 
          dplyr::mutate(evolution_category=as.character(evolution_category)) %>% 
          head(.,1) %>% 
          dplyr::mutate(diversity = ifelse (grepl("pairwise", evolution_category), "pairwise",
                                            ifelse (grepl("watterson", evolution_category), "watterson", NA))) %>% 
          select(diversity) %>% 
          .[1,1]
        
        # Filtro mi DF para las categorías con las que estamos trabajando
        DATAFRAME_filtered=get(DATAFRAME) %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))
        
        # Selecciono cuales van a ser mis comparaciones (eg. ND-ND vs ND-D)
        my_comparisons <- unique(DATAFRAME_filtered$evolution_category) %>% as.vector() %>% list()
        
        # Selecciono los límites del plot
        if (GENOMIC_VARIABLE=="divergence") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$divergence)$stats  # Compute lower and upper whisker limits
        }
        
        if (GENOMIC_VARIABLE=="recombination_rate") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$recombination_rate)$stats  # Compute lower and upper whisker limits
        }
        
        if (GENOMIC_VARIABLE=="GC_content") 
        {
          sts <- boxplot.stats((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep="")))$GC_content)$stats  # Compute lower and upper whisker limits
        }
        
               
        # Mean general ##

        #   ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
        #   coord_cartesian(ylim = c(0,sts[5])) +
        #   stat_summary(aes(colour=evolution_category),fun.data = "mean_cl_boot", size = 0.5) +
        #   scale_colour_viridis_d(option="cividis") +
        #   geom_signif(comparisons = my_comparisons, test="wilcox.test", y_position=sts[4], tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE)) + 
        #   theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
        #   ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_mean_se_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        # 
        # ## Mean feature ##
        # 
        #   ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
        #   coord_cartesian(ylim = c(0,sts[5])) +
        #   facet_grid(col = vars(feature) ) +
        #   stat_summary(aes(colour=evolution_category),fun.data = "mean_cl_boot", size = 0.5) +
        #   scale_colour_viridis_d(option="cividis") +
        #   geom_signif(comparisons = my_comparisons, test="wilcox.test", y_position=sts[4], tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE)) + 
        #   theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
        #   ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_mean_se_features_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
        ## Boxplot general ##
        
        ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
          geom_boxplot(aes(colour=evolution_category),outlier.shape = NA) +
          scale_colour_viridis_d(option="cividis") +
          coord_cartesian(ylim = c(0,max(sts)*2.3)) +
          geom_signif(comparisons = my_comparisons, test="wilcox.test",  y_position=max(sts)*2.1, tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE), map_signif_level=TRUE)  + 
          theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
          ggplot2::ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_boxplot_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
        
        ## Boxplot por feature ##
        
        ggplot((DATAFRAME_filtered %>% filter (type_chr=="autosomes") %>% filter(diversity_category==paste(DIVERSITY_CATEGORY,"_",DIVERSITY_MEASURE_CONSIDERED,sep=""))), aes_string("evolution_category",GENOMIC_VARIABLE)) +
          facet_grid(col = vars(feature) ) +
          geom_boxplot(aes(colour=evolution_category),outlier.shape = NA) +
          scale_colour_viridis_d(option="cividis") +
          coord_cartesian(ylim = c(0,max(sts)*2.3)) +
          geom_signif(comparisons = my_comparisons, test="wilcox.test",  y_position=max(sts)*2.1, tip_length=0, linetype="blank", test.args=list(alternative=TEST_TYPE), map_signif_level=TRUE)  + 
          theme(strip.text.x = element_text(angle = 75), axis.text.x = element_blank()) +
          ggplot2::ggsave(paste(wd_output,ruta_UMW_test,POP1,"_",POP2,"_",GENOMIC_VARIABLE,"_boxplot_features_",DIVERSITY_CATEGORY,"_", DIVERSITY_MEASURE_CONSIDERED, "_alternative_",TEST_TYPE ,".pdf",sep=""))
        
      }
    }
  }
}

```



#----------------------------------------
# Global
### Summary stats 

Como queremos hacerlo relativo al tamaño de cada unidad primero tengo que calcular la tabla con los estadísticos (mean and stdev basado en bootstrapping), y despues plotearla. No puedo hacer las dos cosas en un mismo plot de R como hacía antes usando stat_summary (mean_cl_boot).

### Stats

```{r}
  
# Low values to zero:
  # WATTERSON
  # Mean dr distribution
  # Calculo el valor medio por feature para sacar el maximo y mínimo para luego establecer los límites del plot
mean <- get(DATAFRAME) %>% 
             dplyr::group_by(feature) %>%
             dplyr::summarise(
            delta_watterson_per_unit_normalized_zero_ave=mean(delta_watterson_per_unit_normalized_zero,na.rm=TRUE)) %>% select(-feature) %>% as.vector(.)
# Plot de la media global
 a <- ggplot (get(DATAFRAME)) +
    stat_summary(aes(x=comparison, y=delta_watterson_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
    labs(title=paste("Mean for the Delta comparing Watterson:", POPULATION2,"vs", POPULATION1, sep =" ")) +    
   theme(axis.title = element_blank())+
    coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) 
 
 b <-  ggplot (get(DATAFRAME)) +
    stat_summary(aes(x=feature, y=delta_watterson_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
   labs(x ="Feature", y = "Mean") +
    coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) 

 a.b <- ggarrange(a, b,  
          labels = c("A", "B"),
          ncol = 1, nrow = 2, align="hv", heights=c(1,6))
  
    ggsave (paste(wd_output,DATAFRAME,"_delta_zero_watterson_mean_feature.pdf", sep=""),a.b, width = 20, height = 25, units = "cm")

    ## Plot combinado de mean per feature, con una raya para la media
    ggplot (get(DATAFRAME)) +
      geom_hline(yintercept=mean(get(DATAFRAME)$delta_watterson_per_unit_normalized_zero, na.rm=TRUE), colour ="grey60")  +
      stat_summary(aes(x=feature, y=delta_watterson_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
      labs(x ="Feature", y = "Mean") +
      coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) +
      ggsave (paste(wd_output,DATAFRAME,"_delta_zero_watterson_mean_feature_one_pannel.pdf", sep=""), width = 20, height = 25, units = "cm")
   ## Plot combinado de mean per feature, con una raya para la media wo -1 and +1
    ggplot (get(DATAFRAME) %>% filter(delta_watterson_per_unit_normalized_zero!=1 & delta_watterson_per_unit_normalized_zero!=-1)) +
      geom_hline(data=get(DATAFRAME) %>% filter(delta_watterson_per_unit_normalized_zero!=1 & delta_watterson_per_unit_normalized_zero!=-1), aes(yintercept=mean(delta_watterson_per_unit_normalized_zero, na.rm=TRUE)), color ="grey60")  +
      stat_summary(aes(x=feature, y=delta_watterson_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
      labs(x ="Feature", y = "Mean") +
      coord_flip() +
      ggsave (paste(wd_output,DATAFRAME,"_delta_zero_watterson_mean_feature_one_pannel_wo_1_&_minu1.pdf", sep=""), width = 20, height = 25, units = "cm")
    # Pi # 
    # Mean dr distribution
  mean <- get(DATAFRAME) %>% 
             dplyr::group_by(feature) %>%
             dplyr::summarise(
            delta_pairwise_per_unit_normalized_zero_ave=mean(delta_pairwise_per_unit_normalized_zero,na.rm=TRUE)) %>% select(-feature) %>% as.vector(.)

 a <- ggplot (get(DATAFRAME)) +
    stat_summary(aes(x=comparison, y=delta_pairwise_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
    labs(title=paste("Mean for the Delta comparing pairwise:", POPULATION2,"vs", POPULATION1, sep =" ")) +    
   theme(axis.title = element_blank())+
    coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) 
 
 b <-  ggplot (get(DATAFRAME)) +
    stat_summary(aes(x=feature, y=delta_pairwise_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
   labs(x ="Feature", y = "Mean") +
    coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) 

 
 a.b <- ggarrange(a, b,  
          labels = c("A", "B"),
          ncol = 1, nrow = 2, align="hv", heights=c(1,6))
  
    ggsave (paste(wd_output,DATAFRAME,"_delta_zero_pairwise_mean_feature.pdf", sep=""),a.b, width = 20, height = 25, units = "cm")
    
    ## Plot combinado de mean per feature, con una raya para la media
    
    ggplot (get(DATAFRAME)) +
      geom_hline(yintercept=mean(get(DATAFRAME)$delta_pairwise_per_unit_normalized_zero, na.rm=TRUE), colour ="grey60")  +
      stat_summary(aes(x=feature, y=delta_pairwise_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
      labs(x ="Feature", y = "Mean") +
      coord_flip(ylim = c(min(mean)-0.1, max(mean)+0.1)) +
      ggsave (paste(wd_output,DATAFRAME,"_delta_zero_pairwise_mean_feature_one_pannel.pdf", sep=""), width = 20, height = 25, units = "cm")
    
       ## Plot combinado de mean per feature, con una raya para la media wo -1 and +1
    
    ggplot (get(DATAFRAME) %>% filter(delta_pairwise_per_unit_normalized_zero!=1 & delta_pairwise_per_unit_normalized_zero!=-1)) +
      geom_hline(data=get(DATAFRAME) %>% filter(delta_pairwise_per_unit_normalized_zero!=1 & delta_pairwise_per_unit_normalized_zero!=-1), aes(yintercept=mean(delta_pairwise_per_unit_normalized_zero, na.rm=TRUE)), color ="grey60")  +
      stat_summary(aes(x=feature, y=delta_pairwise_per_unit_normalized_zero), fun.data = "mean_cl_boot", colour = "black", size = 0.5) +
      labs(x ="Feature", y = "Mean") +
      coord_flip() +
      ggsave (paste(wd_output,DATAFRAME,"_delta_zero_pairwise_mean_feature_one_pannel_wo_1_&_minu1.pdf", sep=""), width = 20, height = 25, units = "cm")
   
    


```



##### Plot panel comparison MB & EB

```{r}

library (ggpubr)
library(ggrepel)
library(ggpmisc)
library("gridExtra")

formula <- y ~ x

interaction_stats_all_pops_region_chr_cat_info$comparison <- factor (interaction_stats_all_pops_region_chr_cat_info$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))
interaction_stats_all_pops_feature$comparison <- factor (interaction_stats_all_pops_feature$comparison, levels=c("Kirov-NE_Poland", "Kirov-Norway", "Andujar-Donana"))

DIVERSITY_MEASURES=c( "wmean_pairwise_ave", "wmean_watterson_ave")

  for (DIVERSITY_MEASURE in DIVERSITY_MEASURES)
{
    
# Region and chr 
    
    ## Lynx lynx
    
region_chr_colour_recomb <- ggplot(data=interaction_stats_all_pops_region_chr_cat_info %>% filter(chr!="chrX") , mapping = aes_string(paste(DIVERSITY_MEASURE,".y",sep=""), paste(DIVERSITY_MEASURE,".x",sep="")))+
  facet_grid (region~comparison) +
  scale_fill_viridis(option="magma") +
  scale_colour_viridis(option="magma") +
  geom_abline(colour="grey", linetype = "dashed") +
  geom_point (colour="grey80", fill="grey80") +
  geom_smooth (method = "lm", colour="grey30", fill="grey60", size=0.5) +
  theme(axis.title.x = element_blank(), legend.position = "none") +
  coord_equal(ratio=1) +
  stat_poly_eq(aes(label =  paste("atop(",stat(adj.rr.label),",",stat(eq.label), ")", sep = "")), 
               formula = formula, parse = TRUE,size = 2.5, label.x.npc = "right",
                  label.y.npc = "bottom")

# # Prueba en la que coloreamos por porcentage de genes. 
# region_chr_colour_recomb2 <- ggplot(data=interaction_stats_all_pops_region_chr_cat_info %>% filter(chr!="chrX") , mapping = aes_string(paste(DIVERSITY_MEASURE,".y",sep=""),                                                                                                                        paste(DIVERSITY_MEASURE,".x",sep=""), fill="percentage_genes", colour="percentage_genes"))+
#   facet_grid (region~comparison) +
#   scale_fill_viridis(option="magma") +
#   scale_colour_viridis(option="magma") +
#   geom_abline(colour="grey", linetype = "dashed") +
#   geom_point () + #colour="grey80", fill="grey80"
#   geom_smooth (method = "lm", colour="grey30", fill="grey60", size=0.5) +
#   theme(axis.title.x = element_blank() ) + # legend.position = "none"
#   coord_equal(ratio=1) +
#   stat_poly_eq(aes(label =  paste("atop(",stat(adj.rr.label),",",stat(eq.label), ")", sep = "")),
#                formula = formula, parse = TRUE,size = 2.5, label.x.npc = "right",
#                   label.y.npc = "bottom")
# Esta guardado como: comparisons_region_vs_chr_wmean_pairwise_ave_colour_gene_percentage

# Plot Feature MB vs EB

feature_colour_feature <- ggplot (interaction_stats_all_pops_feature %>% filter(type_chr!="chrX"), 
          mapping=aes_string(paste(DIVERSITY_MEASURE, ".y", sep=""), paste(DIVERSITY_MEASURE, ".x", sep=""))) +
    facet_wrap (~comparison) +
    scale_fill_viridis_d() +
    scale_colour_viridis_d() +
    geom_abline (colour="grey", linetype = "dashed") +
    geom_smooth (method = "lm", colour="grey80", fill="grey80", size=0.5) +
    geom_point (aes(fill=feature, colour=feature)) +
    coord_equal(ratio=1) +
    theme(legend.position = "none") +
    geom_text_repel(aes(label = feature),
                  box.padding   = 1, 
                  point.padding = 0.1,
                  size          = 2.5,
                  force         = 80,
                  segment.size  = 0.2,
                  segment.color = 'grey50') 
  

  
  

 figure <- ggarrange(region_chr_colour_recomb, feature_colour_feature, 
          labels = c("A", "B"),
          ncol = 1, nrow = 2,   heights = c(2, 1))
ggsave(paste(wd_output, "panel_regions_feature_", DIVERSITY_MEASURE,".pdf", sep=""), figure, width = 20, height = 30, units = "cm")

}

```

##### Plot chr solo 
No lo corro
```{r}

  ggplot (interaction_stats_all_pops_chr %>% filter(type_chr=="autosome"), mapping=aes_string(paste(DIVERSITY_MEASURE, ".y", sep=""), paste(DIVERSITY_MEASURE, ".x", sep=""), fill="type_chr", colour="type_chr")) +
  geom_point(interaction_stats_all_pops_chr %>% filter(type_chr=="autosome"), mapping=aes_string(paste(DIVERSITY_MEASURE, ".y", sep=""), paste(DIVERSITY_MEASURE, ".x", sep=""), fill="type_chr", colour="type_chr")) +
  geom_smooth (method = "lm", colour="grey80", fill="grey80", size=0.5) +
  geom_abline(colour="grey", linetype = "dashed") +
  facet_wrap (~comparison) +
  stat_poly_eq(formula = my.formula,coef.digits=2 ,
                aes(label = paste(..eq.label..)), 
                parse = TRUE, label.y = 1.05 * min(interaction_stats_all_pops_chr$wmean_watterson_ave.x)) +  
  stat_poly_eq(formula = my.formula, 
                aes(label = paste(..rr.label..)), 
                parse = TRUE, label.y = min(interaction_stats_all_pops_chr$wmean_watterson_ave.x)) +
  scale_color_viridis(discrete=TRUE) +
  scale_fill_viridis(discrete=TRUE) 
```



##### Plot for ratio

No lo corro.
```{r}

DATAFRAMES=c(
"ratio_median_c_ll_ki_n013_stats_feature_c_ll_no_n008_stats_feature", "ratio_median_c_ll_ki_n013_stats_feature_c_ll_po_n008_stats_feature", "ratio_median_c_lp_sm_n019_stats_feature_c_lp_do_n012_stats_feature")

INDEXES=c("ratio_median_watterson_corrected", "ratio_median_pairwise_corrected", "ratio_median_watterson_corrected", "ratio_median_pairwise_corrected", "ratio_median_watterson_corrected_zero", "ratio_median_pairwise_corrected_zero")

for (DATAFRAME in DATAFRAMES)
{
for (INDEX in INDEXES)
  {

ggplot (get(DATAFRAME) %>% filter (type_chr!="chrX"), aes_string("feature", INDEX, fill="type_chr", colour="type_chr")) +
  geom_point() +
  scale_color_viridis(discrete=TRUE) +
  scale_fill_viridis(discrete=TRUE) +
  theme(text=element_text(family = 'Times', size = 12, color = 'black'), axis.text.x = element_text(angle = 45, hjust = 1))
  
  ggsave(paste(wd_output, DATAFRAME,"_", INDEX,".pdf", sep=""))
}



```



# -------------------------

# -------------------------------------------------

#### 3D Plot divergence and recombination vs ratio

```{r}
library(plotly)

# data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity

plot_ly(x=filter(data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity,feature=="UCNE")$divergence,
        y=filter(data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity,feature=="UCNE")$recombination_rate,
        z=filter(data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity,feature=="UCNE")$ratio_watterson_average)


plot_ly(x=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity$divergence,
        y=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity$recombination_rate,
        z=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity$ratio_watterson_average, 
        color = ~data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity$feature, 
        mode = "markers",marker=list(opacity=0.5))
```

####  Plot boot data for each group

No lo hago

```{r}

library(rlang)

DATAFRAMES=c("ratio_boot_c_ll_ki_n013_c_ll_no_n008_feature", "ratio_boot_c_ll_ki_n013_c_ll_po_n008_feature", "ratio_boot_c_lp_sm_n019_c_lp_do_n012_feature") 

# "ratio_boot_c_ll_ki_n013_c_ll_no_n008_chr", "ratio_boot_c_ll_ki_n013_c_ll_po_n008_chr", "ratio_boot_c_lp_sm_n019_c_lp_do_n012_chr",  "ratio_boot_c_ll_ki_n013_c_ll_no_n008_region", "ratio_boot_c_ll_ki_n013_c_ll_po_n008_region", "ratio_boot_c_lp_sm_n019_c_lp_do_n012_region") 

INDICES=c("ratio_watterson_boot", "ratio_pairwise_boot")




myplot3 <- function(data,title){
  for (INDEX in INDICES)
  {
    print (INDEX)
  pdf(paste(wd_output,title,"_",INDEX,".pdf", sep=""))
  plot <- ggplot(data=data, aes_string(x=colnames(data)[1],y=paste("mean_",INDEX,sep=""), fill="chr_type", colour="chr_type")) +
        geom_point() +
        geom_errorbar(aes_string(ymin=paste("mean_",INDEX,"-sd_",INDEX,sep=""), ymax=paste("mean_",INDEX,"+sd_",INDEX,sep=""))) +
        #scale_y_continuous(trans = 'log10') +
        theme_minimal() +  #theme selection for background and lines
        theme(
              axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1))
  # labs(title = paste (title, "scale_y_transformed", sep=""))
  print (plot)
  dev.off() 
    
  }
}


for(i in DATAFRAMES){
  print(myplot3(get(i), i))
}


```
# -------------------------
# -------------------------

# OLD CODE

##### Heatmap for two groups

```{r}
library( gplots)

# ls(pattern="ratio_median")  y selecciono los que tienen comparación a pares
DATAFRAMES=c(
"ratio_median_c_ll_ki_n013_stats_feature_chr_c_ll_no_n008_stats_feature_chr", "ratio_median_c_ll_ki_n013_stats_feature_chr_c_ll_po_n008_stats_feature_chr", "ratio_median_c_lp_sm_n019_stats_feature_chr_c_lp_do_n012_stats_feature_chr")




INDICE=c("ratio_median_pairwise_corrected", "ratio_median_watterson_corrected")
#########################################################
### A) Installing and loading required packages
#########################################################

if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)}

if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
   }

  
  library(eply)

#########################################################
### B) Reading in data and transform it into matrix format
#########################################################
for (DATA in DATAFRAMES)
{  
  for (INDEX in INDICE)
  {

    DATAFRAME=as.data.frame(get(DATA))
    DATAFRAME$chr <- factor (DATAFRAME$chr, levels=c( "chrD4","chrE1","chrE3","chrF1","chrF2","chrB4","chrD2","chrE2","chrA3","chrB3","chrC1","chrD1","chrD3","chrA2","chrB2","chrA1","chrC2","chrB1","chrX"))
    
    DATAFRAME$feature <- factor (DATAFRAME$feature, levels=c("Intergenic", "Gen_promoter", "5UTR", "CDS", "Intron", "3UTR", "lncRNA_promoter", "lncRNA_exons", "lncRNA_intron", "ncRNA", "UCNE"))

    POP1=as.character(DATAFRAME$pop.x[1])
    POP2=as.character(DATAFRAME$pop.y[1])
    
    temp.data.frame.diversity <- reshape2::acast(DATAFRAME , list(names(DATAFRAME)[1], names(DATAFRAME)[3]), value.var=INDEX)
    
    temp.data.frame.counts <- reshape2::acast(DATAFRAME, list(names(DATAFRAME)[1], names(DATAFRAME)[3]), value.var="total_count.x")

   
   
#########################################################
### C) Customizing and plotting the heat map
#########################################################

# creates a 5 x 5 inch image
png(paste(wd_output,"heatmap_",names(DATAFRAME)[1],"_",names(DATAFRAME)[3],"_",INDEX,"_",POP1,"_",POP2,".png", sep=""),    # create PNG for the heat map        
  width = 5*500,        # 5 x 300 pixels
  height = 5*500,
  res = 300,            # 300 pixels per inch
  pointsize = 8)        # smaller font size


# my_palette <- colorRampPalette(c("yellow", "orange", "red"))(n = 1000)
# colors = c(seq(-3,-2,length=100),seq(-2,0.5,length=100),seq(0.5,6,length=100))

    
heatmap.2(temp.data.frame.diversity,
  cellnote = temp.data.frame.counts,  
  col=viridis_pal(), 
  #breaks=colors,
  main = paste(INDEX,POP1,POP2, sep=" "), # heat map title
  notecol="black",      # change font color of cell labels to black
  # density.info="none",  # turns off density plot inside color legend
  trace="none",         # turns off trace lines inside the heat map
  margins =c(20,10),     # widens margins around plot
  # col=my_palette,       # use on color palette defined earlier
  # breaks=col_breaks,    # enable color transition at specified limits
  dendrogram="none", 
  Rowv=FALSE,
  Colv=FALSE
  )           # turn off column clustering
  
dev.off()               # close the PNG device

  }
}  
  


```
 


# ------------------

# ----------------------------------------------------------------
### Delta in a heat map

Primero genero las tablas que me van a servir para el heatmap.

```{r}

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity")

VARIABLES=c("delta_pairwise_per_unit_normalized_zero", "delta_watterson_per_unit_normalized_zero")


for (DATAFRAME in DATAFRAMES) 
{
  for (VARIABLE in VARIABLES)
  {
    
  POP1=get(DATAFRAME) %>% select(pop.x) %>% mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% select(pop.y) %>% mutate(pop.y=as.character(pop.y)) %>% .[1,1] 

# Feature & region
DATAFRAME_FR <- get(DATAFRAME) %>% filter(chr!="chrF1" & chr!="chrF2" & chr!="chrX") %>%             
             dplyr::group_by(region, feature) %>%
             dplyr::summarise(
               total=n(),
               mean_result = mean(get(VARIABLE), na.rm = TRUE),  
               median_result = median(get(VARIABLE), na.rm = TRUE))  


ggplot(DATAFRAME_FR, aes(x = region, y = feature, fill = mean_result)) + 
  geom_tile() + 
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FR , mapping = aes(x = region, y = feature, label=total), size=4) +  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_region_vs_feature_mean.pdf", sep=""), width = 15, height = 15)


ggplot(DATAFRAME_FR, aes(x = region, y = feature, fill = median_result)) + 
  geom_tile() + 
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FR , mapping = aes(x = region, y = feature, label=total), size=4)+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_region_vs_feature_median.pdf", sep=""), width = 15, height = 15)


# Feature & chromosome
DATAFRAME_FC <- get(DATAFRAME)%>% filter(chr!="chrF1" & chr!="chrF2" & chr!="chrX") %>%          
             dplyr::group_by(chr, feature) %>%
             dplyr::summarise(
               total=n(),
               mean_result = mean(get(VARIABLE), na.rm = TRUE),  
               median_result = median(get(VARIABLE), na.rm = TRUE))  

                                           
ggplot(DATAFRAME_FC, aes(x = chr, y = feature, fill = mean_result)) + 
  geom_tile() + 
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FC , mapping = aes(x = chr, y = feature, label=total), size=4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_chr_vs_feature_mean.pdf", sep=""), width = 15, height = 15)


ggplot(DATAFRAME_FC, aes(x = chr, y = feature, fill = median_result)) + 
  geom_tile() + 
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FC , mapping = aes(x = chr, y = feature, label=total), size=4)+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_chr_vs_feature_median.pdf", sep=""), width = 15, height = 15)


# Feature & chromosome & region
DATAFRAME_FCR <- get(DATAFRAME)%>% filter(chr!="chrF1" & chr!="chrF2" & chr!="chrX") %>%          
             dplyr::group_by(chr, feature, region) %>%
             dplyr::summarise(
               total=n(),
               mean_result = mean(get(VARIABLE), na.rm = TRUE),  
               median_result = median(get(VARIABLE), na.rm = TRUE))  
                                           
ggplot(DATAFRAME_FCR, aes(x = region, y = feature, fill = mean_result)) + 
  geom_tile() +
  facet_wrap(~chr) +
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FCR , mapping = aes(x = region, y = feature, label=total), size=4) +  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_region_vs_chr_vs_feature_mean.pdf", sep=""), width = 25, height = 25)


ggplot(DATAFRAME_FCR, aes(x = region, y = feature, fill = median_result)) + 
  geom_tile() + 
  facet_wrap(~chr) +
  scale_fill_viridis(alpha = 0.9) +
  geom_text(DATAFRAME_FCR , mapping = aes(x = region, y = feature, label=total), size=4) +  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
ggsave (paste(wd_output,"heatmap/", POP1, "_",POP2,"_", VARIABLE, "_region_vs_chr_vs_feature_median.pdf", sep=""), width = 25, height = 25)
                       
  }
}

```

### Delta vs genomic variables
```{r}

DATAFRAMES=c("c_ll_ki_n013_diversity_c_ll_no_n008_diversity", "c_ll_ki_n013_diversity_c_ll_po_n008_diversity", "c_lp_sm_n019_diversity_c_lp_do_n012_diversity")

for (DATAFRAME in DATAFRAMES) 
{
  POP1=get(DATAFRAME) %>% select(pop.x) %>% mutate(pop.x=as.character(pop.x)) %>% .[1,1] 
  POP2=get(DATAFRAME) %>% select(pop.y) %>% mutate(pop.y=as.character(pop.y)) %>% .[1,1] 
    # RECOMBINATION WATTERSON
    ggplot (get(DATAFRAME), aes(x=recomb, y=delta_watterson_average_normalized)) +
    stat_summary(fun.data = "mean_cl_boot", size = 0.5) +
    facet_wrap(~feature, scales="free_y")
    ggsave (paste(wd_output,POP1,"_",POP2, "_recombination_vs_delta_zero_watterson.pdf", sep=""))
  
    # DIVERGENCE WATTERSON
  
    ggplot (get(DATAFRAME), aes(x=divergence, y=delta_watterson_per_unit_normalized_zero)) +
    geom_point() +
    ggsave (paste(wd_output,POP1,"_",POP2, "_divergence_vs_delta_zero_watterson.pdf", sep=""))

     # GC_CONTENT WATTERSON
  
    ggplot (get(DATAFRAME), aes(x=GC_content, y=delta_watterson_per_unit_normalized_zero)) +
    geom_point() +
    ggsave (paste(wd_output,POP1,"_",POP2, "_GC_content_vs_delta_zero_watterson.pdf", sep=""))
    
    # RECOMBINATION pairwise
  
    ggplot (get(DATAFRAME), aes(x=recombination_rate, y=delta_pairwise_per_unit_normalized_zero)) +
    geom_point() +
    ggsave (paste(wd_output,POP1,"_",POP2, "_recombination_vs_delta_zero_pairwise.pdf", sep=""))
  
     # DIVERGENCE pairwise
  
    ggplot (get(DATAFRAME), aes(x=divergence, y=delta_pairwise_per_unit_normalized_zero)) +
    geom_point() +
    ggsave (paste(wd_output,POP1,"_",POP2, "_divergence_vs_delta_zero_pairwise.pdf", sep=""))

     # GC_CONTENT pairwise
  
    ggplot (get(DATAFRAME), aes(x=GC_content, y=delta_pairwise_per_unit_normalized_zero)) +
    geom_point() +
    ggsave (paste(wd_output,POP1,"_",POP2, "_GC_content_vs_delta_zero_pairwise.pdf", sep=""))


}
  
```

# ----------------------------------------------------------------
## Otros

### GAM


```{r}
data_set_yedra <- c_ll_ki_n013_diversity_c_ll_no_n008_diversity %>% filter (!is.na(delta_pairwise_per_unit_normalized_zero)) %>% .[sample(1:nrow(.), 2500, replace=FALSE),] %>% select ("scaffold", "start_cero_based", "end", "recombination_rate", "divergence", "GC_content","feature", "delta_pairwise_per_unit_normalized_zero") %>% arrange(., scaffold, start_cero_based) 

write.csv(data_set_yedra, "/Users/marialucenaperez/Desktop/DATAFRAME_test_Maria.csv",row.names = F, quote = F)

```

https://noamross.github.io/gams-in-r-course/chapter1

```{r}
require(gam)
require(ISLR)
attach(Wage)
require("mgcv")


data_set_yedra <- read.table("/Users/marialucenaperez/Desktop/DATAFRAME_test_Maria.csv", header=T, sep=",")
gam1<-gam(wage~s(age,df=6)+s(year,df=6)+education ,data = Wage)
#in the above function s() is the shorthand for fitting smoothing splines 
#in gam() function
summary(gam1)

#Plotting the Model
par(mfrow=c(1,3)) #to partition the Plotting Window
plot(gam1,se = TRUE) 
#se stands for standard error Bands

gam1<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate),data = data_set_yedra)
# When fitting a GAM, use the syntax y ~ s(x) in the model formula to indicate that y should be a smooth, non-linear function of x.
#in the above function s() is the shorthand for fitting smoothing splines 
#in gam() function

summary(gam1)

par(mfrow=c(3,3)) #to partition the Plotting Window
gam.check(gam1)
# Extract the model coefficients
coef(gam1)

#Plotting the Model
par(mfrow=c(1,3)) #to partition the Plotting Window
plot(gam1,se = TRUE,residuals = TRUE, pch = 1) 
#se stands for standard error Bands

gam2<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate, k = 3) ,data = data_set_yedra)
plot(gam2,se = TRUE,residuals = TRUE, pch = 1) 

gam3<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate, k = 20) ,data = data_set_yedra)
plot(gam3,se = TRUE,residuals = TRUE, pch = 1) 


gam4<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate) ,data = data_set_yedra, sp = 0.1)
plot(gam4,se = TRUE,residuals = TRUE, pch = 1) 


gam5<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate) ,data = data_set_yedra, sp = 0.0001)
plot(gam5,se = TRUE,residuals = TRUE, pch = 1) 


gam6<-gam(delta_pairwise_per_unit_normalized_zero~+s(recombination_rate, k=) ,data = data_set_yedra, sp = 0.1)
plot(gam6,se = TRUE,residuals = TRUE, pch = 1) 


```
#### Divergencia

```{r}
gam1<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence),data = data_set_yedra)
# When fitting a GAM, use the syntax y ~ s(x) in the model formula to indicate that y should be a smooth, non-linear function of x.
#in the above function s() is the shorthand for fitting smoothing splines 
#in gam() function

summary(gam1)


# Extract the model coefficients
coef(gam1)

#Plotting the Model
par(mfrow=c(1,3)) #to partition the Plotting Window
plot(gam1,se = TRUE,residuals = TRUE, pch = 1) 
#se stands for standard error Bands

gam2<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence, k = 3) ,data = data_set_yedra)
plot(gam2,se = TRUE,residuals = TRUE, pch = 1) 

gam3<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence, k = 20) ,data = data_set_yedra)
plot(gam3,se = TRUE,residuals = TRUE, pch = 1) 


gam4<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence) ,data = data_set_yedra, sp = 0.1)
plot(gam4,se = TRUE,residuals = TRUE, pch = 1) 


gam5<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence) ,data = data_set_yedra, sp = 0.0001)
plot(gam5,se = TRUE,residuals = TRUE, pch = 1) 


gam6<-gam(delta_pairwise_per_unit_normalized_zero~+s(divergence, k=) ,data = data_set_yedra, sp = 0.1)
plot(gam6,se = TRUE,residuals = TRUE, pch = 1) 

```
#### GC_content

```{r}
gam1<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content),data = data_set_yedra)
# When fitting a GAM, use the syntax y ~ s(x) in the model formula to indicate that y should be a smooth, non-linear function of x.
#in the above function s() is the shorthand for fitting smoothing splines 
#in gam() function

summary(gam1)


# Extract the model coefficients
coef(gam1)

#Plotting the Model
par(mfrow=c(1,3)) #to partition the Plotting Window
plot(gam1,se = TRUE,residuals = TRUE, pch = 1) 
#se stands for standard error Bands

gam2<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content, k = 3) ,data = data_set_yedra)
plot(gam2,se = TRUE,residuals = TRUE, pch = 1) 

gam3<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content, k = 20) ,data = data_set_yedra)
plot(gam3,se = TRUE,residuals = TRUE, pch = 1) 


gam4<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content) ,data = data_set_yedra, sp = 0.1)
plot(gam4,se = TRUE,residuals = TRUE, pch = 1) 


gam5<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content) ,data = data_set_yedra, sp = 0.0001)
plot(gam5,se = TRUE,residuals = TRUE, pch = 1) 


gam6<-gam(delta_pairwise_per_unit_normalized_zero~+s(GC_content, k=) ,data = data_set_yedra, sp = 0.1)
plot(gam6,se = TRUE,residuals = TRUE, pch = 1) 

```

#### Multivariate

```{r}

gam1<-gam(delta_pairwise_per_unit_normalized_zero ~ s(recombination_rate) +s (divergence) +s (GC_content) ,data = (data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity %>% filter(delta_pairwise_per_unit_normalized_zero!=-1 & delta_pairwise_per_unit_normalized_zero!=1)), method = "REML")
# When fitting a GAM, use the syntax y ~ s(x) in the model formula to indicate that y should be a smooth, non-linear function of x.
#in the above function s() is the shorthand for fitting smoothing splines 
#in gam() function

gam1<-gam(delta_pairwise_per_unit_normalized_zero ~ s(recombination_rate) +s (divergence) +s (GC_content) ,data = data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_no_n008_autosomes_diversity , method = "REML")


summary(gam1)


# Extract the model coefficients
coef(gam1)
cols <- sample(c("red","green","pink"),100,TRUE)


#Plotting the Model
par(mfrow=c(3,1)) #to partition the Plotting Window
plot(gam1,se = TRUE,residuals = F, pch=19, cex=0.75, scheme=1, shade=T,shade.col='gray90')




```


# Conclusión GAM

No se ve un patrón claro, y no se nos ocurren trasnformaciones que solucionen este tema. Así que dejamos lo de los modelos agotado por el momento. 


## Modelo mixto generalizado


```{r}
library(lme4)
library(MuMIn)
library(gvlma)

data=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_po_n008_autosomes_diversity


ki.po.model_iteraction <- lmer (ratio_corrected_by_watterson_average_intergenic_per_unit ~ feature * recombination_rate.y * chr + ( 1 | unique_id ), data=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_po_n008_autosomes_diversity, REML=FALSE)

r.squaredGLMM(ki.po.model_iteraction)

summary (ki.po.model_iteraction)

# Of course you should do model simplification or model averaging in an attempt to get a parsimonious model before you do any of this, but I just wanted to flag this up.
anova_model1 <- anova(ki.po.model_iteraction)


mod <- lm(ratio_corrected_by_watterson_average_intergenic_per_unit ~ feature * recombination_rate.y * chr,data=data_diversity_c_ll_ki_n013_autosomes_diversity_c_ll_po_n008_autosomes_diversity)

# Do I understand correctly it is fine the outcome variable  does not need to be normally distributed itself ?
# Just like general linear models, your outcome variable does not need to be normally distributed as a univariate variable. However, LME models assume that the residuals of the model are normally distributed. So a transformation or adding weights to the model would be a way of taking care of this (and checking with diagnostic plots, of course).
par(mfrow = c(2, 2))
plot(mod)
# problema de convergencia!

gvlma::gvlma(mod)

anova(ki.po.model_nointeraction,ki.po.model_iteraction)

mean(ki.po.model_iteraction$residuals)
# https://www.ncbi.nlm.nih.gov/pubmed/21077903


# Super util para iterpretar los plots: https://stats.stackexchange.com/questions/58141/interpreting-plot-lm/65864#65864
mod.stdres = rstandard(mod)
plot(c_ll_ki_n013_diversity_c_ll_po_n008_diversity_for_model$Populations, mod.stdres, 
  ylab="Standardized Residuals") 

ggplot (data=c_ll_ki_n013_diversity, aes(watterson_ave)) +
  geom_histogram()

kk <- c_ll_ki_n013_diversity %>% filter(feature=="CDS")

shapiro.test(randomssubset$watterson_ave)

a <- seq(1, 700000, by = 150) # option 2
randomssubset <- kk[a,]





```


https://stats.stackexchange.com/questions/101274/how-to-interpret-a-qq-plot
https://www.google.com/search?client=safari&rls=en&q=how+to+correct+for+right+skewness+residuals&ie=UTF-8&oe=UTF-8
https://stats.stackexchange.com/questions/135008/how-to-deal-with-non-normally-distributed-residuals
https://stats.stackexchange.com/questions/58141/interpreting-plot-lm/65864#65864
https://stats.stackexchange.com/questions/76226/interpreting-the-residuals-vs-fitted-values-plot-for-verifying-the-assumptions
https://brownmath.com/stat/shape.htm
https://www.researchgate.net/post/how_to_reduce_skewness_and_kurtosis
https://www.researchgate.net/post/how_to_deal_with_skewness


### Assumptions of the model

Assumption 1
The regression model is linear in parameters

```{r}

```

Assumption 2
The mean of residuals is zero





Again, let’s work through this: First, the output reminds you of the model that you
fit. Then, there’s some general summary statistics such as Akaike’s Information
Criterion, the log-Likelihood etc. We won’t go into the meaning of these different
values in this tutorial because these are conceptually a little bit more involved.
Let’s focus on the output for the random effects first:


You’re being reminded of the formula of the two models that you’re comparing.
Then, you find a Chi-Square value, the associated degrees of freedom and the pvalue2.
You would report this result the following way:
“… politeness affected pitch (χ2(1)=11.62, 

