---
title: "pipeline_contemporary_sfs"
output: html_document
---

# Generating SAF 

## Introduction and populations:

I am going to run SAF in my target populations.

Populations:

c_ll_no -->
c_ll_po -->
c_ll_ki -->

c_lp_do -->
c_lp_sm -->

I checked that all bam files were indexed --> OK

## Global variables:


```{r, engine=bash, eval=FALSE}

RUTA=/home/mlucena/ANGSD_analysis 

cd $RUTA/whole_genome_analysis/sfs

#To launch one by one

# POP="c_lp_do_n012"  # <--CHANGE POP HERE
POP="all pops"
screen -S "$POP"_sfs_wg
# screen -S "$POP"_thetas
#POP="all pops"
#POP="c_lp_do_n012"  # <--CHANGE POP HERE

#script "$POP"_sfs_wg.log
# script "$POP"_thetas.log

# POP="c_lp_do_n012"  # <--CHANGE POP HERE

THREADS=15                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!

RUTA=/home/mlucena/ANGSD_analysis/depth_calculus 
ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
RUTA=/home/mlucena/ANGSD_analysis

POPS=(c_lp_sm_n019 c_ll_ki_n013 c_ll_no_n008 c_ll_po_n008 c_lp_do_n012 c_lp_sm_n019)   # <--CHANGE POP HERE

# 6/3/2018 --> Lo he lanzado en un loop que acaba en SAF posprob. 

for POP in  ${POPS[@]}
do

read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated < $RUTA/depth_calculus/"${POP}"_mean_sd_depthGlobal_lynx_per_pop_mean_folds_0.95.csv

N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)

# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
echo $maxDepth_truncated
echo $minDepth_truncated






```

## Unfolded SAF 

```{r, engine=bash, eval=FALSE}


##########################
#  SAF (likelihood):     
##########################

echo "-------$POP----------SAF (likelihood)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b $RUTA/whole_genome_analysis/"$POP".bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth_truncated -setMinDepth $minDepth_truncated


```

## SFS 

```{r, engine=bash, eval=FALSE}

##########################
#  SFS                   #(I dont require the -rf file as the saf already only contains the -rf sites
##########################
echo "-------$POP----------SFS------------------------------------------------------"

$NGSTOOLS/realSFS "$POP".unfolded-lr.saf.idx  -P $THREADS > "$POP".unfolded-lr.sfs


```

##  SAF (postprob) 

```{r, engine=bash, eval=FALSE}


echo "-------$POP----------SAF (postprob)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b $RUTA/whole_genome_analysis/"$POP".bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr.postprob \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth_truncated -setMinDepth $minDepth_truncated \
-doThetas 1 -pest "$POP".unfolded-lr.sfs 

done

```


## SFS graphical representation


First we copy the SFS files to our local folder.

```{bash}
scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/*sfs /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs

# I need to change name as R does not like "-lr.sfs"

cd /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/
for i in *sfs
do 
#echo $i
echo $i "--->" ${i/-lr.sfs/_lr.sfs}
mv $i ${i/-lr.sfs/_lr.sfs}
done


```

Now we run the script. 

```{r}
library(dplyr)
library(magrittr)

wd<-"/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/"

my_files_unfolded = list.files(path = wd, pattern="*sfs$")    
for (i in 1:length(my_files_unfolded)) 
{ assign(my_files_unfolded[i], (scan(paste(wd,my_files_unfolded[i], sep = ""), sep = " ", dec = ".")) %>% .[!is.na(.)])}

norm <- function(x) x/sum(x)

for (i in 1:length (my_files_unfolded))
{
  temp1=(eval(parse(text=my_files_unfolded[i])))
  temp2=norm(temp1[-1])
  temp3=norm(temp2[-c(length(temp2))])
  barplot(temp3,xlab="Chromosomes",
          names=1:length(temp3),ylab="Proportions",main=my_files_unfolded[i],col='grey')
  dev.print(device=postscript, paste("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/",my_files_unfolded[i],".eps", sep =""), onefile=FALSE, horizontal=FALSE)
  dev.off()
  rm(temp1)
  rm(temp2)
  rm(temp3)
}


```


##  Make bed theta        

```{r, engine=bash, eval=FALSE}

echo "-------$POP----------Make bed theta -----------------------------------------"

$NGSTOOLS/thetaStat print $POP.unfolded-lr.postprob.thetas.idx > $POP.printed.stats

# Transform the coordinates to use 0Based bedtools

awk 'NR>1 {print  $1" "$2-1" "$2}' $POP.printed.stats > $POP.0basedcoordinates.borrar # --> 1, 2, 3
# log transformation of watterson and pairwise:
awk 'NR>1 {print  "e("$3")"}' $POP.printed.stats | bc -l > "$POP".watterson.borrar # --> 4
awk 'NR>1 {print  "e("$4")"}' $POP.printed.stats | bc -l > "$POP".pairwise.borrar # --> 5
                                                                                  # --> PAIRWISE - WATTERSON
awk 'NR>1 {print  "e("$5")"}' $POP.printed.stats | bc -l > "$POP".thetaFL.borrar # --> 6
awk 'NR>1 {print  "e("$6")"}' $POP.printed.stats | bc -l > "$POP".thetaH.borrar # --> 7
awk 'NR>1 {print  "e("$7")"}' $POP.printed.stats | bc -l > "$POP".thetaL.borrar # --> 8

# Create a header for the "$POP".transformedThetas file

echo "scaffold position1 position2 watterson pairwise pairwise-watterson thetaFL thetaH thetaL" > "$POP".transformedThetas

paste $POP.0basedcoordinates.borrar "$POP".watterson.borrar "$POP".pairwise.borrar "$POP".thetaFL.borrar "$POP".thetaH.borrar "$POP".thetaL.borrar | \
awk -v OFS='\t'  '{print $1, $2, $3, $4, $5, $5-$4, $6, $7, $8}' >> "$POP".transformedThetas

rm "$POP".*.borrar
mv $POP.printed.stats /backup/mlucena/intermediate_files_ANGSD/

# done

```


## Window analysis


```{bash}

RUTA=/home/mlucena/ANGSD_analysis 
cd $RUTA/whole_genome_analysis/sfs/window_analysis

#To launch one by one
POP="c_lp_sm_n019"  # <--CHANGE POP HERE
screen -S "$POP"_sfs_window
POP="c_lp_sm_n019"  # <--CHANGE POP HERE
script "$POP"_sfs_window.log
POP="c_lp_sm_n019"  # <--CHANGE POP HERE

THREADS=10                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!

NGSTOOLS="/opt/angsd/angsd/misc"

WINDOWSIZE=50000
WINDOWSTEP=10000 

RUTA=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs

echo "-------$POP---------- Window analysis -----------------------------------------"

$NGSTOOLS/thetaStat do_stat $RUTA/$POP.unfolded-lr.postprob.thetas.idx -win $WINDOWSIZE -step $WINDOWSTEP  -outnames $POP.unfolded-lr.postprob.thetasWindow_$WINDOWSIZE.$WINDOWSTEP.gz

```
Values in this output file are the sum of the per-site estimates for the whole window.


## Window R representation


Download data:

```{bash}

scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/*PG /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/window_analysis/

```


```{r}

# Representación de las ventanas en manhattan plots. 

library(dplyr)
library(qqman)
library(ggplot2)
library ("GGally")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/window_analysis/" 
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/"

finsThetas = list.files(path = wd, pattern="*.pestPG$")

for (i in 1:length(finsThetas))
{
  datThetas <- read.csv (paste(wd,finsThetas[i],sep=""), header = T, sep = '\t')  
  name_thetas <- unlist(strsplit(finsThetas[i], "[.]"))
  datThetas$CHR <- as.numeric(gsub('^.{6}', '', datThetas$Chr))
  datThetas$Nsites <- as.numeric(datThetas$nSites)
  datThetas$SNPS_window = paste(datThetas$Chr, datThetas$WinCenter, sep='_')
  datThetas[,-c(1,2)][datThetas[, -c(1,2)] < 0] <- 0
  # Manhattan plot Theta Waterson
  max_thetas <- max (datThetas$tW)
  min_thetas <- min(datThetas$tW)
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_watterson_manhattan.eps', sep=''))
  manhattan(datThetas, chr = "CHR", bp = "WinCenter", p = "tW", snp = "SNPS_window", ylim = c(0 , max_thetas+10), ylab = "tW", xlab = "Scaffold", logp=F, genomewideline = F, suggestiveline = F, main = paste ("Manhattan plot of tW ",name_thetas[1], sep =''))
  dev.off() 
  # Correlation between diversity indixes
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_diversity_correlation.eps', sep=''), horizontal = FALSE, onefile = FALSE, paper = "special")
  print(ggpairs(datThetas, columns=4:8, title=paste("Theta correlations ", name_thetas[1], sep='')))
  dev.off()
  # Correlation between neutrality indixes
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_neutrality_correlation.eps', sep=''), horizontal = FALSE, onefile = FALSE, paper = "special")
  print(ggpairs(datThetas, columns=9:13, title=paste("Neutrality indexes correlations ", name_thetas[1], sep='')))
  dev.off()
}


```


```{bash}
scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/*eps /Users/marialucenaperez/Desktop

```



## Global diversity per feature

I made a gff3 file with all the characteristics that we want to evaluate. 
Now that I have the file to run over the calculus per unit. To do it in a efficient way I will split this file into as many files as scaffolds. I will do the same with my thetas file. 
That way I will just have to search over the file that contains the scaffold, and bedtools will be faster.  

ESTE ES EL ARCHIVO: 

/home/mlucena/grupolince/notation/LYPA23C.GENE.mRNA.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3

### First I separate transformedThetas per scaffold. 

```{bash}

cd /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit

NEW_FOLDER=("c_ll_ki_separated_by_scaffold" "c_ll_po_separated_by_scaffold" "c_ll_no_separated_by_scaffold" "c_lp_do_separated_by_scaffold" "c_lp_sm_separated_by_scaffold")

for FOLDER in "${NEW_FOLDER[@]}"
do
echo "---------------------------------------------------$FOLDER---------------------------------------------------"
mkdir $FOLDER;
done
 
 
POPS=("c_ll_no_n008" "c_ll_po_n008"  "c_lp_do_n012")



########################


for POP in "${POPS[@]}"
do
echo $POP
cd /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/
mv "$POP".transformedThetas "$POP"_separated_by_scaffold/
cd  /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/"$POP"_separated_by_scaffold
# Create multiple files base on one column: transformedThetas:
awk '{print >> $1; close($1)}' "$POP".transformedThetas
rm scaffold # esto borra la primera linea que forma un archivo llamado scaffold.
# Rename the files: transformedThetas

for file in lp23*
do
echo $file
mv $file ${file/lp23/"$POP".transformedThetas_lp23}
done
done

```

### Change folder name: Esta es una corrección ya aplicada que no tendría que aplicarse puesto que el script de arriba está bien. 

Prefiero añadir un nombre con el número de individuos (que aunque estaba así arriba porque lo he corregido para sm no lo había hecho así), así que lo cambio.

```{bash}

mv c_lp_sm_separated_by_scaffold c_lp_sm_n019_separated_by_scaffold
cd c_lp_do_n012_separated_by_scaffold

for file in c_lp_do.*
do
echo $file 
mv $file ${file/c_lp_do/c_lp_do_n012}
done

```



### Diversity per feature.

Populations:

c_ll_ki_n013
c_ll_no
c_ll_po

c_lp_sm
c_lp_do

## OJO! parece que tengo valores de -1 en los NAS, ¿eso por qué? corregir e incorporar directamente el valor de informative sites en el loop.

```{bash}

### Change location!

# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!
# screen -S "$POP"_scaffold_per_unit
# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!
# script log_screen_"$POP"_scaffold_per_unit.log
# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!

## WG variables

screen -S Scaffold_per_unit
script log_screen_Scaffold_per_unit

POPS=("c_ll_ki_n013" "c_ll_no_n008" "c_ll_po_n008" "c_lp_do_n012" "c_lp_sm_n019")

SCAFFOLDS_FOLDER=/GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.corrected.PerScaffold/
cd $SCAFFOLDS_FOLDER
SCAFFOLDS=($(find /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.corrected.PerScaffold/ -name "LYPA23C*" -exec ls {} \; | awk '{ print substr( $0, length($0) - 10, length($0) ) }'  | LANG=en_EN sort | uniq ))
DIVERSITY_PER_UNIT_FOLDER=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit/
SFS_FOLDER=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/


for POP in "${POPS[@]}"
do

echo "---------------------------------------------------$POP---------------------------------------------------"

# Create headers for the outfile
echo -e "scaffold\tstart\tend\tlength\tNAs\tfeature\tstrandness\tframe\tid_gene\tid\twatterson_ave\twatterson_sd\tpairwise_ave\tpairwise_sd\ttajimaD\tpop" > $DIVERSITY_PER_UNIT_FOLDER"$POP".per.unit.averages.tsv

for SCAFFOLD in "${SCAFFOLDS[@]}"
do
echo "---------------------$SCAFFOLD---------------------"

#For each unit

cd "$SFS_FOLDER""$POP"_separated_by_scaffold

while read LOCATION METHOD FEATURE START_ONEBASED END POINT STRANDNESS FRAME IDRAW;

do

echo "--------$SCAFFOLD:$FEATURE--------"

LENGTH=$(expr $END - $START_ONEBASED) 

if [ "$METHOD" = "PipeR" ] 
then
ID_GENE=$(echo $IDRAW | awk -F "_" '{ split ($0, a, "ID="); split (a[2],b,"_"); print b[1]"_"b[2] }')
else
ID_GENE=$(echo $IDRAW | awk '{ split($0,a,"ID="); split (a[2],b,";"); split(b[1],c,"T"); print c[1] }') 
fi

if [ "$FEATURE" = "CDS" ]
then
ID=$(echo $IDRAW | awk -F "_" '{split ($0,a,"Target="); split(a[2],b,";"); print b[1]}' |  awk '{print $1"_"$2"_"$3}' )
else
ID=$(echo $IDRAW | awk '{ split($0,a,"ID="); split (a[2],b,";"); print b[1] }')
fi

# Intersect bed of the unit (first remove header)¿¿??

START_CEROBASED=($(echo $START_ONEBASED-1 | bc))

cat "$POP".transformedThetas_"$SCAFFOLD" | bedtools intersect -a stdin -b <(echo -e "$LOCATION\t$START_CEROBASED\t$END") > "$POP".iter.transformedThetas.borrar

WATTERSON_AVERAGE_PER_UNIT=$(cut -f 4 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g')
WATTERSON_SD_PER_UNIT=$(cut -f 4 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g')

PAIRWISE_AVERAGE_PER_UNIT=$(cut -f 5 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g') 
PAIRWISE_SD_PER_UNIT=$(cut -f 5  "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT=$(cut -f 6  "$POP".iter.transformedThetas.borrar   |  awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR ))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

TAJIMAS_D_PER_UNIT=$(echo "(($PAIRWISE_AVERAGE_PER_UNIT) - ($WATTERSON_AVERAGE_PER_UNIT))/($DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT)" | bc -l | awk '{printf ("%.10e\n",$1)}' |  sed 's/[eE]+\{0,1\}/*10^/g' )

NR=$(wc -l "$POP".iter.transformedThetas.borrar | cut -d" " -f1)

NAs=$(expr $LENGTH - $NR)

INFORMATIVESITES=$(expr $LENGTH - $NAs)


#Before printing check all variables are full, if empty use NA
if [ -z ${LOCATION} ]; then  LOCATION=NA;  fi
if [ -z ${START_ONEBASED} ]; then  START_ONEBASED=NA;  fi
if [ -z ${END} ]; then  END=NA;  fi
if [ -z ${LENGTH} ]; then  LENGTH=NA;  fi
if [ -z ${NAs} ]; then  NAs=NA;  fi
if [ -z ${INFORMATIVESITES} ]; then  INFORMATIVESITES=NA;  fi
if [ -z ${FEATURE} ]; then  FEATURE=NA;  fi
if [ -z ${STRANDNESS} ]; then  STRANDNESS=NA;  fi
if [ -z ${FRAME} ]; then  FRAME=NA;  fi
if [ -z ${ID_GENE} ]; then  ID_GENE=NA;  fi
if [ -z ${ID} ]; then  ID=NA;  fi
if [ -z ${WATTERSON_AVERAGE_PER_UNIT} ]; then  WATTERSON_AVERAGE_PER_UNIT=NA;  fi
if [ -z ${WATTERSON_SD_PER_UNIT} ]; then  WATTERSON_SD_PER_UNIT=NA;  fi
if [ -z ${PAIRWISE_AVERAGE_PER_UNIT} ]; then  PAIRWISE_AVERAGE_PER_UNIT=NA;  fi
if [ -z ${PAIRWISE_SD_PER_UNIT} ]; then  PAIRWISE_SD_PER_UNIT=NA;  fi
if [ -z ${TAJIMAS_D_PER_UNIT} ]; then  TAJIMAS_D_PER_UNIT=NA;  fi
if [ -z ${POP} ]; then  POP=NA;  fi

#Paste averages, and standatd deviations

paste \
<(echo $LOCATION ) \
<(echo $START_ONEBASED ) \
<(echo $END ) \
<(echo $LENGTH ) \
<(echo $NAs) \
<(echo $INFORMATIVESITES) \
<(echo $FEATURE ) \
<(echo $STRANDNESS ) \
<(echo $FRAME ) \
<(echo $ID_GENE ) \
<(echo $ID) \
<(echo $WATTERSON_AVERAGE_PER_UNIT) \
<(echo $WATTERSON_SD_PER_UNIT) \
<(echo $PAIRWISE_AVERAGE_PER_UNIT) \
<(echo $PAIRWISE_SD_PER_UNIT) \
<(echo $TAJIMAS_D_PER_UNIT) \
<(echo $POP) |\
sed 's/ /\t/g'| sed 's/\t\+/\t/g'  >>  $DIVERSITY_PER_UNIT_FOLDER"$POP".per.unit.averages.tsv

#Reset all (but POP) variables before next iteration

unset LOCATION
unset START_ONEBASED
unset END
unset LENGTH
unset NAs
unset INFORMATIVESITES
unset FEATURE
unset STRANDNESS
unset FRAME
unset ID_GENE
unset ID
unset WATTERSON_AVERAGE_PER_UNIT
unset WATTERSON_SD_PER_UNIT
unset PAIRWISE_AVERAGE_PER_UNIT
unset PAIRWISE_SD_PER_UNIT
unset TAJIMAS_D_PER_UNIT

done < <(cat "$SCAFFOLDS_FOLDER"LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3_"$SCAFFOLD") 

done &

rm "$POP".iter.transformedThetas.borrar


```

#### Correcciones diversity per feature!!

```{bash}

# Calculo el número de sitios informativos, para todos los archivos:

FILES_DIVERSITY=("c_ll_ki_n013.per.unit.averages.tsv" "c_ll_po_n008.per.unit.averages.tsv" "c_lp_sm_n019.per.unit.averages.tsv" "c_ll_no_n008.per.unit.averages.tsv" "c_lp_do_n012.per.unit.averages.tsv")


## OJO! Que lo está imprimiendo en otro orden!

# Añado una columna con el número de sitios informativos. 
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
echo ${i/.tsv/_informative_sites.tsv}
cat <(awk -v OFS='\t' '(NR==1) {print $0,"informative_sites"}' $i ) <(awk -v OFS='\t' 'NR >= 2 {print $0,$4-$5}' $i) > ${i/.tsv/_informative_sites.tsv}
mv ${i/.tsv/_informative_sites.tsv} $i
done

# Añado una columna en el header con la poblacion
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
POP=$(echo ${i/.per.unit.averages.tsv/} )
awk -v OFS='\t' 'NR==1{print $0,"pop";next}{print}' $i > ${i/.per.unit.averages/_withPOP.per.unit.averages}
mv ${i/.per.unit.averages/_withPOP.per.unit.averages} $i
done


# Añado una columna con la specie
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
POP=$(echo ${i/.per.unit.averages.tsv/} )
SPECIE=$(echo $POP | cut -d"_" -f 1-2)
echo $SPECIE
cat <(awk -v OFS='\t' '(NR==1) {$18="specie"; print}' $i) <(awk -v OFS='\t' -v SPECIE="$SPECIE" 'NR >= 2 {print $0, SPECIE}' $i) > ${i/.tsv/_corrected.tsv}
mv ${i/.tsv/_corrected.tsv} $i
done


# 8/9/2017: Haciendo un plot me he dado cuenta que hay algunas filas duplicadas en el archivo original .gff3, corresponden a UTR. 
# En la carpeta /home/mlucena/grupolince/notation, están especificadas cuales son. He corregido el gff3, pero como no voy a volver a correr mis archivos para sacar los números, voy a corregir todos los originales para tenerlo todo bien desde el principio. 

# Check:

# sort c_ll_ki_n013.per.unit.averages.tsv | uniq -c | sort -nr | head

# 2 lp23.s36682     338697  338929  232     -1      3UTR    +       .       LYPA23C000008   LYPA23C000008   1.4564973179*10^-06     4.5837356691*10^-06     4.4811971320*10^-07     1.3890498033*10^-06     -3.1562808160*1
# 2 lp23.s36682     328523  328962  439     75      5UTR    -       .       LYPA23C000003   LYPA23C000003   5.0803158132*10^-06     1.2267826914*10^-05     1.8178262696*10^-06     4.0698820431*10^-06     -3.9712504595*1
# 2 lp23.s36682     279677  279811  134     -1      3UTR    +       .       LYPA23C000015   LYPA23C000015   1.0437203878*10^-06     4.1265547119*10^-06     3.1206706874*10^-07     1.2333416354*10^-06     -2.5288512495*1
# 2 lp23.s36682     100823  101610  787     82      3UTR    +       .       LYPA23C000004   LYPA23C000004   3.7669060166*10^-04     9.8624719772*10^-03     3.9965744098*10^-04     1.0556168131*10^-02     3.3103961265*10
# 2 lp23.s26402     92633   95884   3251    166     3UTR    +       .       LYPA23C000017   LYPA23C000017   9.8348133084*10^-06     3.1577310510*10^-04     4.3617798050*10^-06     1.5841654621*10^-04     -3.4633370151*1
# 2 lp23.s10719     96754   96794   40      -1      5UTR    -       .       LYPA23C000041   LYPA23C000041   3.0378406270*10^-06     1.0043475395*10^-06     1.0387353248*10^-06     3.9400597979*10^-07     -3.2716817179*1
# 2 lp23.s10719     156260  156378  118     96      5UTR    -       .       LYPA23C000045   LYPA23C000045   1.3834272997*10^-05     2.2261745529*10^-06     6.3570507930*10^-06     1.2906641862*10^-06     -7.9734277877*1
# 1 scaffold        start   end     length  NAs     feature strandness      frame   id_gene id      watterson_ave   watterson_sd    pairwise_ave    pairwise_sd     tajimaD informative_sites       pop     specie
# 1 lp23.s41700     1       200     199     199     intergenic      -       .       intergenic_region_65795 intergenic_region_65795         0       c_ll_ki_n013    c_ll
# 1 lp23.s41699     1       200     199     199     intergenic      -       .       intergenic_region_65794 intergenic_region_65794         0       c_ll_ki_n013    c_ll


# 491637 c_ll_ki_n013.per.unit.averages.tsv

# Efectivamente aquí también está duplicados, así que corro el loop para corregirlo:

for i in  *.per.unit.averages.tsv
do
echo $i
POP=$(echo ${i} | cut -d "." -f 1 )
SPECIE=$(echo $POP | cut -d"_" -f 1-2)
awk '!seen[$0]++' $i > ${i/.tsv/.corrected.tsv} # remove duplicated rows.
mv ${i/.tsv/.corrected.tsv} $i
done



# Create a unique file with all the information

awk 'FNR==1 && NR!=1{next;}{print}' c_lp*averages.tsv > c_lp_do_n012-c_lp_sm_n019.per.unit.averages.per_specie.tsv
awk 'FNR==1 && NR!=1{next;}{print}' c_ll*averages.tsv > c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv


# Compruebo que ha funcionado:

# sort c_ll_ki_n013.per.unit.averages.tsv | uniq -c | sort -nr | head
# 1 scaffold	start	end	length	NAs	feature	strandness	frame	id_gene	id	watterson_ave	watterson_sd	pairwise_ave	pairwise_sd	tajimaD	informative_sites	pop	specie
# 1 lp23.s41700	1	200	
      
# Perfecto!!
      
```





# OJO!!

Calcular luego en R medidas pareadas!!! cada unidad con su equivalente en las tres poblaciones!


```{bash}
nota


wc -l c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv
1474888 c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv


awk '{if ($18 == "") print $0}' c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv  | wc -l
134260

# Tengo 134260 lineas que no tienen todos los campos, eso supone un 9,1% de total. 
# ¿A cuantas les faltan todos los campos y por tanto no ha perjudicados los cálculos? (A todas las que tengas informative sites ($11), igual a cero)

awk '{if ($18 == "") print $0}' c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv  | awk '{if ($11=="0") print $0}' | wc -l
133589

# Por tanto, de las 134260, a 133589 le faltan todos los datos, por tanto sólo hay 671 lineas con algunos valores sí y otros no. 

133589-134260=671

awk '{if ($18 == "") print $0}' c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv  | awk '{if ($11!="0") print $0}' | wc -l
671



# De estos, parece que hay muchos con sitios informativos=1, por lo que asumo que no puede calcular las tajimaD. 

awk '{if ($18 == "") print $0}' c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv  | awk '{if ($11!="0") print $0}' | head
lp23.s00005	1472211	1472211	0	-1	CDS	-	0	LYPA23C006624	LYPA23C006624P4_1_1	9.1722861064*10^-06	0.0000000000*10^00	3.7772361236*10^-06	0.0000000000*10^00		1	c_ll_ki_n013	c_ll
lp23.s00006	42798	43028	230	229	intron	+	0	LYPA23C020681	LYPA23C020681T1_intron_13	7.9066451013*10^-06	0.0000000000*10^00	3.1403854839*10^-06	0.0000000000*10^00		1	c_ll_ki_n013	c_ll
lp23.s00010	215823	215849	26	25	lncRNA	+

# Concretamente encontramos,

 awk '{if ($18 == "") print $0}' c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv  | awk '{if ($11!="0") print $0}' | awk '{if ($15="1") print $0}' | wc -l
671

# Todo cuadra. 

# 671 supoone un 0.045 del total, así que yo creo que no influye o no mucho en los resultados que tenog. En todo caso, como en algunos casos la coumna de informative site 01 está donde watterson debería estar, ahí si pero en el resto no. 


```




#### Representation per pop --> I rm this files, but I keep the script in case is useful. 

Este script nos devuelve una gráfica por cada población. Hay un segundo script que devuelve una gráfica para lynx lynx y otra para lynx pardinus. 

```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per.unit.averages.tsv$")
INFORMATIVE_SITES=1
PERCENTAGE_COVERED=0.5

for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))

#### Para que almacene varias dataframe con el nombre correcto
#  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
#  assign(dataframename, read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".", colClasses=c("watterson_ave"="character")))

dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")

dataframe = dataframe[-1,]
dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
#dataframe  <- dataframe[complete.cases(dataframe),] ## --> No se que pasa que la columna de specie sale NA

########################################################
# Filtering
dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>PERCENTAGE_COVERED) %>% dplyr::filter(informative_sites>INFORMATIVE_SITES)
name_diversity[1] <- paste(name_diversity[1],".percentagecovered",PERCENTAGE_COVERED,"_mininformativesites",INFORMATIVE_SITES, sep="")
########################################################

df.pm <- ddply(dataframe, "feature", summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites))

df.wm <- ddply(dataframe, "feature", summarise, weightedmean=weighted.mean(watterson_ave, informative_sites))


# pdf(file = paste(wd,"/Diversity_by_feature_pairwise",name_diversity[1],".pdf", sep = ""))
ggplot(data = dataframe, aes(x=feature, y = pairwise_ave)) + 
  geom_boxplot()+
  geom_point(data=df.pm,aes(x=feature, y = weightedmean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.pdf", sep = ""), device = "pdf")
# dev.off()



# pdf(file = paste(wd,"/Diversity_by_feature_watterson",name_diversity[1],".pdf", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = watterson_ave)) + 
  geom_boxplot()+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())

ggsave(filename = paste(wd,"/", name_diversity[1],".diversity_feature_watterson.pdf", sep = ""), device = "pdf")


ggplot(data = dataframe, aes(x=feature, y = tajimaD)) + 
  geom_boxplot()+
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  #  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave(filename = paste(wd,"/",name_diversity[1],".percentagecovered",PERCENTAGE_COVERED,"_mininformativesites",INFORMATIVE_SITES,".diversity_feature_tajimaD.pdf", sep = ""), device = "pdf")


# Exploratory plots

jpeg(paste(wd,"/",name_diversity[1],".plot1.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot2.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$watterson_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot3.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot4.jpg", sep=""))
plot (dataframe$watterson_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot5.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$pairwise_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot6.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$watterson_ave)
dev.off()


}


# Cosas útiles outliers.shape=NA.

```

# Comparación por pares unidad


```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library(dplyr)

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_region"

finsdiversity = list.files(path = wd,pattern="*.per.unit.averages.tsv$")

for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))
dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")

dataframe = dataframe[-1,]
dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))

#### Para que almacene varias dataframe con el nombre correcto
  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
  assign(dataframename, dataframe)
  
  

}


pair_comparison <- full_join (data_diversity_c_ll_ki_n013, data_diversity_c_ll_no_n008, by=c("row.names"="row.names","scaffold"="scaffold","start"="start", "end"="end",  "feature"="feature", "strandness"="strandness", "frame"="frame", "id_gene"="id_gene", "id"="id")  )

colourCount = length(unique(pair_comparison$feature))
getPalette = colorRampPalette(brewer.pal(9, "Paired"))


ggplot (filter(pair_comparison), aes(x= watterson_ave.x, y=watterson_ave.y , fill=feature, colour=feature)) +
  geom_point(alpha=1/100) +
  geom_smooth(method = "lm", se = FALSE) +
  geom_abline(intercept = 0) +
  scale_fill_manual(values = getPalette(colourCount))+
  scale_colour_manual(values = getPalette(colourCount)) +
   theme(legend.position="bottom") +
  guides(fill=guide_legend(nrow=2))

```







#### Representation per specie

Esta gráfica representa por especie.

```{r}


# stat_summary(fun.y=mean,

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library("tidyr")
library ("dplyr")
# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per_specie.tsv$")
INFORMATIVE_SITES=1
PERCENTAGE_COVERED=0


for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))
  dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")
  dataframe = dataframe[-1,]
  dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
  dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
  dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
  dataframe  <- dataframe[complete.cases(dataframe),]
  dataframe$pop <- as.factor(dataframe$pop)
  dataframe$pop <- factor(dataframe$pop, levels=c("c_ll_ki_n013", "c_ll_po_n008", "c_ll_no_n008", "c_lp_sm_n019", "c_lp_do_n012"))
  #### Para que almacene varias dataframe con el nombre correcto
  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
  assign(dataframename, dataframe)
  
  ########################################################
# Filtering
  
  dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>PERCENTAGE_COVERED) %>% dplyr::filter(informative_sites>INFORMATIVE_SITES)
  
}



### LYNX PARDINUS #### 

dataframe <- `data_diversity_c_lp_do_n012-c_lp_sm_n019`
name_diversity[1] <- "c_lp_do_n012-c_lp_sm_n019"
name_diversity[1] <- paste(name_diversity[1],".percentagecovered",PERCENTAGE_COVERED,"_mininformativesites",INFORMATIVE_SITES, sep="")


# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites), mean=mean(pairwise_ave))

df.pm1 <- df.pm %>% select (feature,pop, weightedmean) %>% spread (., pop, weightedmean) %>% select (feature, wm_sm= c_lp_sm_n019,wm_do=c_lp_do_n012)
df.pm2 <- df.pm %>% select (feature,pop, mean) %>% spread (., pop, mean) %>% select (feature, m_sm= c_lp_sm_n019,m_do=c_lp_do_n012)

df.pm_guardar <- full_join(df.pm1, df.pm2)


df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(watterson_ave, informative_sites), mean=mean(watterson_ave))

df.wm1 <- df.pm %>% select (feature,pop, weightedmean) %>% spread (., pop, weightedmean) %>% select (feature, wm_sm= c_lp_sm_n019,wm_do=c_lp_do_n012)
df.wm2 <- df.pm %>% select (feature,pop, mean) %>% spread (., pop, mean) %>% select (feature, m_sm= c_lp_sm_n019,m_do=c_lp_do_n012)

df.wm_guardar <- full_join(df.pm1, df.pm2)


# LYNX PARDINUS
# Hago spread para calcular el porcentage que representa do respecto a sm
df.pm_lost <- df.pm %>% select(.,-mean) %>% tidyr::spread(., pop, weightedmean) %>%
 mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "weightedmean", -specie, -feature, -percentage_do_sm)
# Hago spread para calcular el porcentage de perdida
df.wm_lost <-df.wm %>% select(.,-mean) %>% tidyr::spread(., pop, weightedmean) %>% mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "weightedmean", -specie, -feature, -percentage_do_sm)


# Guardo el porcentage de perdida. 
write.table(df.pm_guardar, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""), row.names = F, dec=",", sep=";")
write.table(df.wm_guardar, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""), row.names = F, dec=",", sep=";")


ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot() +
  geom_point(data=df.pm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+ 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
  geom_text(data = df.pm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)

ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.eps", sep = ""), device = "eps")
# dev.off()


ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) + 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
  geom_text(data = df.wm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_feature_watterson.eps", sep = ""), device="eps")



### LYNX LYNX ####

dataframe <- `data_diversity_c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008`
name_diversity[1] <- "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008"
name_diversity[1] <- paste(name_diversity[1],".percentagecovered",PERCENTAGE_COVERED,"_mininformativesites",INFORMATIVE_SITES, sep="")

# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites), mean=mean(pairwise_ave))

df.pm1 <- df.pm %>% select (feature,pop, weightedmean) %>% spread (., pop, weightedmean) %>% select (feature, wm_ki= c_ll_ki_n013,wm_no=c_ll_no_n008,wm_po=c_ll_po_n008)
df.pm2 <- df.pm %>% select (feature,pop, mean) %>% spread (., pop, mean) %>% select (feature, m_ki= c_ll_ki_n013,m_no=c_ll_no_n008,m_po=c_ll_po_n008)

df.pm_guardar <- full_join(df.pm1, df.pm2)


df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(watterson_ave, informative_sites), mean=mean(watterson_ave))

df.wm1 <- df.wm %>% select (feature,pop, weightedmean) %>% spread (., pop, weightedmean) %>% select (feature, wm_ki= c_ll_ki_n013,wm_no=c_ll_no_n008,wm_po=c_ll_po_n008)
df.wm2 <- df.wm %>% select (feature,pop, mean) %>% spread (., pop, mean) %>% select (feature, m_ki= c_ll_ki_n013,m_no=c_ll_no_n008,m_po=c_ll_po_n008)

df.wm_guardar <- full_join(df.pm1, df.pm2)


df.pm_lost <- df.pm %>% select(.,-mean) %>% tidyr::spread(., pop, weightedmean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "weightedmean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki)

df.wm_lost <- df.wm %>% select(.,-mean) %>% tidyr::spread(., pop, weightedmean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "weightedmean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki) 

# Guardo el porcentage de perdida. 
write.table(df.pm_guardar, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""), row.names = F, dec=",", sep=";")
write.table(df.wm_guardar, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""), row.names = F, dec=",", sep=";")



ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  #scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+ 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
  geom_text(data = df.pm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.eps", sep = ""), device = "eps")
# dev.off()



ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA, position=position_dodge(width=1.5))+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=1.5), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) + 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
        geom_text(data = df.wm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")

ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_feature_watterson.eps", sep =""),device="eps")


# Cosas útiles 

# geom_box(outliers.shape=NA)

# Si quiero plotear todos los indices a la vez:
#index = c("watterson_ave", "pairwise_ave")
#for (i in 1:length(index))
#{
#  ggplot(data = dataframe, aes_string(x='feature', y = index[1], fill='pop'))+



```


#### Pairwise Pi diversity

```{r}


library ("ggplot2")
library ("plyr")
library ("dplyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library("tidyr")
library(viridis)


# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/Users/marialucenaperez/Desktop/"


# finsdiversity = list.files(path = wd,pattern="*per.unit.averages.per_specie.tsv$") %>% grep("informative_sites", ., invert = T, value=T)

infile <- "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv"

INFORMATIVE_SITES=1
PERCENTAGE_COVERED=0




# for (i in 1:length(finsdiversity))
# {
#  infile <- finsdiversity[i]
  
  dataframe <- read.csv (paste (wd,infile, sep="/"), header = T, sep = '\t', stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".") 
  dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
  dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
  dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
# Filtering
  dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>PERCENTAGE_COVERED) %>% dplyr::filter(informative_sites>INFORMATIVE_SITES)
  

  
# scaffold_start_end_feature_id, length NAs feature strandness frame id_gene watterson_ave watterson_sd pairwise_ave pairwise_sdtajimaD informative_sites pop specie length NAs
distinct_pop <- dataframe %>% distinct(pop)
distinct_vector_pop = distinct_pop$pop

# distinct_vector_pop=c("don","smo","po","cat","bia")

combinations <- t(combn(distinct_vector_pop, 2)) # %>% mutate (combination=paste(V1,V2,sep="/")) %>% select(combination)

# for(a in 1:nrow(combinations)) {
#    row <- NULL
    row <- combinations[a,]
    # do stuff with row
    POP1 <- NULL
    POP2 <- NULL
    POP1 <- row[1]
    POP2 <- row[2]
    NAME <- paste(POP1,"_",POP2,".percentagecovered",PERCENTAGE_COVERED,"_mininformativesites",INFORMATIVE_SITES, sep="")

    
    df_POP1 <- filter (dataframe, pop == POP1) %>% mutate(., feature2=feature) %>% unite (., scaffold, start, end, feature2, id, col="scaffold_start_end_feature_id")
    
    df_POP2 <- filter (dataframe, pop == POP2) %>% mutate(., feature2=feature) %>% unite (., scaffold, start, end, feature2, id, col="scaffold_start_end_feature_id") 
    

    vector_positions_POP1 <- unique(df_POP1$scaffold_start_end_feature_id) %>% sort(.)
    vector_positions_POP2 <- unique(df_POP2$scaffold_start_end_feature_id) %>% sort(.)
    
    vector_common_positions <- as.vector(intersect(vector_positions_POP1, vector_positions_POP2))
    
    df_POP1_filtered <- df_POP1 %>% filter(scaffold_start_end_feature_id  %in% vector_common_positions) %>% arrange(., scaffold_start_end_feature_id) %>% select (., scaffold_start_end_feature_id, feature, id_gene, pairwise_ave, informative_sites, length, NAs, pop )
    
    df_POP2_filtered <- df_POP2 %>% filter(scaffold_start_end_feature_id  %in% vector_common_positions) %>% arrange(., scaffold_start_end_feature_id) %>% select (., scaffold_start_end_feature_id, feature, id_gene, pairwise_ave, informative_sites, length, NAs, pop )

if (identical(df_POP1_filtered$scaffold_start_end_feature_id, df_POP2_filtered$scaffold_start_end_feature_id))
{ df_pairwise <- cbind (
  setNames(data.frame(df_POP1_filtered), c("scaffold_start_end_feature_id", "feature", "id_gene", "pairwise_ave1","informative_sites1", "length1", "NAs1", "pop1")),
  setNames(data.frame(df_POP2_filtered), c("scaffold_start_end_feature_id2", "feature2", "id_gene2", "pairwise_ave2","informative_sites2", "length2", "NAs2", "pop2"))) %>% select_ (., "scaffold_start_end_feature_id", "feature", "id_gene", "pairwise_ave1","informative_sites1", "length1", "NAs1", "pop1","pairwise_ave2","informative_sites2", "length2", "NAs2", "pop2")
           } else {print("First column not matching") }

    
ggplot(df_pairwise, aes(informative_sites1, informative_sites2)) +
  geom_point() +
  labs(x=paste("Informative sites", POP1, sep =" "),y=paste("Informative sites", POP2, sep =" ")) +
  theme_classic()

ggsave(filename = paste(wd,"/",NAME,"_informativesites.eps", sep = ""), device="eps")


ggplot(df_pairwise, aes(pairwise_ave1, pairwise_ave2,  color = informative_sites1)) +
  geom_point() +
  facet_wrap(~feature) +
  # scale_color_gradient(low="red", high="blue") +
  #scale_color_gradientn(colours = rainbow(5)) +
  # scale_colour_gradient(limits = c(0, quantile(df_pairwise$informative_sites1, 0.5))) +
  # scale_colour_gradientn(colours = myPalette(100), values=nrm.vals) +
  labs(colour = paste("Informative sites", POP1, sep =" ")) +
  labs(x=paste("Pi diversity", POP1, sep =" "),y=paste("Pi diversity", POP2, sep =" ")) +
  scale_color_viridis(option="magma",limits = c(0, quantile(df_pairwise$informative_sites1, 0.5))) +
  theme_classic()

ggsave(filename = paste(wd,"/",NAME,"_informative_sites1_Pi_pairwise.eps", sep = ""), device="eps")

}

}

# https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html

```



#### Representation per pop

Este script nos devuelve una gráfica por cada población. Hay un segundo script que devuelve una gráfica para lynx lynx y otra para lynx pardinus. 



```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_region"

finsdiversity = list.files(path = wd,pattern="*.per.region.averages.tsv$")

for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))

#### Para que almacene varias dataframe con el nombre correcto
#  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
#  assign(dataframename, read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".", colClasses=c("watterson_ave"="character")))

dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")

dataframe = dataframe[-1,]
dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
#dataframe  <- dataframe[complete.cases(dataframe),] ## --> No se que pasa que la columna de specie sale NA

########################################################
# Filtering
dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>0.5) %>% dplyr::filter(informative_sites>10)
name_diversity[1] <- paste(name_diversity[1],".percentagecovered05_mininformativesites10", sep="")
########################################################

ddply(dataframe, "feature", summarise, weightedmean=mean(pairwise_ave))
df.pm <- ddply(dataframe, "feature", summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites))


ddply(dataframe, "feature", summarise, weightedmean=mean(watterson_ave))
df.wm <- ddply(dataframe, "feature", summarise, weightedmean=weighted.mean(watterson_ave, informative_sites))


# pdf(file = paste(wd,"/Diversity_by_region_pairwise",name_diversity[1],".pdf", sep = ""))
ggplot(data = dataframe, aes(x=feature, y = pairwise_ave)) + 
  geom_boxplot()+
  geom_point(data=df.pm,aes(x=feature, y = weightedmean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.pdf", sep = ""), device = "pdf")
# dev.off()



# pdf(file = paste(wd,"/Diversity_by_region_watterson",name_diversity[1],".pdf", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = watterson_ave)) + 
  geom_boxplot()+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())

ggsave(filename = paste(wd,"/", name_diversity[1], ".diversity_region_watterson.pdf", sep = ""), device = "pdf")


ggplot(data = dataframe, aes(x=feature, y = tajimaD)) + 
  geom_boxplot()+
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  #  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_tajimaD.pdf", sep = ""), device = "pdf")


# Exploratory plots

jpeg(paste(wd,"/",name_diversity[1],".plot1.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot2.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$watterson_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot3.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot4.jpg", sep=""))
plot (dataframe$watterson_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot5.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$pairwise_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot6.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$watterson_ave)
dev.off()


}


# Cosas útiles outliers.shape=NA.

```









#### Representation per specie

Esta gráfica representa por especie.

```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library("tidyr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per_specie.tsv$")


for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))
  dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")
  dataframe = dataframe[-1,]
  dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
  dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
  dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
  dataframe  <- dataframe[complete.cases(dataframe),]
  dataframe$pop <- as.factor(dataframe$pop)
  dataframe$pop <- factor(dataframe$pop, levels=c("c_ll_ki_n013", "c_ll_po_n008", "c_ll_no_n008", "c_lp_sm_n019", "c_lp_do_n012"))
  #### Para que almacene varias dataframe con el nombre correcto
  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
  assign(dataframename, dataframe)
}


##### Lynx pardinus


dataframe <- `data_diversity_c_lp_do_n012-c_lp_sm_n019`
name_diversity[1] <- "c_lp_do_n012-c_lp_sm_n019"


# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites), mean=mean(pairwise_ave))
df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(watterson_ave, informative_sites), mean=mean(watterson_ave))



# LYNX PARDINUS
# Hago spread para calcular el porcentage que representa do respecto a sm
df.pm_lost <- df.pm %>%  select (., -mean) %>% tidyr::spread(., pop, weightedmean) %>%
 mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "weightedmean", -specie, -feature, -percentage_do_sm)

# Hago spread para calcular el porcentage de perdida
df.wm_lost <- df.pm %>%  select (., -mean) %>% tidyr::spread(., pop, weightedmean) %>% mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "weightedmean", -specie, -feature, -percentage_do_sm)


# Guardo el porcentage de perdida. 
write.csv(df.pm, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""), row.names = F)
write.csv(df.wm, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""), row.names = F)


ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
 # scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+ 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
  geom_text(data = df.pm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.eps", sep = ""), device = "eps")
# dev.off()


ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) + 
        stat_summary(fun.y=mean, colour="darkred", geom="point", shape=18, size=3,show_guide = FALSE) + 
  geom_text(data = df.wm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_watterson.eps", sep = ""), device="eps")

#### Lynx lynx

dataframe <- `data_diversity_c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008`
name_diversity[1] <- "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008"

# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(pairwise_ave, informative_sites), mean=mean(pairwise_ave))
df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, weightedmean=weighted.mean(watterson_ave, informative_sites), mean=mean(watterson_ave))


df.pm_lost <- df.pm %>%  select (., -mean) %>% tidyr::spread(., pop, weightedmean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "weightedmean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki)

df.wm_lost <- df.wm %>%  select (., -mean) %>% tidyr::spread(., pop, weightedmean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "weightedmean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki) 



# Guardo el porcentage de perdida. 
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  #scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+
  geom_text(data = df.pm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")

ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.eps", sep = ""), device = "eps")
# dev.off()



ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA, position=position_dodge(width=1.5))+
  geom_point(data=df.wm,aes(x=feature, y = weightedmean, fill=pop),shape = 23, position=position_dodge(width=1.5), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) +
        geom_text(data = df.wm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")


ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_watterson.eps", sep = ""), device="eps")


# Cosas útiles outliers.shape=NA.



```





### Copy

```{bash}

scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit/*eps /Users/marialucenaperez/ownCloud/publico/WG_diversity/diversity_plots/diversity_per_unit 

scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit/*csv /Users/marialucenaperez/ownCloud/publico/WG_diversity/diversity_plots/diversity_per_unit 


scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region/*eps /Users/marialucenaperez/ownCloud/publico/WG_diversity/diversity_plots/diversity_per_region/



```





#### Diversity per region: Cat Coordinates

He probado con el script de Fede de perl:

```{bash}

# Primero le quito el cabecero porque si no, no puede usarse bedintersect:
# sed '1d' c_ll_ki_n013.per.region.averages.tsv  > c_ll_ki_n013.per.region.averages.NOHEADER.tsv 
# Ahora lo corro tal y como indica el propio script:
# perl lynx2cat.pl c_ll_ki_n013.per.region.averages.NOHEADER.tsv > c_ll_ki_n013.per.region.averages.CatCoordinates.tsv
# Este script me ha dado el siguiente error:
#Use of uninitialized value $s in concatenation (.) or string at lynx2cat.pl line 14, <I> line 156394492.
#Use of uninitialized value $e in concatenation (.) or string at lynx2cat.pl line 14, <I> line 156394492.
# Además me he dado cuenta que solo reporta las coordenadas de uno y otro, y yo quiero toda la información. 

```

Por tanto pruebo con el básico:

```{bash}

intersectBed -sorted -wo -a c_ll_ki_n013.per.region.averages.NOHEADER.tsv -b lynx2cat_wTiger.bed > c_ll_ki_n013.per.region.averages.CatCoordinates.tsv

# Me da el error:

# ERROR: chromomsome sort ordering for file c_ll_ki_n013.per.region.averages.NOHEADER.tsv is inconsistent with other files. Record was:
# lp23.s00015     2903992 3006783 102791  101509  telomers0-10m   na      na      na      na      6.5495647224*10^-06     1.4900294206*10^-05        2.5287837844*10^-06     5.8944110986*10^-06     -4.4603274016*10^-01    1282    c_ll_ki_n013    c_ll

```


Así que ordeno antes:

```{bash}

 bedtools sort c_ll_ki_n013.per.region.averages.NOHEADER.tsv > c_ll_ki_n013.per.region.averages.NOHEADER.SORTED.tsv

```


### Diversity per region & feature


```{bash}




```



