---
title: "pipeline_contemporary_sfs"
output: html_document
---

# Generating SAF 

## Introduction and populations:

I am going to run SAF in my target populations.

Populations:

c_ll_no -->
c_ll_po -->
c_ll_ki -->

c_lp_do -->
c_lp_sm -->

I checked that all bam files were indexed --> OK


## Global variables:


```{r, engine=bash, eval=FALSE}

RUTA=/home/mlucena/ANGSD_analysis 

cd $RUTA/whole_genome_analysis/sfs

#To launch one by one

POP="c_lp_do_n012"  # <--CHANGE POP HERE

screen -S "$POP"_sfs_wg
# screen -S "$POP"_thetas

POP="c_lp_do_n012"  # <--CHANGE POP HERE

script "$POP"_sfs_wg.log
# script "$POP"_thetas.log

POP="c_lp_do_n012"  # <--CHANGE POP HERE

THREADS=15                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!

RUTA=/home/mlucena/ANGSD_analysis/depth_calculus 

read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated < $RUTA/"${POP}"_mean_sd_depthGlobal_lynx_per_pop_mean_folds_0.95.csv

ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)
RUTA=/home/mlucena/ANGSD_analysis 


# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
echo $maxDepth_truncated
echo $minDepth_truncated

POPS=(c_lp_sm_n019 )   # <--CHANGE POP HERE


```

## Unfolded SAF 

```{r, engine=bash, eval=FALSE}

for POP in  ${POPS[@]}
do

##########################
#  SAF (likelihood):     
##########################

echo "-------$POP----------SAF (likelihood)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b $RUTA/whole_genome_analysis/"$POP".bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth_truncated -setMinDepth $minDepth_truncated


```

## SFS 

```{r, engine=bash, eval=FALSE}

##########################
#  SFS                   #(I dont require the -rf file as the saf already only contains the -rf sites
##########################
echo "-------$POP----------SFS------------------------------------------------------"

$NGSTOOLS/realSFS "$POP".unfolded-lr.saf.idx  -P $THREADS > "$POP".unfolded-lr.sfs


```

##  SAF (postprob) 

```{r, engine=bash, eval=FALSE}


echo "-------$POP----------SAF (postprob)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b $RUTA/whole_genome_analysis/"$POP".bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr.postprob \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth_truncated -setMinDepth $minDepth_truncated \
-doThetas 1 -pest "$POP".unfolded-lr.sfs 

```



## SFS graphical representation


First we copy the SFS files to our local folder.

```{bash}
scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/*sfs /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs

# I need to change name as R does not like "-lr.sfs"

cd /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/
for i in *sfs
do 
#echo $i
echo $i "--->" ${i/-lr.sfs/_lr.sfs}
mv $i ${i/-lr.sfs/_lr.sfs}
done


```

Now we run the script. 

```{r}
library(dplyr)
library(magrittr)

wd<-"/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/"

my_files_unfolded = list.files(path = wd, pattern="*sfs$")    
for (i in 1:length(my_files_unfolded)) 
{ assign(my_files_unfolded[i], (scan(paste(wd,my_files_unfolded[i], sep = ""), sep = " ", dec = ".")) %>% .[!is.na(.)])}

norm <- function(x) x/sum(x)

for (i in 1:length (my_files_unfolded))
{
  temp1=(eval(parse(text=my_files_unfolded[i])))
  temp2=norm(temp1[-1])
  temp3=norm(temp2[-c(length(temp2))])
  barplot(temp3,xlab="Chromosomes",
          names=1:length(temp3),ylab="Proportions",main=my_files_unfolded[i],col='grey')
  dev.print(device=postscript, paste("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/",my_files_unfolded[i],".eps"), onefile=FALSE, horizontal=FALSE)
  dev.off()
  rm(temp1)
  rm(temp2)
  rm(temp3)
}


```


##  Make bed theta        

```{r, engine=bash, eval=FALSE}

echo "-------$POP----------Make bed theta -----------------------------------------"

# $NGSTOOLS/thetaStat print $POP.unfolded-lr.postprob.thetas.idx > $POP.printed.stats

# Transform the coordinates to use 0Based bedtools

awk 'NR>1 {print  $1" "$2-1" "$2}' $POP.printed.stats > $POP.0basedcoordinates.borrar # --> 1, 2, 3
# log transformation of watterson and pairwise:
awk 'NR>1 {print  "e("$3")"}' $POP.printed.stats | bc -l > "$POP".watterson.borrar # --> 4
awk 'NR>1 {print  "e("$4")"}' $POP.printed.stats | bc -l > "$POP".pairwise.borrar # --> 5
                                                                                  # --> PAIRWISE - WATTERSON
awk 'NR>1 {print  "e("$5")"}' $POP.printed.stats | bc -l > "$POP".thetaFL.borrar # --> 6
awk 'NR>1 {print  "e("$6")"}' $POP.printed.stats | bc -l > "$POP".thetaH.borrar # --> 7
awk 'NR>1 {print  "e("$7")"}' $POP.printed.stats | bc -l > "$POP".thetaL.borrar # --> 8

# Create a header for the "$POP".transformedThetas file

echo "scaffold position1 position2 watterson pairwise pairwise-watterson thetaFL thetaH thetaL" > "$POP".transformedThetas

paste $POP.0basedcoordinates.borrar "$POP".watterson.borrar "$POP".pairwise.borrar "$POP".thetaFL.borrar "$POP".thetaH.borrar "$POP".thetaL.borrar | \
awk -v OFS='\t'  '{print $1, $2, $3, $4, $5, $5-$4, $6, $7, $8}' >> "$POP".transformedThetas

rm "$POP".*.borrar
mv $POP.printed.stats /backup/mlucena/intermediate_files_ANGSD/

done

```


## Window analysis


```{bash}

RUTA=/home/mlucena/ANGSD_analysis 
cd $RUTA/whole_genome_analysis/sfs/window_analysis

#To launch one by one
POP="c_lp_sm_n019"  # <--CHANGE POP HERE
screen -S "$POP"_sfs_window
POP="c_lp_sm_n019"  # <--CHANGE POP HERE
script "$POP"_sfs_window.log
POP="c_lp_sm_n019"  # <--CHANGE POP HERE

THREADS=10                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!

NGSTOOLS="/opt/angsd/angsd/misc"

WINDOWSIZE=50000
WINDOWSTEP=10000 

RUTA=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs

echo "-------$POP---------- Window analysis -----------------------------------------"

$NGSTOOLS/thetaStat do_stat $RUTA/$POP.unfolded-lr.postprob.thetas.idx -win $WINDOWSIZE -step $WINDOWSTEP  -outnames $POP.unfolded-lr.postprob.thetasWindow_$WINDOWSIZE.$WINDOWSTEP.gz

```



## Window R representation


Download data:

```{bash}
scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/*PG /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/window_analysis/
```


```{r}

# Representación de las ventanas en manhattan plots. 

library(dplyr)
library(qqman)
library(ggplot2)
library ("GGally")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/window_analysis/" 
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/"

finsThetas = list.files(path = wd, pattern="*.pestPG$")

for (i in 1:length(finsThetas))
{
  datThetas <- read.csv (paste(wd,finsThetas[i],sep=""), header = T, sep = '\t')  
  name_thetas <- unlist(strsplit(finsThetas[i], "[.]"))
  datThetas$CHR <- as.numeric(gsub('^.{6}', '', datThetas$Chr))
  datThetas$Nsites <- as.numeric(datThetas$nSites)
  datThetas$SNPS_window = paste(datThetas$Chr, datThetas$WinCenter, sep='_')
  datThetas[,-c(1,2)][datThetas[, -c(1,2)] < 0] <- 0
  # Manhattan plot Theta Waterson
  max_thetas <- max (datThetas$tW)
  min_thetas <- min(datThetas$tW)
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_watterson_manhattan.eps', sep=''))
  manhattan(datThetas, chr = "CHR", bp = "WinCenter", p = "tW", snp = "SNPS_window", ylim = c(0 , max_thetas+10), ylab = "tW", xlab = "Scaffold", logp=F, genomewideline = F, suggestiveline = F, main = paste ("Manhattan plot of tW ",name_thetas[1], sep =''))
  dev.off() 
  # Correlation between diversity indixes
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_diversity_correlation.eps', sep=''), horizontal = FALSE, onefile = FALSE, paper = "special")
  print(ggpairs(datThetas, columns=4:8, title=paste("Theta correlations ", name_thetas[1], sep='')))
  dev.off()
  # Correlation between neutrality indixes
  setEPS()
  postscript(file=paste(wd,name_thetas[1], '_neutrality_correlation.eps', sep=''), horizontal = FALSE, onefile = FALSE, paper = "special")
  print(ggpairs(datThetas, columns=9:13, title=paste("Neutrality indexes correlations ", name_thetas[1], sep='')))
  dev.off()
}


```


```{bash}
scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/window_analysis/*eps /Users/marialucenaperez/Desktop

```



## Global diversity per feature

I made a gff3 file with all the characteristics that we want to evaluate. 
Now that I have the file to run over the calculus per unit. To do it in a efficient way I will split this file into as many files as scaffolds. I will do the same with my thetas file. 
That way I will just have to search over the file that contains the scaffold, and bedtools will be faster.  

ESTE ES EL ARCHIVO: 

/home/mlucena/grupolince/notation/LYPA23C.GENE.mRNA.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3

### First I separate transformedThetas per scaffold. 

```{bash}

cd /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit

NEW_FOLDER=("c_ll_ki_separated_by_scaffold" "c_ll_po_separated_by_scaffold" "c_ll_no_separated_by_scaffold" "c_lp_do_separated_by_scaffold" "c_lp_sm_separated_by_scaffold")

for FOLDER in "${NEW_FOLDER[@]}"
do
echo "---------------------------------------------------$FOLDER---------------------------------------------------"
mkdir $FOLDER;
done
 
 
POPS=("c_ll_no_n008" "c_ll_po_n008"  "c_lp_do_n012")


"c_ll_ki" 
#### Falta este!! ###
"c_lp_sm_n019"
########################


for POP in "${POPS[@]}"
do
echo $POP
cd /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/
mv "$POP".transformedThetas "$POP"_separated_by_scaffold/
cd  /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/"$POP"_separated_by_scaffold
# Create multiple files base on one column: transformedThetas:
awk '{print >> $1; close($1)}' "$POP".transformedThetas
# rm scaffold ### no se pq esto?
# Rename the files: transformedThetas

for file in lp23*
do
echo $file
mv $file ${file/lp23/"$POP".transformedThetas_lp23}
done
done

```

### Change folder name: Esta es una corrección ya aplicada que no tendría que aplicarse puesto que el script de arriba está bien. 

Prefiero añadir un nombre con el número de individuos (que aunque estaba así arriba porque lo he corregido para sm no lo había hecho así), así que lo cambio.

```{bash}

mv c_lp_sm_separated_by_scaffold c_lp_sm_n019_separated_by_scaffold
cd c_lp_do_n012_separated_by_scaffold

for file in c_lp_do.*
do
echo $file 
mv $file ${file/c_lp_do/c_lp_do_n012}
done

```



### Diversity per feature.

Populations:

c_ll_ki_n013
c_ll_no
c_ll_po

c_lp_sm
c_lp_do


```{bash}

### Change location!

# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!
# screen -S "$POP"_scaffold_per_unit
# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!
# script log_screen_"$POP"_scaffold_per_unit.log
# POP=c_ll_ki_n013 # check if I need to include n_013 <--- Change pop here!

## WG variables

screen -S Scaffold_per_unit
script log_screen_Scaffold_per_unit

POPS=("c_ll_ki_n013" "c_ll_no_n008" "c_ll_po_n008" "c_lp_do_n012")


SCAFFOLDS_FOLDER=/home/mlucena/grupolince/notation/LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3.PerScaffold/
cd $SCAFFOLDS_FOLDER
SCAFFOLDS=($(find /home/mlucena/grupolince/notation/LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3.PerScaffold/ -name "LYPA23C*" -exec ls {} \; | awk '{ print substr( $0, length($0) - 10, length($0) ) }'  | LANG=en_EN sort | uniq ))
DIVERSITY_PER_UNIT_FOLDER=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit/
SFS_FOLDER=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/


for POP in "${POPS[@]}"
do

echo "---------------------------------------------------$POP---------------------------------------------------"

# Create headers for the outfile
echo -e "scaffold\tstart\tend\tlength\tNAs\tfeature\tstrandness\tframe\tid_gene\tid\twatterson_ave\twatterson_sd\tpairwise_ave\tpairwise_sd\ttajimaD" > $DIVERSITY_PER_UNIT_FOLDER"$POP".per.unit.averages.tsv

for SCAFFOLD in "${SCAFFOLDS[@]}"
do
echo "---------------------$SCAFFOLD---------------------"

#For each unit

cd "$SFS_FOLDER""$POP"_separated_by_scaffold

while read LOCATION METHOD FEATURE START_ONEBASED END POINT STRANDNESS FRAME IDRAW;

do

echo "--------$SCAFFOLD:$FEATURE--------"

LENGTH=$(expr $END - $START_ONEBASED) 

if [ "$METHOD" = "PipeR" ] 
then
ID_GENE=$(echo $IDRAW | awk -F "_" '{ split ($0, a, "ID="); split (a[2],b,"_"); print b[1]"_"b[2] }')
else
ID_GENE=$(echo $IDRAW | awk '{ split($0,a,"ID="); split (a[2],b,";"); split(b[1],c,"T"); print c[1] }') 
fi

if [ "$FEATURE" = "CDS" ]
then
ID=$(echo $IDRAW | awk -F "_" '{split ($0,a,"Target="); split(a[2],b,";"); print b[1]}' |  awk '{print $1"_"$2"_"$3}' )
else
ID=$(echo $IDRAW | awk '{ split($0,a,"ID="); split (a[2],b,";"); print b[1] }')
fi

# Intersect bed of the unit (first remove header)¿¿??

START_CEROBASED=($(echo $START_ONEBASED-1 | bc))

cat "$POP".transformedThetas_"$SCAFFOLD" | bedtools intersect -a stdin -b <(echo -e "$LOCATION\t$START_CEROBASED\t$END") > "$POP".iter.transformedThetas.borrar

WATTERSON_AVERAGE_PER_UNIT=$(cut -f 4 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g')
WATTERSON_SD_PER_UNIT=$(cut -f 4 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g')

PAIRWISE_AVERAGE_PER_UNIT=$(cut -f 5 "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g') 
PAIRWISE_SD_PER_UNIT=$(cut -f 5  "$POP".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT=$(cut -f 6  "$POP".iter.transformedThetas.borrar   |  awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR ))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

TAJIMAS_D_PER_UNIT=$(echo "(($PAIRWISE_AVERAGE_PER_UNIT) - ($WATTERSON_AVERAGE_PER_UNIT))/($DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT)" | bc -l | awk '{printf ("%.10e\n",$1)}' |  sed 's/[eE]+\{0,1\}/*10^/g' )

NR=$(wc -l "$POP".iter.transformedThetas.borrar | cut -d" " -f1)

NAs=$(expr $LENGTH - $NR)

# Calculate averages, and standatd deviations and paste them

paste \
<(echo $LOCATION ) \
<(echo $START_ONEBASED ) \
<(echo $END ) \
<(echo $LENGTH ) \
<(echo $NAs) \
<(echo $FEATURE ) \
<(echo $STRANDNESS ) \
<(echo $FRAME ) \
<(echo $ID_GENE ) \
<(echo $ID) \
<(echo $WATTERSON_AVERAGE_PER_UNIT) \
<(echo $WATTERSON_SD_PER_UNIT) \
<(echo $PAIRWISE_AVERAGE_PER_UNIT) \
<(echo $PAIRWISE_SD_PER_UNIT) \
<(echo $TAJIMAS_D_PER_UNIT) |\
sed 's/ /\t/g'| sed 's/\t\+/\t/g' >>  $DIVERSITY_PER_UNIT_FOLDER"$POP".per.unit.averages.tsv

done < <(cat "$SCAFFOLDS_FOLDER"LYPA23C.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.intergenic.nr.gff3_"$SCAFFOLD") 

done &
done

rm "$POP".iter.transformedThetas.borrar

# Calculo el número de sitios informativos, para todos los archivos:

FILES_DIVERSITY=("c_ll_ki_n013.per.unit.averages.tsv" "c_ll_po_n008.per.unit.averages.tsv" "c_lp_sm_n019.per.unit.averages.tsv" "c_ll_no_n008.per.unit.averages.tsv" "c_lp_do_n012.per.unit.averages.tsv")

# Añado una columna con el número de sitios informativos. 
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
echo ${i/.tsv/_informative_sites.tsv}
cat <(awk -v OFS='\t' '(NR==1) {print $0,"informative_sites"}' $i ) <(awk -v OFS='\t' 'NR >= 2 {print $0,$4-$5}' $i) > ${i/.tsv/_informative_sites.tsv}
mv ${i/.tsv/_informative_sites.tsv} $i
done

# Añado una columna con la poblacion
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
POP=$(echo ${i/.per.unit.averages.tsv/} )
awk -v OFS='\t' -v POP="$POP" '{print $0, POP}' $i > ${i/.per.unit.averages/_withPOP.per.unit.averages}
mv ${i/.per.unit.averages/_withPOP.per.unit.averages} $i
done

# Se me ha olvidado añadir el cabecero asi que lo modifico aparte. 
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
cat <(awk -v OFS='\t' '(NR==1) {$17="pop"; print}' $i) <(awk -v OFS='\t' 'NR >= 2 {print $0}' $i) > ${i/.tsv/_corrected.tsv}
mv ${i/.tsv/_corrected.tsv} $i
done


# Create a unique file with all the information

awk '
    FNR==1 && NR!=1 { while (/^<header>/) getline; }
    1 {print}' c_lp*.per.unit.averages.tsv > c_lp_do_n012-c_lp_sm_n019.per.unit.averages.tsv

awk '
    FNR==1 && NR!=1 { while (/^<header>/) getline; }
    1 {print}' c_ll*.per.unit.averages.tsv > c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.tsv



mv c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.tsv c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv 


mv c_lp_do_n012-c_lp_sm_n019.per.unit.averages.tsv c_lp_do_n012-c_lp_sm_n019.per.unit.averages.per_specie.tsv


FILES_DIVERSITY=("c_ll_ki_n013.per.unit.averages.tsv" "c_ll_po_n008.per.unit.averages.tsv" "c_lp_sm_n019.per.unit.averages.tsv" "c_ll_no_n008.per.unit.averages.tsv" "c_lp_do_n012.per.unit.averages.tsv" "c_lp_do_n012-c_lp_sm_n019.per.unit.averages.per_specie.tsv" "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.unit.averages.per_specie.tsv")

# Añado una columna con la specie
for i in "${FILES_DIVERSITY[@]}"
do
echo $i
POP=$(echo ${i/.per.unit.averages.tsv/} )
SPECIE=$(echo $POP | cut -d"_" -f 1-2)
echo $SPECIE
cat <(awk -v OFS='\t' '(NR==1) {$18="specie"; print}' $i) <(awk -v OFS='\t' -v SPECIE="$SPECIE" 'NR >= 2 {print $0, SPECIE}' $i) > ${i/.tsv/_corrected.tsv}
mv ${i/.tsv/_corrected.tsv} $i
done


```



#### Representation per pop

Este script nos devuelve una gráfica por cada población. Hay un segundo script que devuelve una gráfica para lynx lynx y otra para lynx pardinus. 

```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per.unit.averages.tsv$")

for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))

#### Para que almacene varias dataframe con el nombre correcto
#  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
#  assign(dataframename, read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".", colClasses=c("watterson_ave"="character")))

dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")

dataframe = dataframe[-1,]
dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
#dataframe  <- dataframe[complete.cases(dataframe),] ## --> No se que pasa que la columna de specie sale NA

########################################################
# Filtering
dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>0.5) %>% dplyr::filter(informative_sites>10)
name_diversity[1] <- paste(name_diversity[1],".percentagecovered05_mininformativesites10", sep="")
########################################################

ddply(dataframe, "feature", summarise, rating.mean=mean(pairwise_ave))
df.pm <- ddply(dataframe, "feature", summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))
write.csv(df.pm, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))


ddply(dataframe, "feature", summarise, rating.mean=mean(watterson_ave))
df.wm <- ddply(dataframe, "feature", summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))
write.csv(df.wm, paste(wd,"/", name_diversity[1], ".watterson_mean.csv", sep = ""))


# pdf(file = paste(wd,"/Diversity_by_feature_pairwise",name_diversity[1],".pdf", sep = ""))
ggplot(data = dataframe, aes(x=feature, y = pairwise_ave)) + 
  geom_boxplot()+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.pdf", sep = ""), device = "pdf")
# dev.off()



# pdf(file = paste(wd,"/Diversity_by_feature_watterson",name_diversity[1],".pdf", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = watterson_ave)) + 
  geom_boxplot()+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())

ggsave(filename = paste(wd,"/", name_diversity[1], ".diversity_feature_watterson.pdf", sep = ""), device = "pdf")


ggplot(data = dataframe, aes(x=feature, y = tajimaD)) + 
  geom_boxplot()+
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  #  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_feature_tajimaD.pdf", sep = ""), device = "pdf")


# Exploratory plots

jpeg(paste(wd,"/",name_diversity[1],".plot1.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot2.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$watterson_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot3.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot4.jpg", sep=""))
plot (dataframe$watterson_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot5.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$pairwise_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot6.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$watterson_ave)
dev.off()


}


# Cosas útiles outliers.shape=NA.

```

#### Representation per specie

Esta gráfica representa por especie.

```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library("tidyr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per_specie.tsv$")


for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))
  dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")
  dataframe = dataframe[-1,]
  dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
  dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
  dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
  dataframe  <- dataframe[complete.cases(dataframe),]
  dataframe$pop <- as.factor(dataframe$pop)
  dataframe$pop <- factor(dataframe$pop, levels=c("c_ll_ki_n013", "c_ll_po_n008", "c_ll_no_n008", "c_lp_sm_n019", "c_lp_do_n012"))
  #### Para que almacene varias dataframe con el nombre correcto
  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
  assign(dataframename, dataframe)
}


```

##### Lynx pardinus

```{r}

dataframe <- `data_diversity_c_lp_do_n012-c_lp_sm_n019`
name_diversity[1] <- "c_lp_do_n012-c_lp_sm_n019"


# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))

df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))


# LYNX PARDINUS
# Hago spread para calcular el porcentage que representa do respecto a sm
df.pm_lost <-tidyr::spread(df.pm, pop, rating.mean) %>%
 mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "rating.mean", -specie, -feature, -percentage_do_sm)
# Hago spread para calcular el porcentage de perdida
df.wm_lost <-tidyr::spread(df.wm, pop, rating.mean) %>% mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "rating.mean", -specie, -feature, -percentage_do_sm)


# Guardo el porcentage de perdida. 
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""))


ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
 # scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+
  geom_text(data = df.pm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.eps", sep = ""), device = "eps")
# dev.off()


ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) +
  geom_text(data = df.wm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_feature_watterson.eps", sep = ""), device="eps")






# Cosas útiles 

# geom_box(outliers.shape=NA)

# Si quiero plotear todos los indices a la vez:
#index = c("watterson_ave", "pairwise_ave")
#for (i in 1:length(index))
#{
#  ggplot(data = dataframe, aes_string(x='feature', y = index[1], fill='pop'))+
  
```

##### Lynx lynx


```{r}


dataframe <- `data_diversity_c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008`
name_diversity[1] <- "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008"

# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))

df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))

df.pm_lost <-tidyr::spread(df.pm, pop, rating.mean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "rating.mean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki)

df.wm_lost <-tidyr::spread(df.wm, pop, rating.mean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "rating.mean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki) 



# Guardo el porcentage de perdida. 
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  #scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+
  geom_text(data = df.pm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")

ggsave (file = paste(wd,"/",name_diversity[1],".diversity_feature_pairwise.eps", sep = ""), device = "eps")
# dev.off()



ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA, position=position_dodge(width=1.5))+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=1.5), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic feature") +
  xlab(label = "Genomic feature") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by feature",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) +
        geom_text(data = df.wm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")



ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_feature_watterson.eps", sep =""), device="eps")


# Cosas útiles outliers.shape=NA.



```


### Diversity per region.

Populations:

c_ll_ki
c_ll_no
c_ll_po

c_lp_sm
c_lp_do


```{bash}

#### CHANGE PATH!!!!

mkdir /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region

POP=c_ll_ki # check if I need to include n_013 <--- Change pop here!
screen -S "$POP"_scaffold_per_region

POP=c_ll_ki # check if I need to include n_013 <--- Change pop here!
script log_screen_"$POP"_scaffold_per_region.log

POP=c_ll_ki # check if I need to include n_013 <--- Change pop here!

## WG variables

RUTA=/GRUPOS/grupolince/telomers_centromers_definition/

cd $RUTA

REGIONS_PerScaffold=($(ls | grep "PerScaffold" | uniq))

for REGION_PerScaffold in "${REGIONS_PerScaffold[@]}"

do

echo $REGION_PerScaffold
cd $RUTA$REGION_PerScaffold

REGION=$(echo $REGION_PerScaffold | cut -d"." -f 1 )
SCAFFOLDS=($(ls * | cut -d'_' -f7 | uniq))

# mkdir /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region/$REGION/  # --> Ya están creadas.
DIVERSITY_PER_REGION_FOLDER=/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region/$REGION/
echo $REGION

echo "---------------------------------------------------$POP---------------------------------------------------"

cd $DIVERSITY_PER_REGION_FOLDER

# Create headers for the outfile
echo -e "scaffold\tstart\tend\tlength\tNAs\twatterson_ave\twatterson_sd\tpairwise_ave\tpairwise_sd\ttajimaD" > "$POP"."$REGION".averages.tsv

for SCAFFOLD in "${SCAFFOLDS[@]}"
do
echo "---------------------$SCAFFOLD---------------------"

while read LOCATION START_CEROBASED END CAT_CHR;

do

START_ONEBASED=($(echo $START_CEROBASED+1 | bc))
LENGTH=$(expr $END - $START_ONEBASED) 

#Intersect bed of the unit (first remove header)¿¿??

START_CEROBASED=($(echo $START_ONEBASED-1 | bc))



cat "$POP".transformedThetas_"$SCAFFOLD" | bedtools intersect -a stdin -b <(echo -e "$LOCATION\t$START_CEROBASED\t$END") > "$POP"."$REGION".iter.transformedThetas.borrar

WATTERSON_AVERAGE_PER_UNIT=$(cut -f 4 "$POP"."$REGION".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g')
WATTERSON_SD_PER_UNIT=$(cut -f 4 "$POP"."$REGION".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g')

PAIRWISE_AVERAGE_PER_UNIT=$(cut -f 5 "$POP"."$REGION".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g') 
PAIRWISE_SD_PER_UNIT=$(cut -f 5  "$POP"."$REGION".iter.transformedThetas.borrar | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT=$(cut -f 6  "$POP"."$REGION".iter.transformedThetas.borrar   |  awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR ))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

TAJIMAS_D_PER_UNIT=$(echo "(($PAIRWISE_AVERAGE_PER_UNIT) - ($WATTERSON_AVERAGE_PER_UNIT))/($DIFFERENCE_PAIRWISE_WATTERSON_SD_PER_UNIT)" | bc -l | awk '{printf ("%.10e\n",$1)}' |  sed 's/[eE]+\{0,1\}/*10^/g' )

NR=$(wc -l "$POP"."$REGION".iter.transformedThetas.borrar | cut -d" " -f1)

NAs=$(expr $LENGTH - $NR)

# Calculate averages, and standatd deviations and paste them

paste \
<(echo $LOCATION ) \
<(echo $START_ONEBASED ) \
<(echo $END ) \
<(echo $LENGTH ) \
<(echo $NAs) \
<(echo $WATTERSON_AVERAGE_PER_UNIT) \
<(echo $WATTERSON_SD_PER_UNIT) \
<(echo $PAIRWISE_AVERAGE_PER_UNIT) \
<(echo $PAIRWISE_SD_PER_UNIT) \
<(echo $TAJIMAS_D_PER_UNIT) |\
sed 's/ /\t/g'| sed 's/\t\+/\t/g' >>  "$POP"."$REGION".averages.tsv

done < <(cat "$SCAFFOLDS_FOLDER""$SCAFFOLD") 
done &
done

rm "$POP"*iter.transformedThetas.borrar


#############################





# Muevo todos los archivos a la carpeta madre

mv /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region/*tsv /home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region

# Elimino las carpetas donde estaban que ahora están vacías
 
rm -r telomers2m/
rm -r telomers0-10m/
rm -r centromers/


FILES_DIVERSITY=("c_lp_do_n012.centromers.averages.tsv" "c_ll_ki_n013.centromers.averages.tsv" "c_ll_po_n008.centromers.averages.tsv" "c_ll_no_n008.centromers.averages.tsv" "c_lp_sm_n019.centromers.averages.tsv" "c_lp_do_n012.telomers0-10m.averages.tsv" "c_ll_po_n008.telomers0-10m.averages.tsv" "c_ll_no_n008.telomers0-10m.averages.tsv" "c_ll_ki_n013.telomers0-10m.averages.tsv" "c_lp_sm_n019.telomers0-10m.averages.tsv" "c_lp_do_n012.telomers2m.averages.tsv" "c_ll_po_n008.telomers2m.averages.tsv" "c_ll_ki_n013.telomers2m.averages.tsv" "c_ll_no_n008.telomers2m.averages.tsv" "c_lp_sm_n019.telomers2m.averages.tsv")



# Añado varias columnas simulando el archivo de per unit por si alguna vez queremos juntarlo en uno, con esto estoy añadiendo numero de sitios informativos, feature (que en este caso es tel, centr), pop, specie, y dejo en blanco pero también añado otras columnas como frame, id_gene. 

for i in "${FILES_DIVERSITY[@]}"
do
echo $i
POP=$(echo ${i} | cut -d "." -f 1 )
SPECIE=$(echo $POP | cut -d"_" -f 1-2)
FEATURE=$(echo ${i} | cut -d "." -f 2 )
cat <(awk -v OFS='\t' '(NR==1) {print $1, $2, $3, $4, $5,"feature","strandness","frame","id_gene","id", $6,$7,$8,$9,$10,"informative_sites","pop","specie"}' $i) <( awk -v OFS='\t' -v POP="$POP" -v SPECIE="$SPECIE" -v FEATURE="$FEATURE" 'NR >= 2 {print $1, $2, $3, $4, $5, FEATURE, "na","na","na","na",$6,$7,$8,$9,$10,$4-$5, POP, SPECIE}' $i) > ${i/.tsv/_corrected.tsv}
mv ${i/.tsv/_corrected.tsv} $i
done


# Create a unique file with all the information per specie
# ojo que para per feature he usado otro script, que no se porque no me funcionaba aquí.

awk 'FNR==1 && NR!=1{next;}{print}' c_lp*averages.tsv > c_lp_do_n012-c_lp_sm_n019.per.region.averages.per_specie.tsv

awk 'FNR==1 && NR!=1{next;}{print}' c_ll*averages.tsv > c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008.per.region.averages.per_specie.tsv

# Each pop

awk 'FNR==1 && NR!=1{next;}{print}' c_ll_ki_n013*averages.tsv > c_ll_ki_n013.per.region.averages.tsv

awk 'FNR==1 && NR!=1{next;}{print}' c_ll_no_n008*averages.tsv > c_ll_no_n008.per.region.averages.tsv

awk 'FNR==1 && NR!=1{next;}{print}' c_ll_po_n008*averages.tsv > c_ll_po_n008.per.region.averages.tsv

awk 'FNR==1 && NR!=1{next;}{print}' c_lp_do_n012*averages.tsv > c_lp_do_n012.per.region.averages.tsv

awk 'FNR==1 && NR!=1{next;}{print}' c_lp_sm_n019*averages.tsv > c_lp_sm_n019.per.region.averages.tsv



```



#### Representation per pop

Este script nos devuelve una gráfica por cada población. Hay un segundo script que devuelve una gráfica para lynx lynx y otra para lynx pardinus. 



```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_region"

finsdiversity = list.files(path = wd,pattern="*.per.region.averages.tsv$")

for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))

#### Para que almacene varias dataframe con el nombre correcto
#  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
#  assign(dataframename, read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".", colClasses=c("watterson_ave"="character")))

dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")

dataframe = dataframe[-1,]
dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
#dataframe  <- dataframe[complete.cases(dataframe),] ## --> No se que pasa que la columna de specie sale NA

########################################################
# Filtering
dataframe <- dataframe %>% 
  mutate(percentage_covered=ifelse(NAs>0, (as.numeric(length)-as.numeric(NAs))/as.numeric(length), 1)) %>% dplyr::filter(percentage_covered>0.5) %>% dplyr::filter(informative_sites>10)
name_diversity[1] <- paste(name_diversity[1],".percentagecovered05_mininformativesites10", sep="")
########################################################

ddply(dataframe, "feature", summarise, rating.mean=mean(pairwise_ave))
df.pm <- ddply(dataframe, "feature", summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))
write.csv(df.pm, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))


ddply(dataframe, "feature", summarise, rating.mean=mean(watterson_ave))
df.wm <- ddply(dataframe, "feature", summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))
write.csv(df.wm, paste(wd,"/", name_diversity[1], ".watterson_mean.csv", sep = ""))


# pdf(file = paste(wd,"/Diversity_by_region_pairwise",name_diversity[1],".pdf", sep = ""))
ggplot(data = dataframe, aes(x=feature, y = pairwise_ave)) + 
  geom_boxplot()+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.pdf", sep = ""), device = "pdf")
# dev.off()



# pdf(file = paste(wd,"/Diversity_by_region_watterson",name_diversity[1],".pdf", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = watterson_ave)) + 
  geom_boxplot()+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean),shape = 23, 
             size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())

ggsave(filename = paste(wd,"/", name_diversity[1], ".diversity_region_watterson.pdf", sep = ""), device = "pdf")


ggplot(data = dataframe, aes(x=feature, y = tajimaD)) + 
  geom_boxplot()+
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  #  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_tajimaD.pdf", sep = ""), device = "pdf")


# Exploratory plots

jpeg(paste(wd,"/",name_diversity[1],".plot1.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot2.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$watterson_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],".plot3.jpg", sep=""))
plot (dataframe$pairwise_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot4.jpg", sep=""))
plot (dataframe$watterson_ave, dataframe$tajimaD)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot5.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$pairwise_ave)
dev.off()

jpeg(paste(wd,"/",name_diversity[1],"plot6.jpg", sep=""))
plot (dataframe$informative_sites, dataframe$watterson_ave)
dev.off()


}


# Cosas útiles outliers.shape=NA.

```

#### Representation per specie

Esta gráfica representa por especie.

```{r}

library ("ggplot2")
library ("plyr")
library ("RColorBrewer")
library ("scales")
library ("magrittr")
library("tidyr")

# wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/whole_genome_analysis/sfs/diversity_per_unit"
wd <- "/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region"
# wd <- "/Users/marialucenaperez/Desktop/Diversity_per_unit"

finsdiversity = list.files(path = wd,pattern="*per_specie.tsv$")


for (i in 1:length(finsdiversity))
{
  infile <- finsdiversity[i]
  name_diversity <- unlist(strsplit(finsdiversity[i], "[.]"))
  dataframe <- read.csv (paste(wd,infile, sep="/"), header = T, sep = '\t',stringsAsFactors = FALSE,  row.names=NULL, na.strings = "", dec=".")
  dataframe = dataframe[-1,]
  dataframe$watterson_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$watterson_ave))
  dataframe$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",dataframe$pairwise_ave))
  dataframe$tajimaD <- as.numeric(gsub("\\*10\\^","e",dataframe$tajimaD))
  dataframe  <- dataframe[complete.cases(dataframe),]
  dataframe$pop <- as.factor(dataframe$pop)
  dataframe$pop <- factor(dataframe$pop, levels=c("c_ll_ki_n013", "c_ll_po_n008", "c_ll_no_n008", "c_lp_sm_n019", "c_lp_do_n012"))
  #### Para que almacene varias dataframe con el nombre correcto
  dataframename <- paste("data_diversity", name_diversity[1], sep ='_')
  assign(dataframename, dataframe)
}


```

##### Lynx pardinus

```{r}

dataframe <- `data_diversity_c_lp_do_n012-c_lp_sm_n019`
name_diversity[1] <- "c_lp_do_n012-c_lp_sm_n019"


# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))

df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))


# LYNX PARDINUS
# Hago spread para calcular el porcentage que representa do respecto a sm
df.pm_lost <-tidyr::spread(df.pm, pop, rating.mean) %>%
 mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "rating.mean", -specie, -feature, -percentage_do_sm)
# Hago spread para calcular el porcentage de perdida
df.wm_lost <-tidyr::spread(df.wm, pop, rating.mean) %>% mutate(., percentage_do_sm=(c_lp_do_n012 *100)/c_lp_sm_n019)
# Hago gathered para poder plotearlo.
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "rating.mean", -specie, -feature, -percentage_do_sm)


# Guardo el porcentage de perdida. 
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""))


ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
 # scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+
  geom_text(data = df.pm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.eps", sep = ""), device = "eps")
# dev.off()


ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) +
  geom_text(data = df.wm.gathered, aes(feature, percentage_do_sm, label=(percentage_do_sm %>% round(2))),angle = 45, size=2)
ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_watterson.eps", sep = ""), device="eps")






# Cosas útiles 

# geom_box(outliers.shape=NA)

# Si quiero plotear todos los indices a la vez:
#index = c("watterson_ave", "pairwise_ave")
#for (i in 1:length(index))
#{
#  ggplot(data = dataframe, aes_string(x='feature', y = index[1], fill='pop'))+
  
```

##### Lynx lynx


```{r}


dataframe <- `data_diversity_c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008`
name_diversity[1] <- "c_ll_ki_n013-c_ll_no_n008-c_ll_po_n008"

# Calculo weighted mean y lo guardo en un df.

df.pm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(pairwise_ave, informative_sites))

df.wm <- ddply(dataframe, c("feature","specie","pop"), summarise, rating.mean=weighted.mean(watterson_ave, informative_sites))

df.pm_lost <-tidyr::spread(df.pm, pop, rating.mean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.pm.gathered <- tidyr::gather(df.pm_lost, "pop", "rating.mean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki)

df.wm_lost <-tidyr::spread(df.wm, pop, rating.mean) %>% 
  mutate(., percentage_po_ki=(c_ll_po_n008 *100)/c_ll_ki_n013) %>% 
  mutate(., percentage_no_ki=(c_ll_no_n008*100)/c_ll_ki_n013)
df.wm.gathered <- tidyr::gather(df.wm_lost, "pop", "rating.mean", -specie, -feature,  -percentage_po_ki, -percentage_no_ki) 



# Guardo el porcentage de perdida. 
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".pairwise_mean.csv", sep = ""))
write.csv(df.wm_lost, paste(wd,"/", name_diversity[1],".watterson_mean.csv", sep = ""))

ggplot(data = dataframe, aes(x=feature, y = pairwise_ave, fill=pop)) +
  geom_boxplot(outlier.shape=NA)+
  geom_point(data=df.pm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=0.75), size = 3, inherit.aes=FALSE) +
  # scale_y_continuous(limits = c(-20,-3), expand =c(0,0), labels = comma) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Pairwise average") + # y title
  theme_bw() +  #theme selection for background and lines
  #scale_x_continuous(expand = c(0, 1)) +# + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank())+
  geom_text(data = df.pm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")

ggsave (file = paste(wd,"/",name_diversity[1],".diversity_region_pairwise.eps", sep = ""), device = "eps")
# dev.off()



ggplot(data = dataframe, aes(x=feature, y = watterson_ave, fill =pop)) + 
  geom_boxplot(outlier.shape=NA, position=position_dodge(width=1.5))+
  geom_point(data=df.wm,aes(x=feature, y = rating.mean, fill=pop),shape = 23, position=position_dodge(width=1.5), size = 3, inherit.aes=FALSE) +
  facet_grid(.~feature, scales="free") +
  scale_y_continuous(trans = 'log10') +
  scale_x_discrete("Genomic region") +
  xlab(label = "Genomic region") + #x title
  ylab(label = "Watterson average") + # y title
  theme_bw() +  #theme selection for background and lines
  # scale_x_continuous(expand = c(0, 0)) + scale_y_continuous(expand = c(0, 0)) +
  ggtitle(paste ("Diversity by region",name_diversity[1], sep = " " )) +
  theme(axis.line.x = element_line(color="black", size = 0.5),
        axis.line.y = element_line(color="black", size = 0.5),
        axis.text.x = element_text(angle = 60, vjust = 0.8, hjust = 1),
        axis.text.y = element_text(vjust = 0.2, hjust = 0.2),
        axis.title.y = element_text(margin=margin(r=0.3, unit="cm")),
        axis.title.x = element_text(margin=margin(t=0.5, unit="cm")),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank()) +
        geom_text(data = df.wm.gathered, aes(feature, percentage_po_ki, label=paste((percentage_po_ki %>% round(2)),(percentage_no_ki%>% round(2)), sep='\n')), angle = 45, size=2, vjust="inward",hjust="inward")


ggsave(filename = paste(wd,"/",name_diversity[1],".diversity_region_watterson.eps", sep = ""), device="eps")


# Cosas útiles outliers.shape=NA.



```

#### Diversity per region: Cat Coordinates

He probado con el script de Fede de perl:

```{bash}

# Primero le quito el cabecero porque si no, no puede usarse bedintersect:
# sed '1d' c_ll_ki_n013.per.region.averages.tsv  > c_ll_ki_n013.per.region.averages.NOHEADER.tsv 
# Ahora lo corro tal y como indica el propio script:
# perl lynx2cat.pl c_ll_ki_n013.per.region.averages.NOHEADER.tsv > c_ll_ki_n013.per.region.averages.CatCoordinates.tsv
# Este script me ha dado el siguiente error:
#Use of uninitialized value $s in concatenation (.) or string at lynx2cat.pl line 14, <I> line 156394492.
#Use of uninitialized value $e in concatenation (.) or string at lynx2cat.pl line 14, <I> line 156394492.
# Además me he dado cuenta que solo reporta las coordenadas de uno y otro, y yo quiero toda la información. 

```

Por tanto pruebo con el básico:

```{bash}

intersectBed -sorted -wo -a c_ll_ki_n013.per.region.averages.NOHEADER.tsv -b lynx2cat_wTiger.bed > c_ll_ki_n013.per.region.averages.CatCoordinates.tsv

# Me da el error:

# ERROR: chromomsome sort ordering for file c_ll_ki_n013.per.region.averages.NOHEADER.tsv is inconsistent with other files. Record was:
# lp23.s00015     2903992 3006783 102791  101509  telomers0-10m   na      na      na      na      6.5495647224*10^-06     1.4900294206*10^-05        2.5287837844*10^-06     5.8944110986*10^-06     -4.4603274016*10^-01    1282    c_ll_ki_n013    c_ll

```


Así que ordeno antes:

```{bash}

 bedtools sort c_ll_ki_n013.per.region.averages.NOHEADER.tsv > c_ll_ki_n013.per.region.averages.NOHEADER.SORTED.tsv

```




# Copy

```{bash}

scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_unit/*eps /Users/marialucenaperez/ownCloud/publico/WG_diversity/diversity_plots/diversity_per_unit 


scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/whole_genome_analysis/sfs/diversity_per_region/*eps /Users/marialucenaperez/ownCloud/publico/WG_diversity/diversity_plots/diversity_per_region/



```




### Diversity per region & feature


```{bash}




```



