---
title: "pipeline_contemporary_sfs"
output: html_document
---

# Medidas en ventanas

# Generating SAF for intergenic data

## Defining variables:

I checked that all bam files were indexed --> OK
Ruta a bam files intergenic:

Global variables:

```{r, engine=bash, eval=FALSE}

RUTA=/home/mlucena/ANGSD_analysis 

cd $RUTA/intergenic_analysis/intergenic_sfs

#To launch one by one

POP="c_ll_ur"  # <--CHANGE POP HERE

#screen -S "$POP"_sfs
screen -S "$POP"_thetas

POP="c_ll_ur"  # <--CHANGE POP HERE

#script "$POP"_sfs.log
script "$POP"_thetas.log

POP="c_ll_ur"  # <--CHANGE POP HERE

THREADS=25                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!
RUTA=/home/mlucena/ANGSD_analysis/depth_calculus 

read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated < $RUTA/"${POP}"_n*_mean_sd_depthGlobal_lynx_per_pop_mean_folds_0.95.csv


ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)

# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
echo $minDepth


```

## Unfolded SAF 

```{r, engine=bash, eval=FALSE}

#The read command will read one line at a time from mean_sd_depthGlobal_lynx_per_pop_sd_folds2.csv
# put the first field (fields being blank separated, and where backslash can be used to escape the field or line separator) in the variable $discard, the second in $x, the third in $y and the rest of the fields in $discard.
#while read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated; 
#do

# Anyway I will do it one by one:

##########################
#  SAF (likelihood):     
##########################

echo "-------$POP----------SAF (likelihood)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b /home/mlucena/ANGSD_analysis/intergenic_analysis/"$POP".intergenic.bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth -setMinDepth $minDepth

```

## SFS 

```{r, engine=bash, eval=FALSE}

##########################
#  SFS                   #(I dont require the -rf file as the saf already only contains the -rf sites
##########################
echo "-------$POP----------SFS------------------------------------------------------"

$NGSTOOLS/realSFS "$POP".unfolded-lr.saf.idx  -P $THREADS > "$POP".unfolded-lr.sfs

```

##  SAF (postprob) 

```{r, engine=bash, eval=FALSE}


echo "-------$POP----------SAF (postprob)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b /home/mlucena/ANGSD_analysis/intergenic_analysis/"$POP".intergenic.bamlist -ref $REF -anc $ANC \
-out "$POP".unfolded-lr.postprob \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth \
-doThetas 1 -pest "$POP".unfolded-lr.sfs 

```




##  Make bed theta        

```{r, engine=bash, eval=FALSE}

echo "-------$POP----------Make bed theta -----------------------------------------"

$NGSTOOLS/thetaStat print $POP.unfolded-lr.postprob.thetas.idx > $POP.printed.stats

# Transform the coordinates to use 0Based bedtools

awk 'NR>1 {print  $1" "$2-1" "$2}' $POP.printed.stats > $POP.0basedcoordinates.borrar # --> 1, 2, 3
# log transformation of watterson and pairwise:
awk 'NR>1 {print  "e("$3")"}' $POP.printed.stats | bc -l > "$POP".watterson.borrar # --> 4
awk 'NR>1 {print  "e("$4")"}' $POP.printed.stats | bc -l > "$POP".pairwise.borrar # --> 5
                                                                                  # --> PAIRWISE - WATTERSON
awk 'NR>1 {print  "e("$5")"}' $POP.printed.stats | bc -l > "$POP".thetaFL.borrar # --> 6
awk 'NR>1 {print  "e("$6")"}' $POP.printed.stats | bc -l > "$POP".thetaH.borrar # --> 7
awk 'NR>1 {print  "e("$7")"}' $POP.printed.stats | bc -l > "$POP".thetaL.borrar # --> 8

# Create a header for the "$POP".transformedThetas file

echo "scaffold position1 position2 watterson pairwise pairwise-watterson thetaFL thetaH thetaL" > "$POP".transformedThetas

paste $POP.0basedcoordinates.borrar "$POP".watterson.borrar "$POP".pairwise.borrar "$POP".thetaFL.borrar "$POP".thetaH.borrar "$POP".thetaL.borrar | \
awk -v OFS='\t'  '{print $1, $2, $3, $4, $5, $5-$4, $6, $7, $8}' >> "$POP".transformedThetas



# Hasta aquí!

rm "$POP".*.borrar


scp *.printed.stats /backup/mlucena/intermediate_files_ANGSD/intergenic_analysis/



```



```{bash}
nano heterozigosity_calculus.R
```

```{bash}
#in R
library (dplyr)
args <- commandArgs(TRUE)
print(args)
a<-scan(args[1])
heterozigosity <- a[2]/sum(a)
write (heterozigosity,paste0(args[1],"_heterozigosity.csv"))

```

```{bash}

for i in *sfs;
do
echo $i
Rscript heterozigosity_calculus.R $i 
done

rm heterozigosity_per_pop.csv

for i in *heterozigosity_borrar.csv
do
echo $i
heterozigosity=$(cat $i)
echo ${i/.unfolded-lr.sfs_heterozigosity_borrar.csv/} $heterozigosity >> heterozigosity_per_pop.csv
done

rm *_heterozigosity_borrar.csv

```


# Pairwise-Watterson estimation genome wide. 


```{bash}

screen -S diveristy_summary_per_pop
script diveristy_summary_per_pop.log


POPS=($(ls *sfs |  sed -e "s/.unfolded-lr.sfs//" ))


# Create headers for the outfile
echo -e "pop\tpositions\twatterson_ave\twatterson_sd\tpairwise_ave\tpairwise_sd\ttajimaD" > diversity.per.pop.averages.tsv

for POP in "${POPS[@]}"
do
echo "---------------------------------------------------$POP---------------------------------------------------"

WATTERSON_AVERAGE=$(cut -f 4 "$POP".transformedThetas | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g')

WATTERSON_SD=$(cut -f 4 "$POP".transformedThetas | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g')

PAIRWISE_AVERAGE=$(cut -f 5 "$POP".transformedThetas | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

PAIRWISE_SD=$(cut -f 5 "$POP".transformedThetas | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

DIFFERENCE_PAIRWISE_WATTERSON_SD=$(cut -f 6  "$POP".transformedThetas |  awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR ))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

TAJIMAS_D=$(echo "(($PAIRWISE_AVERAGE) - ($WATTERSON_AVERAGE))/($DIFFERENCE_PAIRWISE_WATTERSON_SD)" | bc -l | awk '{printf ("%.10e\n",$1)}' |  sed 's/[eE]+\{0,1\}/*10^/g' )

NR=$(wc -l  "$POP".transformedThetas  | cut -d" " -f1)

paste \
<(echo $POP) \
<(echo $NR) \
<(echo $WATTERSON_AVERAGE) \
<(echo $WATTERSON_SD) \
<(echo $PAIRWISE_AVERAGE) \
<(echo $PAIRWISE_SD) \
<(echo $TAJIMAS_D) |\
sed 's/ /\t/g'| sed 's/\t\+/\t/g' >>  diversity.per.pop.averages.tsv

done 

```


## SFS graphical representation


First we copy the SFS files to our local folder.

```{bash}
scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*sfs /Users/marialucenaperez/Dropbox/phd/contemporary/ANGSD/intergenic_analysis/analysis_sfs

# I need to change name as R does not like "-lr.sfs"

cd /Users/marialucenaperez/Dropbox/phd/contemporary/ANGSD/intergenic_analysis/analysis_sfs


for i in *sfs
do 
#echo $i
echo $i "--->" ${i/-lr.sfs/_lr.sfs}
mv $i ${i/-lr.sfs/_lr.sfs}
rm $i
done

## OJO!!!!! READMEEEEEE!!!!! #######

# Le he quitado a mano a c_ll_to-c_ll_ka-c_ll_og_n008.unfolded_lr.sfs los guiones medios porque si no, no corre en R!!

```

Now we run the script. 

 
```{bash}
nano SFS_graphical_representation
```

```{r}
library(dplyr)
library(magrittr)

wd<-"/Users/marialucenaperez/Dropbox/phd/contemporary/ANGSD/intergenic_analysis/analysis_sfs/"

my_files_unfolded = list.files(path = wd, pattern="*sfs$")    
for (i in 1:length(my_files_unfolded)) 
{ assign(my_files_unfolded[i], (scan(paste(wd,my_files_unfolded[i], sep = ""), sep = " ", dec = ".")) %>% .[!is.na(.)])}

norm <- function(x) x/sum(x)

for (i in 1:length (my_files_unfolded))
{
  temp1=(eval(parse(text=my_files_unfolded[i])))
  temp2=norm(temp1[-1])
  temp3=norm(temp2[-c(length(temp2))])
  barplot(temp3,xlab="Minor allele frequency",
          names=1:length(temp3),ylab="Proportions",main=my_files_unfolded[i],col='grey')
  dev.print(device=postscript, paste("/Users/marialucenaperez/Dropbox/phd/contemporary/ANGSD/intergenic_analysis/analysis_sfs/",my_files_unfolded[i],".eps", sep=""), onefile=FALSE, horizontal=FALSE)
  dev.off()
  rm(temp1)
  rm(temp2)
  rm(temp3)
}

```

# SAF variable_sites 1

As we find that balkans have a wierd behaviour, we are going to repeat all the stats, using onel sites that were already filter out by SNP-pval 1e-4 for PCA and ADMIXTURE. We found that this threshold is already stable, and therefore, SNPs called are trustworthy. 

First we are going to select this sites, using PCA maf file.


```{bash}
# This is how it looks:

mkdir /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/intergenic_sfs_filtered_sites

cd /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_PCA
zcat c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n080.intergenic.mafs.gz | head
# chromo	position	major	minor	ref	knownEM	pK-EM	nInd
# lp23.s00001	3926	G	A	N	0.020146	9.075109e-08	80
# lp23.s00001	5895	T	A	N	0.031340	3.633791e-08	80
# lp23.s00001	6927	C	T	C	0.015890	0.000000e+00	80
# lp23.s00001	9059	C	T	N	0.064255	0.000000e+00	80
# lp23.s00001	13418	G	A	A	0.065885	0.000000e+00	80
# lp23.s00001	13869	T	G	T	0.025897	0.000000e+00	80
# lp23.s00001	14821	T	G	T	0.041811	0.000000e+00	80

# We are making a site file using the first column (See ANGSD documentation)

cd /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/intergenic_sfs_filtered_sites

zcat /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_PCA/c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n080.intergenic.mafs.gz | awk  'BEGIN{OFS="\t";} ; NR>1{print $1,$2}' > PCA_Maf1_SNP-pval1e_4.sites

angsd sites index PCA_Maf1_SNP-pval1e_4.sites

```


```{bash}

screen -S diversity_calculus_subset

POPS=$(ls /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.transformedThetas | cut -d"/" -f 7)


echo -e "pop\tpairwise\twatterson"  > Maf1_SNP-pval1e_4_diversity_averages_per_pop.tsv

for POP in ${POPS[@]}

do

echo "grep for" $POP

grep -wFf PCA_Maf1_SNP-pval1e_4.sites /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/$POP > ${POP/.transformedThetas/_Maf1_SNP-pval1e_4.transformedThetas.borrar}

WATTERSON_AVERAGE=$(cut -f 4 ${POP/.transformedThetas/_Maf1_SNP-pval1e_4.transformedThetas.borrar} | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g')

PAIRWISE_AVERAGE=$(cut -f 5 ${POP/.transformedThetas/_Maf1_SNP-pval1e_4.transformedThetas.borrar} | awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i }} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sum[i]/NR )}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

DIFFERENCE_PAIRWISE_WATTERSON_SD=$(cut -f 6  ${POP/.transformedThetas/_Maf1_SNP-pval1e_4.transformedThetas.borrar} |  awk -v OFS='\t' '{for(i=1;i<=NF;i++) {sum[i] += $i; sumsq[i] += ($i)^2}} END {for (i=1;i<=NF;i++) {printf ("%.10e\n", sqrt((sumsq[i]-sum[i]^2/NR)/NR ))}}' | sed 's/[eE]+\{0,1\}/*10^/g') 

TAJIMAS_D=$(echo "(($PAIRWISE_AVERAGE) - ($WATTERSON_AVERAGE))/($DIFFERENCE_PAIRWISE_WATTERSON_SD)" | bc -l | awk '{printf ("%.10e\n",$1)}' |  sed 's/[eE]+\{0,1\}/*10^/g' )

paste \
<(echo ${POP/.transformedThetas/}) \
<(echo $WATTERSON_AVERAGE) \
<(echo $PAIRWISE_AVERAGE) \
<(echo $TAJIMAS_D) |\
sed 's/ /\t/g'| sed 's/\t\+/\t/g' >> Maf1_SNP-pval1e_4_diversity_averages_per_pop.tsv

rm *borrar

done





```


# SAF variable_sites 2

I will also check if this estimation is correct by running one of my pops from the very beginning using this sites, and then compare the diversity. 


```{r, engine=bash, eval=FALSE}

cd /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs_filtered_sites

#To launch one by one

POP="c_ll_ur"  # <--CHANGE POP HERE

#screen -S "$POP"_sfs
screen -S "$POP"_thetas_filtered

POP="c_ll_ur"  # <--CHANGE POP HERE

#script "$POP"_sfs.log
script "$POP"_thetas_filtered.log

POP="c_ll_ur"  # <--CHANGE POP HERE

THREADS=25                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!
RUTA=/home/mlucena/ANGSD_analysis/depth_calculus 

read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated < $RUTA/"${POP}"_n*_mean_sd_depthGlobal_lynx_per_pop_mean_folds_0.95.csv

ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)
SITES="PCA_Maf1_SNP-pval1e_4.sites"


# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
echo $minDepth
echo $SITES

```

## Unfolded SAF 

```{r, engine=bash, eval=FALSE}

#The read command will read one line at a time from mean_sd_depthGlobal_lynx_per_pop_sd_folds2.csv
# put the first field (fields being blank separated, and where backslash can be used to escape the field or line separator) in the variable $discard, the second in $x, the third in $y and the rest of the fields in $discard.
#while read POP mean sd mean_truncated sd_truncated maxDepth minDepth maxDepth_truncated minDepth_truncated; 
#do

# Anyway I will do it one by one:

##########################
#  SAF (likelihood):     
##########################

echo "-------$POP----------SAF (likelihood)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b /home/mlucena/ANGSD_analysis/intergenic_analysis/"$POP".intergenic.bamlist -ref $REF -anc $ANC \
-out "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr \
$FILTER1 \
$FILTER2 \
-sites $SITES \
-GL 1 -doSaf 1  \
-minInd  $MIN_IND -setMaxDepth $maxDepth -setMinDepth $minDepth

```

Sanity check

```{bash}

$NGSTOOLS/realSFS print  c_ll_ur_n006_filtered_Maf1_SNP-pval1e_4.unfolded_lr.saf.idx | wc -l
# 2058550

wc -l PCA_Maf1_SNP-pval1e_4.sites
# 2096552 PCA_Maf1_SNP-pval1e_4.sites

$NGSTOOLS/realSFS print  c_ll_ur_n006_filtered_Maf1_SNP-pval1e_4.unfolded_lr.saf.idx | awk  'BEGIN{OFS="\t";} ;{print $1, $2}'  > sitios_SAF.sites

 wc -l sitios_SAF.sites 
# 2058550 sitios_SAF.sites


diff sitios_SAF.sites PCA_Maf1_SNP-pval1e_4.sites | head
# 23a24
# > lp23.s00001	53180
# 50a52
# > lp23.s00001	95912
# 57a60
# > lp23.s00001	120814
# 59a63,65
# > lp23.s00001	121922
# > lp23.s00001	121934
# > lp23.s00001	121970

# ¿Todos los sitios de sitios_SAF.sites están en PCA_Maf1_SNP-pval1e_4.sites?

diff sitios_SAF.sites PCA_Maf1_SNP-pval1e_4.sites | grep "<"  
#Nada! Ergo, sí. 

# Voy a buscar este sitio en el SAF de todo el genoma:
# lp23.s00001	53180

cd ../intergenic_sfs

# $NGSTOOLS/realSFS print c_ll_ur_n006.unfolded-lr.saf.idx | grep "lp23.s00001" | grep "53180"


```



## SFS 

```{r, engine=bash, eval=FALSE}

##########################
#  SFS                   #(I dont require the -rf file as the saf already only contains the -rf sites
##########################
echo "-------$POP----------SFS------------------------------------------------------"

$NGSTOOLS/realSFS "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr.saf.idx  -P $THREADS > "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr.sfs

```

##  SAF (postprob) 

```{r, engine=bash, eval=FALSE}


echo "-------$POP----------SAF (postprob)-----------------------------------------"

$ANGSD/angsd -P $THREADS -b /home/mlucena/ANGSD_analysis/intergenic_analysis/"$POP".intergenic.bamlist -ref $REF -anc $ANC \
-out "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr.postprob \
$FILTER1 \
$FILTER2 \
-GL 1 -doSaf 1  \
-sites $SITES \
-minInd  $MIN_IND -setMaxDepth $maxDepth \
-doThetas 1 -pest "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr.sfs 

```




##  Make bed theta        

```{r, engine=bash, eval=FALSE}

echo "-------$POP----------Make bed theta -----------------------------------------"


# Ojo si pop tiene la n006 o no!

$NGSTOOLS/thetaStat print "$POP"_filtered_Maf1_SNP-pval1e_4.unfolded_lr.postprob.thetas.idx > "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats

# Transform the coordinates to use 0Based bedtools

awk 'NR>1 {print  $1" "$2-1" "$2}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats > "$POP"_filtered_Maf1_SNP-pval1e_4.0basedcoordinates.borrar # --> 1, 2, 3
# log transformation of watterson and pairwise:
awk 'NR>1 {print  "e("$3")"}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats | bc -l > "$POP"_filtered_Maf1_SNP-pval1e_4.watterson.borrar # --> 4
awk 'NR>1 {print  "e("$4")"}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats | bc -l > "$POP"_filtered_Maf1_SNP-pval1e_4.pairwise.borrar # --> 5
                                                                                  # --> PAIRWISE - WATTERSON
awk 'NR>1 {print  "e("$5")"}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats | bc -l > "$POP"_filtered_Maf1_SNP-pval1e_4.thetaFL.borrar # --> 6
awk 'NR>1 {print  "e("$6")"}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats | bc -l > "$POP"_filtered_Maf1_SNP-pval1e_4.thetaH.borrar # --> 7
awk 'NR>1 {print  "e("$7")"}' "$POP"_filtered_Maf1_SNP-pval1e_4.printed.stats | bc -l > "$POP"_filtered_Maf1_SNP-pval1e_4.thetaL.borrar # --> 8

# Create a header for the "$POP"_filtered_Maf1_SNP-pval1e_4.transformedThetas file

echo "scaffold position1 position2 watterson pairwise pairwise-watterson thetaFL thetaH thetaL" > "$POP"_filtered_Maf1_SNP-pval1e_4.transformedThetas

paste "$POP"_filtered_Maf1_SNP-pval1e_4.0basedcoordinates.borrar "$POP"_filtered_Maf1_SNP-pval1e_4.watterson.borrar "$POP"_filtered_Maf1_SNP-pval1e_4.pairwise.borrar "$POP"_filtered_Maf1_SNP-pval1e_4.thetaFL.borrar "$POP"_filtered_Maf1_SNP-pval1e_4.thetaH.borrar "$POP"_filtered_Maf1_SNP-pval1e_4.thetaL.borrar | \
awk -v OFS='\t'  '{print $1, $2, $3, $4, $5, $5-$4, $6, $7, $8}' >> "$POP"_filtered_Maf1_SNP-pval1e_4.transformedThetas



# Hasta aquí!

rm "$POP"_filtered_Maf1_SNP-pval1e_4.*.borrar


scp *.printed.stats /backup/mlucena/intermediate_files_ANGSD/intergenic_analysis/


  
  



```

##

```{bash}
scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/diversity.per.pop.averages.tsv /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/
```

## Plotting diversity

```{r}
library(dplyr) 
library(RColorBrewer)
library(ggplot2)

wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/" 
diversity <- read.csv(paste(wd, "diversity.per.pop.averages.tsv", sep =""), sep ="\t", dec=".", header=T) %>%  
 mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA)))))))))))))))  %>% 
filter(Populations!="Balkans") 
  
diversity$watterson_ave <- as.numeric(gsub("\\*10\\^","e",diversity$watterson_ave))
diversity$pairwise_ave <- as.numeric(gsub("\\*10\\^","e",diversity$pairwise_ave))
diversity$tajimaD <- as.numeric(gsub("\\*10\\^","e",diversity$tajimaD))
  
diversity$Populations <- factor (diversity$Populations, levels=c("Norway", "Balkans", "Carpathians", "Bialowieza", "Latvia", "Kirov", "Urals", "Tuva", "Khentii", "Töv", "Ömnögovi", "Mongolia",  "Yakutia", "Vladivostok"))

 cols <- c("Carpathians"=brewer.pal(12,"Paired")[9], 
           "Kirov"=brewer.pal(12,"Paired")[1],  
           "Latvia"=brewer.pal(12,"Paired")[3],  
           "Norway"=brewer.pal(12,"Paired")[2],  
           "Bialowieza"=brewer.pal(12,"Paired")[4],  
           "Mongolia"=brewer.pal(12,"Paired")[12], 
           "Töv"=brewer.pal(12,"Paired")[12], 
          "Khentii"=brewer.pal(12,"Paired")[7], 
          "Ömnögovi"=brewer.pal(12,"Paired")[11], 
              "Tuva"=brewer.pal(12,"Paired")[8],  
           "Urals"=brewer.pal(11,"BrBG")[9],  
           "Vladivostok"=brewer.pal(12,"Paired")[5], 
           "Yakutia"=brewer.pal(12,"Paired")[6], 
           "Balkans"=brewer.pal(12,"Paired")[10])  

 ggplot (diversity, aes(x=Populations, y= pairwise_ave, fill=Populations)) + 
   geom_point(color="dimgrey", size=5, shape=22) + 
   scale_fill_manual(values = cols) + 
  xlab ("Populations") + ylab ("Nucleotide diversity") +  theme_classic() + 
theme (legend.text=element_text(size=18),legend.title = element_text(size=20), axis.title = element_text(size = 20), axis.text = element_text(size=18), title=element_text(size=25), axis.text.x = element_blank() ) 

ggsave(paste(wd,"pi_per_pop.pdf", sep=""), device = pdf,  width = 20, height = 10, units = "cm") 

ggplot (diversity, aes(x=Populations, y=watterson_ave, fill=Populations)) + 
   geom_point(color="dimgrey", size=5, shape=22) + 
   scale_fill_manual(values = cols) + 
  xlab ("Populations") + ylab ("Watterson") +  theme_classic() + 
theme (legend.text=element_text(size=18),legend.title = element_text(size=20), axis.title = element_text(size = 20), axis.text = element_text(size=18), title=element_text(size=25), axis.text.x = element_blank() ) 

ggsave(paste(wd,"watterson_per_pop.pdf", sep=""), device = pdf,  width = 20, height = 10, units = "cm") 

```



<!-- ## Ploting heterozigosity --> No tiene sentido!! -->


<!-- ```{r} -->

<!-- library(dplyr) -->

<!-- wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/" -->
<!-- heterozigosity <- read.csv(paste(wd, "heterozigosity_per_pop.csv", sep =""), sep =" ", dec=".", header=F) %>%  -->
<!-- mutate (., Populations =  ifelse (V1 == "c_ll_po_n008", "Bialowieza", -->
<!--                           ifelse (V1 == "c_ll_ur_n006", "Urals", -->
<!--                           ifelse (V1 == "c_ll_ki_n013", "Kirov", -->
<!--                           ifelse (V1 == "c_ll_la_n006", "Latvia", -->
<!--                           ifelse (V1 == "c_ll_no_n008", "Norway", -->
<!--                           ifelse (V1 == "x_ll_ba_n003", "Balkans" , -->
<!--                           ifelse (V1 == "c_ll_cr_n006","Carpathians", -->
<!--                           ifelse (V1 == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", -->
<!--                           ifelse (V1 == "c_ll_tu_n006", "Tuva", -->
<!--                           ifelse (V1 == "c_ll_vl_n008", "Vladivostok",  -->
<!--                           ifelse (V1 == "c_ll_ya_n008", "Yakutia", NA)))))))))))) %>%  -->
<!--   filter(Populations!="Balkans") -->

<!-- heterozigosity$Populations <- factor (heterozigosity$Populations, levels=c("Norway", "Balkans", "Carpathians", "Bialowieza", "Latvia", "Kirov", "Urals", "Tuva", "Mongolia",  "Yakutia", "Vladivostok")) -->


<!-- cols <- c("Carpathians"=brewer.pal(12,"Paired")[9],  -->
<!--           "Kirov"=brewer.pal(12,"Paired")[1],  -->
<!--           "Latvia"=brewer.pal(12,"Paired")[3],  -->
<!--           "Norway"=brewer.pal(12,"Paired")[2],  -->
<!--           "Bialowieza"=brewer.pal(12,"Paired")[4],  -->
<!--           "Mongolia"=brewer.pal(12,"Paired")[12],   -->
<!--           "Tuva"=brewer.pal(12,"Paired")[8],  -->
<!--           "Urals"=brewer.pal(11,"BrBG")[9],  -->
<!--           "Vladivostok"=brewer.pal(12,"Paired")[5],  -->
<!--           "Yakutia"=brewer.pal(12,"Paired")[6], -->
<!--           "Balkans"=brewer.pal(12,"Paired")[10])  -->

<!-- ggplot (heterozigosity, aes(x=Populations, y= V2, fill=Populations)) + -->
<!--   geom_point(color="dimgrey", size=5, shape=22) + -->
<!--   scale_fill_manual("Población",labels = c("Noruega","Carpatos", "Bialowieza","Letonia","Kirov","Urales","Tuva", "Mongolia","Yakutia" ,"Vladivostok"), values = cols) + -->
<!--   xlab ("Populations") + ylab ("Heterozigosidad promedio") +  theme_classic() + -->
<!-- theme (legend.text=element_text(size=18),legend.title = element_text(size=20), axis.title = element_text(size = 20), axis.text = element_text(size=18), title=element_text(size=25), axis.text.x = element_blank() ) -->

<!-- ggsave(paste(wd,"heterozigosity_per_pop.pdf", sep=""), device = pdf,  width = 20, height = 10, units = "cm") -->

<!-- ``` -->






## Moving to datos!!!!

```{bash}

mkdir /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs

scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.printed.stats /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.transformedThetas /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.unfolded-lr.postprob.thetas.gz /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.unfolded-lr.postprob.thetas.idx /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.unfolded-lr.saf.gz /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.unfolded-lr.saf.idx /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs
scp /home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs/*.unfolded-lr.saf.pos.gz /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs


# Cuando esté pasado --> borrarlos! 

```


# Window analysis

Para tener una medida de dispersión de la diversidad vamos a hacer ventanas, para ello vamos a trabajar en datos que es donde están los archivos brutos. 

```{bash}

cd /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs/

ls -ltr *.postprob.thetas.idx | sort | uniq

# c_ll_no_n008.unfolded-lr.postprob.thetas.idx --> Done
# c_ll_po_n008.unfolded-lr.postprob.thetas.idx --> Done
# c_ll_ki_n013.unfolded-lr.postprob.thetas.idx --> Done
# c_ll_to-c_ll_ka-c_ll_og_n008.unfolded-lr.postprob.thetas.idx --> Done
# c_ll_ya_n008.unfolded-lr.postprob.thetas.idx --> Lanzado
# c_ll_vl_n008.unfolded-lr.postprob.thetas.idx --> Lanzado
# c_ll_la_n006.unfolded-lr.postprob.thetas.idx --> Lanzado
# c_ll_cr_n006.unfolded-lr.postprob.thetas.idx --> Lanzado
# c_ll_tu_n006.unfolded-lr.postprob.thetas.idx --> Lanzado
# x_ll_ba_n003.unfolded-lr.postprob.thetas.idx --> No lanzado para todo el intergénico. 
# c_ll_ur_n006.unfolded-lr.postprob.thetas.idx --> Lanzado


#To launch one by one

POP="c_ll_ur_n006"  # <--CHANGE POP HERE
#screen -S "$POP"_sfs
screen -S "$POP"_window_analysis
POP="c_ll_ur_n006"  # <--CHANGE POP HERE
#script "$POP"_sfs.log
script "$POP"_window_analysis.log
POP="c_ll_ur_n006"  # <--CHANGE POP HERE

NGSTOOLS="/opt/angsd/angsd/misc"

THREADS=10                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!

WINDOWSIZE=50000
WINDOWSTEP=50000 


echo "-------$POP---------- Window analysis -----------------------------------------"

$NGSTOOLS/thetaStat do_stat $POP.unfolded-lr.postprob.thetas.idx -win $WINDOWSIZE -step $WINDOWSTEP  -outnames $POP.unfolded-lr.postprob.thetasWindow_$WINDOWSIZE.$WINDOWSTEP.gz


```

Now I cat window files with the proper name

```{bash}
# Primero defino para cada archivos que ventanas son x y que ventanas no. 

# El archivo BED con las posiciones del X está en:

/home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs_intergenic_Xchr/Xchr.intergenic.bed



# Primero voy a comprobar que todos los archivos tienen el mismo número de ventanas.

wc -l *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41432 c_ll_cr_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41428 c_ll_ki_n013.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41427 c_ll_la_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41422 c_ll_no_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41431 c_ll_po_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41431 c_ll_to-c_ll_ka-c_ll_og_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41430 c_ll_tu_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41433 c_ll_ur_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41430 c_ll_vl_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
#   41428 c_ll_ya_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG

# No las tienen. Así que tengo que buscar una manera de numerar a las ventanas del 1 a x que tenga en cuenta todas las ventanas posibles de manera que si falta una ventana se salte ese número.

# La primera ventana nunca está.

# Para ello voy primero a imprimir en un mismo archivo totdas las ventanas distitnas que puedo tener.

# Primero cuantas ventanas totales tengo.

awk  -v OFS='\t' 'FNR==1 && NR!=1{next;}{print $2"_"$1}' *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG | cut -d"(" -f 1,4 | sed 's/(//g' | sed 's/)//g' | sed 's/,/_/g' |  sort | uniq | wc -l
# 41440

# Ahora les añado un número sequencial.

awk 'FNR==1 && NR!=1{next;}{print $2,$1}' *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG | cut -d"(" -f 1,4 | sed 's/(//g' | sed 's/)//g' | sed 's/,/ /g' | LANG=en_EN sort -k1,1 -k2,2n | uniq | sed 's/ /_/g' | awk  -v OFS='\t' 'FNR==1 && NR!=1{next;}{print $0, NR-1}' | sed 's/#//g' | sed 's/ /_/g' | awk 'FNR==1{$2="WinNumber"};1'  > list_of_window_identity_and_number.tsv


# Ahora vamos a ordenar las columnas del archivo de ventanas para que me sean útiles. 

for i in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG
do
echo $i
cut -d"(" -f 4  $i | sed 's/)//g' | sed 's/,/_/g' | awk  -v OFS='\t' 'FNR==1 && NR!=1{next;}{print $2"_"$1,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' > ${i/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif}
done


# Ahora tengo que hacer un join para nombrar cada ventana en cada archivo.  

for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif
do
echo $file
LANG=en_EN join -1 1 -2 1 <(LANG=en_EN sort -k1,1 $file)  <(LANG=en_EN sort -k1,1 list_of_window_identity_and_number.tsv) | LANG=en_EN  sort -k14,14n | sed $'s/_/\t/g' | sed $'s/ /\t/g'  > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2}
done


# Ahora ya tengo el numero de ventana, así que aunuq pierda las posiciones al hacer el merge lo voy a poder recuperar luego. 

############################################################################################################

############################################################################################################

##################X-CHR##################


# Primero con X.

FILE_TO_MERGE="/home/mlucena/ANGSD_analysis/intergenic_analysis/intergenic_sfs_intergenic_Xchr/Xchr.intergenic.bed" 


for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2
do
echo $file
bedtools intersect -a <(tail -n +2  $file) -b $FILE_TO_MERGE > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_X_chr}
done

# Ahora calculo el numero de bases por ventana del X
# Primero creo la resta y luego ya sumo la resta:

for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_X_chr
do
echo $file
awk  -v OFS='\t' '{print $0, $3-$2}' $file | \
awk -v OFS='\t' '{ count[$16]++; sum[$16]+=$17}END {
    for(number in count) {print number, sum[number]} }' | LANG=en_EN sort -k1,1n  > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_X_chr/number_of_window_X_chr_positions} 
done

############################################################################################################

############################################################################################################

##################AUTOSOMIC##################


# Ahora con autosomico. He creado este archivo su ruta está en remove_x_y_chromosome. 

FILE_TO_MERGE="/GRUPOS/grupolince/Lynx_AUTOSOMALchr/AUTOSOMAL.intergenic.bed" 

for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2
do
echo $file
bedtools intersect -a <(tail -n +2  $file) -b $FILE_TO_MERGE > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_AUTOSOME_chr}
done

# Ahora calculo el numero de bases por ventana del AUTOSOMICO
# Primero creo la resta y luego ya sumo la resta:

for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_AUTOSOME_chr
do
echo $file
awk  -v OFS='\t' '{print $0, $3-$2}' $file | \
awk -v OFS='\t' '{ count[$16]++; sum[$16]+=$17}END {
    for(number in count) {print number, sum[number]} }' | LANG=en_EN sort -k1,1n  > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_AUTOSOME_chr/number_of_window_AUTOSOMIC_chr_positions} 
done

############################################################################################################

############################################################################################################

# Ahora concateno el numero de posiciones X y autosomicas.

for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2
do
echo $file
# the a1 a2 if for doing a full join. -e '0' only works with -o
LANG=en_EN join -a1 -a2 -1 16 -2 1 -e '0' -o '1.1,1.2,1.3,1.4,1.5,1.6,1.7,1.8,1.9,1.10,1.11,1.12,1.13,1.14,1.15,0,2.2' <(tail -n +2 $file | LANG=en_EN sort -k16,16 ) <(LANG=en_EN sort -k1,1 ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2/number_of_window_AUTOSOMIC_chr_positions}) | \
LANG=en_EN join  -a1 -a2 -1 16 -2 1 -e '0' -o '1.1,1.2,1.3,1.4,1.5,1.6,1.7,1.8,1.9,1.10,1.11,1.12,1.13,1.14,1.15,0,1.17,2.2' - <(LANG=en_EN sort -k1,1 ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2/number_of_window_X_chr_positions}) | cat <( head -n1 $file ) - | awk 'FNR==1{$17="AUTOSOMICpositions"};1' | awk 'FNR==1{$18="Xpositions"};1' |  sed $'s/ /\t/g' | LANG=en_EN sort -k16,16n > ${file/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2/unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions}
done

# Borro todos los archivos intermedios.

rm list_of_window_identity_and_number.tsv 
rm *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif
rm *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_modif2
rm *number_of_window_AUTOSOMIC_chr_positions
rm *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_AUTOSOME_chr
rm *number_of_window_X_chr_positions
rm *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG_synteny_X_chr

# Ahora paso a concatenar los archivos para poder importarlos en R y hacer todos los calculos necesarios.

awk 'FNR==1 && NR!=1{next;}{print $0, FILENAME}' *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions | awk 'FNR==1{$19="population"};1' | sed 's/.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions//g' | sed $'s/ /\t/g'  > c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions 



scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs/c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/analysis_diversity_window

```

## R window analysis

Más abajo está el análisis definitivo!



```{r}

library(dplyr)
library(ggplot2)
library (tidyr)

wd <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/analysis_diversity_window"

index="tP"


window_diversity <- read.table("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/analysis_diversity_window/c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions", sep="\t", dec=".", header=T, na.strings = "-nan") %>% mutate (XvsTOTAL= Xpositions/(Xpositions+AUTOSOMICpositions))

window_divergence <- read.table("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_divergence_window/c_ll.divergence_with_rufus", header =T)

ggplot(data = window_diversity, aes(window_diversity$XvsTOTAL)) +
  geom_histogram() #+
# scale_y_continuous(limits = c(0, 200))

# Podemos ver que la mayoría de ventanas son o bien autosomicas o bien del X, sin embargo si añadimos scale_y_continuous(limits = c(0, 200)), vemos que sí hay algunas que tiene valores intermedios. En mi caso y dado que tenemos suficientes, voy a filtrar esas ventanas y me quedo con las que son del x y autosómicas 100%.

window_diversity_filtered <- window_diversity %>%  filter(XvsTOTAL=="1" | XvsTOTAL=="0" ) %>% mutate(window_type=ifelse(
  XvsTOTAL==1 , "Xchr", "Autosomic"))


# The number of rows are 325010, and before we had 414282, so how many of those are na:

sum(is.na(window_diversity$XvsTOTAL))
# 88262

# The total is 413272; so, 414282-413272=1010. Only 1010 windows are between 0 and 1 and has a value! We found the gap!!!!! :)

window_diversity_filtered_autosomic <- window_diversity_filtered %>%  filter(XvsTOTAL=="0")

ggplot(data=window_diversity_filtered_autosomic, aes_string(x="population", y=index))+
 geom_violin() +
 theme(axis.text.x = element_text(angle = 90, hjust = 1))


# Este grafico es interesante. Ahora computo un boxplot sobre estos datos.

ggplot(data=window_diversity_filtered_autosomic, aes_string(x="population", y=index))+
 geom_boxplot(outlier.alpha = 0.5)  +
 theme(axis.text.x = element_text(angle = 90, hjust = 1))
         
# By pop, average and standard deviation

summary_diversity <- window_diversity_filtered_autosomic %>% group_by(population) %>% summarise_at(vars(tW,tP),funs(mean, sd))


ggplot (data = summary_diversity, aes_string(x="population", y=paste0(index,"_mean")))+
  geom_point() +
  geom_errorbar(aes_string(ymin=paste(index,"_mean - ",index,"_sd", sep =""),ymax=paste(index,"_mean + ",index,"_sd", sep ="")), colour="black", width=.1) 



# Ahora pruebo la misma funcion eliminando outliers.

remove_outliers <- function(x, na.rm = TRUE, ...) {
  qnt <- quantile(x, probs=c(.25, .75), na.rm = na.rm, ...)
  H <- 1.5 * IQR(x, na.rm = na.rm)
  y <- x
  y[x < (qnt[1] - H)] <- NA
  y[x > (qnt[2] + H)] <- NA
  y
}

window_diversity_filtered_autosomic$tW_noOutliers <- remove_outliers(window_diversity_filtered_autosomic$tW)
window_diversity_filtered_autosomic$tP_noOutliers <- remove_outliers(window_diversity_filtered_autosomic$tP)

summary_diversity_no_outliers <- window_diversity_filtered_autosomic %>% group_by(population) %>% summarise_at(vars(tW_noOutliers,tP_noOutliers),funs(mean(.,na.rm=TRUE), sd(.,na.rm=TRUE)))


ggplot (data = summary_diversity_no_outliers, aes_string(x="population", y=paste0(index,"_noOutliers_mean")))+
  geom_point() +
  geom_errorbar(aes_string(ymin=paste(index,"_noOutliers_mean - ",index,"_noOutliers_sd", sep =""),ymax=paste(index,"_noOutliers_mean + ",index,"_noOutliers_sd", sep ="")), colour="black", width=.1) 


summary_diversity_ratio_autosome_XChr <- window_diversity_filtered %>% group_by(population, window_type) %>% summarise_at(vars(tW,tP),funs(mean, sd)) %>% select (population, window_type, paste0(index,"_mean")) %>% spread(., window_type, paste0(index,"_mean")) %>% mutate (ratio=Xchr/Autosomic)


ggplot (data = summary_diversity_no_outliers, aes_string(x="population", y=paste0(index,"_noOutliers_mean")))+
  geom_point() +
  geom_errorbar(aes_string(ymin=paste(index,"_noOutliers_mean - ",index,"_noOutliers_sd", sep =""),ymax=paste(index,"_noOutliers_mean + ",index,"_noOutliers_sd", sep ="")), colour="black", width=.1) 

# u_f^2  = ((u_a^2 / a^2) + (u_b^2 / b^2)) * f^2
# u_f=sqrt(((u_a^2 / a^2) + (u_b^2 / b^2)) * f^2)

# Del artículo esta es la versión naif: CVratio~ sqrt(CVx^2 + CVy^2)


```

Pregunta en research gate:

You have a ratio between 2 variables so the propagation of uncertainties for ratio applies. If I call "a" your first variable and "b" the second one, then the uncertainty of their ratio (I call it here "f = a/b") can be obtained via simple propagation as follows: (u_f^2 / f^2) = (u_a^2 / a^2) + (u_b^2 / b^2) where u_f is the uncertainty of the ratio f=a/b, u_a is the uncertainty of the variable a and u_b is the uncertainty of variable b. In your case variables a and b are arithmetic means (I guess) so the uncertainty of arithmetic means (u_a and u_b in our examples) can be again calculated using the propagation of uncertainties. This is a classical formula in propagation of uncertainties and can be further simplified if the uncertainties of each single measurement are the same.

Me lleva al artículo en cuestión:
Clin Biochem. 2007 Jun;40(9-10):728-34. Epub 2007 Mar 12.
Error propagation in calculated ratios.
Holmes DT1, Buhr KA.

Por otro lado,

https://stats.stackexchange.com/questions/161042/ratio-of-standard-deviation-to-mean:

My interval is like this: µ ± σ, where σ is the standard deviation and µ is the mean. 
I want to know if the ratio σ/µ has any interpretation:

That ratio is called the Coefficient of Variation(CV) and it shows the magnitude of variation in relation to the population mean. It is used to describe the variation within the data without depending on the measurement unit of the data, so you can easily compare the dispersion across different distributions of data 


https://stats.stackexchange.com/questions/204264/interpreting-standard-deviation-ratio-is-std-deviation-ratio-equal-to-coefficie




### First I will create a file that contain only windows in common in all pops.


```{bash}

cd /home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs


for file in *unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions
do
echo $file
# Dia 29/06/2018 me he dado cuenta que las ventanas con cero sitios a veces aparecen, pero estas no nos interesan, así que de esta manera, con la frase que acabo de añadir las filtraría y ahora sí haría las ventanas en común. 
awk '$15!=0 {print $0}' $file |\ 
awk '{print $1"_"$2"_"$3, $0}'  > ${file/mic_positions/mic_positions_borrar}
done


LANG=en_EN join -t'  ' -11 -21 <(LANG=en_EN sort -k 1,1 c_ll_cr_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) <(LANG=en_EN sort -k 1,1 c_ll_ki_n013.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <(LANG=en_EN sort -k 1,1 c_ll_la_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_no_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_po_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_to-c_ll_ka-c_ll_og_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_tu_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_ur_n006.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_vl_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) | LANG=en_EN join  -t'  ' -11 -21 - <( LANG=en_EN sort -k 1,1 c_ll_ya_n008.unfolded-lr.postprob.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_borrar) > c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_only_window_in_common

# 29/06/2018 antes tenía 41419 y ahora tengo 32773. Sí esta filtrando


awk '{print $1}' c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions_only_window_in_common > windows_in_common_list.txt 

rm *borrar

# El que tiene todo ocupa 414283 lineas pero esto se debe a que están concatenadas una tras otra. 


# Salgo del servidor:

scp mlucena@genomics-b.ebd.csic.es://home/mlucena/datos/intermediate_ANGSD_files/intergenic_analysis/intergenic_sfs/windows_in_common_list.txt /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/analysis_diversity_window

```

# R final
## Diversity vs divergence

```{r}

library(dplyr)
library(ggplot2)
library (tidyr)
library(boot)
library(broom)
library(RColorBrewer)

wd_diversity <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_sfs/analysis_diversity_window/"
# index="tP_per_site" --> Esto ya no me hace falta. 

# Bajasmos los archivos de diversidad y seleccionamos solo aquellas ventanas que estén en común para todas las poblaciones. 

diversity <- read.table(paste (wd_diversity, "c_ll_no_n008-c_ll_po_n008-c_ll_ki_n013-c_ll_to-c_ll_ka-c_ll_og_n008-c_ll_ya_n008-c_ll_vl_n008-c_ll_la_n006-c_ll_cr_n006-c_ll_tu_n006-c_ll_ur_n006.thetasWindow_50000.50000.gz.pestPG.with_x_autosomic_positions", sep=""), sep="\t", dec=".", header=T, na.strings = "-nan")  %>% 
mutate (XvsTOTAL= Xpositions/(Xpositions+AUTOSOMICpositions)) %>% 
mutate (windows_identity = paste(Chr,"_",WinStart,"_", WinStop, sep ="")) %>% 
filter(nSites>10000) %>% 
filter(XvsTOTAL=="1" | XvsTOTAL=="0" ) %>% 
mutate(window_type=ifelse(XvsTOTAL==1 , "Xchr", "Autosomic")) %>% # Lo que hago es quedarme con las posiciones que son solo X o sólo autosomicas.
mutate(tP_per_site=tP/nSites) %>% 
mutate(tW_per_site=tW/nSites) %>% 
mutate(tajima_per_site=Tajima/nSites)
 

## En esta parte filtraba por la ventanas en común en bash, por lo que no tenía en cuenta las que se filtran a posteriori y las que no. 
# Ahora voy a filtrar sobre las que ya están filtradas para numero de sitios, por ejemplo. 

# Esto me convierte el dataframe de una variable en un vector sobre el que filtrar. 
#windows_in_common <- read.table(paste(wd_diversity, "windows_in_common_list.txt", sep =""), header = T)  [[1]]

#diversity_only_in_common_bash <- filter(diversity, windows_identity %in% windows_in_common) 
# Tiene 327720 lineas, lo que encaja puesto que anteriormente tenía: 32772 (*10 poblaciones)


#hist(diversity_only_in_common$nSites)


# NUEVO FILTRO PARA VENTANAS EN COMÚN

## Ahora he filtrado por las ventanas en común que tenían al principio, pero no he filtrado por las ventanas comunes que quedan despues de aplicar todos los filtros, por tanto tengo que hacer un filtro por estas vetanas en común:

unique(diversity$population)
diversity_c_ll_cr_n006 <- diversity %>%  filter (population=="c_ll_cr_n006") %>% select(windows_identity) %>% unique()
diversity_c_ll_ki_n013 <- diversity %>%  filter (population=="c_ll_ki_n013") %>% select(windows_identity) %>% unique()
diversity_c_ll_la_n006 <- diversity %>%  filter (population=="c_ll_la_n006") %>% select(windows_identity) %>% unique()
diversity_c_ll_no_n008 <- diversity %>%  filter (population=="c_ll_no_n008") %>% select(windows_identity) %>% unique()
diversity_c_ll_po_n008 <- diversity %>%  filter (population=="c_ll_po_n008") %>% select(windows_identity) %>% unique()
diversity_c_ll_to_c_ll_ka_c_ll_og_n008   <- diversity %>%  filter (population=="c_ll_to-c_ll_ka-c_ll_og_n008") %>% select(windows_identity) %>% unique()
diversity_c_ll_tu_n006 <- diversity %>%  filter (population=="c_ll_tu_n006") %>% select(windows_identity) %>% unique()
diversity_c_ll_ur_n006 <- diversity %>%  filter (population=="c_ll_ur_n006") %>% select(windows_identity) %>% unique()
diversity_c_ll_vl_n008 <- diversity %>%  filter (population=="c_ll_vl_n008") %>% select(windows_identity) %>% unique()
diversity_c_ll_ya_n008 <- diversity %>%  filter (population=="c_ll_ya_n008") %>% select(windows_identity) %>% unique()


windows_in_common_after_filtering <- inner_join(diversity_c_ll_cr_n006, diversity_c_ll_ki_n013) %>% inner_join(diversity_c_ll_la_n006) %>% inner_join(diversity_c_ll_no_n008) %>% inner_join(diversity_c_ll_po_n008) %>% inner_join(diversity_c_ll_to_c_ll_ka_c_ll_og_n008) %>% inner_join(diversity_c_ll_tu_n006) %>% inner_join(diversity_c_ll_ur_n006) %>% inner_join(diversity_c_ll_vl_n008) %>% inner_join(diversity_c_ll_ya_n008) %>%  .[[1]]

rm (diversity_c_ll_cr_n006)
rm (diversity_c_ll_ki_n013)
rm (diversity_c_ll_la_n006)
rm (diversity_c_ll_no_n008)
rm (diversity_c_ll_po_n008)
rm (diversity_c_ll_to_c_ll_ka_c_ll_og_n008)
rm (diversity_c_ll_tu_n006)
rm (diversity_c_ll_ur_n006)
rm (diversity_c_ll_vl_n008)
rm (diversity_c_ll_ya_n008)

diversity_only_in_common <- filter(diversity, windows_identity %in% windows_in_common_after_filtering) 

rm (diversity)

#########


# Corrección de la diversidada para las ventanas que son X, en el caso de watterson (mirar oja de calculo con la correción de watterson en owncloud).

diversity_only_in_common <- diversity_only_in_common %>% 
  mutate ( watterson_ave_corrected = 
                           ifelse (population == "c_ll_po_n008" & window_type=="Xchr", tW_per_site*1.12,
                           ifelse (population == "c_ll_ur_n006" & window_type=="Xchr", tW_per_site*1.14, 
                           ifelse (population == "c_ll_ki_n013" & window_type=="Xchr", tW_per_site*1.09, 
                           ifelse (population == "c_ll_la_n006" & window_type=="Xchr", tW_per_site*1.10, 
                           ifelse (population == "c_ll_no_n008" & window_type=="Xchr", tW_per_site*1.15, 
                           ifelse (population == "c_ll_cr_n006" & window_type=="Xchr", tW_per_site*1.06, 
                           ifelse (population == "c_ll_to-c_ll_ka-c_ll_og_n008" & window_type=="Xchr", tW_per_site*1.12, 
                           ifelse (population == "c_ll_tu_n006" & window_type=="Xchr", tW_per_site*1.06, 
                           ifelse (population == "c_ll_vl_n008" & window_type=="Xchr", tW_per_site*1.12,  
                           ifelse (population == "c_ll_ya_n008" & window_type=="Xchr", tW_per_site*1.20, tW_per_site))))))))))) 


diversity_only_in_common$tW_per_site <- diversity_only_in_common$watterson_ave_corrected

  ####

# Ahora bajamos divergencia

wd_divergence <- "/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/intergenic_analysis/analysis_divergence_window/"

divergence <- read.table(paste (wd_divergence,"c_ll.divergence_with_rufus", sep =""), sep=" ", dec=".", header=T) %>% mutate (windows_identity = paste(scaffold,"_",winstart,"_", winend, sep =""))

####

## Hacemos un join.
# Aquí tenemos la divergencia y la diversidad.

diversity_divergence <- inner_join(diversity_only_in_common,divergence, by="windows_identity" )  %>% 
  mutate(tW_per_site_corrected=tW_per_site/divergence) %>% mutate(tP_per_site_corrected=tP_per_site/divergence)
  
#index_corrected=paste0(index,"_corrected")

rm (diversity_only_in_common)
rm (divergence)



# # Autosomico:
# 
# diversity_divergence_autosomic <- diversity_divergence %>%  filter(XvsTOTAL=="0")
# 
# ## Some summary results:
# # By pop, average and standard deviation
# 
# summary_diversity_autosomic <- diversity_divergence_autosomic %>% group_by(population) %>% summarise_at(vars(tW_per_site_corrected,tP_per_site_corrected),funs(mean, sd, n()))
# 
# 
# ## Violin plot:
# 
# ggplot(data=diversity_divergence_autosomic, aes_string(x="population", y=index_corrected))+
#  geom_violin() +
#  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# 
# # Este grafico es interesante. Ahora computo un boxplot sobre estos datos.
# 
# ggplot(data=diversity_divergence_autosomic, aes_string(x="population", y=index_corrected))+
#  geom_boxplot( outlier.shape = NA) +#outlier.alpha = 0.5) +
#  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
#  coord_cartesian(ylim = quantile(diversity_divergence_autosomic[,which(colnames(diversity_divergence_autosomic)==index_corrected)], c(0.1, 0.9))) +
#  scale_y_continuous(limits = quantile(diversity_divergence_autosomic[,which(colnames(diversity_divergence_autosomic)==index_corrected)], c(0.1, 0.9))) +
#  stat_summary(fun.y = mean, geom="point", colour="darkred", size=3) +
#  stat_summary(fun.y = mean, geom="text", vjust=-0.7, aes( label=round(..y.., digits=5)))
# 
# 
# # X
# 
# diversity_divergence_X <- diversity_divergence %>%  filter(XvsTOTAL=="1")
# 
# ## Some summary results:
# # By pop, average and standard deviation
# 
# summary_diversity_X <- diversity_divergence_X %>% group_by(population) %>% summarise_at(vars(tW_per_site_corrected,tP_per_site_corrected),funs(mean, sd, n()))
# 
# 
# ## Violin plot:
# 
# ggplot(data=diversity_divergence_X, aes_string(x="population", y=index_corrected))+
#  geom_violin() +
#  theme(axis.text.x = element_text(angle = 90, hjust = 1))
# 
# # Este grafico es interesante. Ahora computo un boxplot sobre estos datos.
# 
# ggplot(data=diversity_divergence_X, aes_string(x="population", y=index_corrected))+
#  geom_boxplot( outlier.shape = NA) +#outlier.alpha = 0.5) +
#  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
#  coord_cartesian(ylim = quantile(diversity_divergence_X[,which(colnames(diversity_divergence_X)==index_corrected)], c(0.1, 0.9))) +
#  scale_y_continuous(limits = quantile(diversity_divergence_X[,which(colnames(diversity_divergence_X)==index_corrected)], c(0.1, 0.9))) +
#  stat_summary(fun.y = mean, geom="point", colour="darkred", size=3) +
#  stat_summary(fun.y = mean, geom="text", vjust=-0.7, aes( label=round(..y.., digits=5)))
# 



# ----------------------------------------------------------------------------------------------------------------------------
# 29/05/2018

# Hasta aquí he trabajado con los datos sin bootstrap. Tiene sentido que las ventanas tengan una dispersión muy grande, puesto que hay ventanas con un valor de diversidad muy alto y otras con valores muy bajos. Por tanto, lo que vamos a hacer es un bootstrap con reemplazamiento y vamos a calcular sobre este bootstrap nuestra dispersión. Es decir, calculo 1000 veces la media cogiendo al azar tantas ventanas como hay (permitiendo coger dos veces la misma = bootstrap), y despues calculo la dispersión de estas medias. Igual voy a hacer con la dispersión del ratio. 

# https://uoftcoders.github.io/studyGroup/lessons/r/resampling/lesson/


# Create a function to take a resample of the values, and then calculate the mean

boot_mean <- function(original_vector, resample_vector) {
    mean(original_vector[resample_vector])
}

# R is number of replications

ITERATION=10000

# Mi dataframe de diversity_divergence ya tiene valores que o son 0 o 1 para autosomico/X. Así que la puedo usar para filtrar. 

summary_boot_mean_autosomic_tW <- data.frame(mean=numeric())
summary_boot_mean_X_tW <- data.frame(mean=numeric())
summary_boot_ratio_X_autosomic_tW <- data.frame(mean=numeric())

summary_boot_mean_autosomic_tP <- data.frame(mean=numeric())
summary_boot_mean_X_tP <- data.frame(mean=numeric())
summary_boot_ratio_X_autosomic_tP <- data.frame(mean=numeric())




for(pop in unique(diversity_divergence$population)) {
  
  print (pop)
  
  # Hago dos tablas temporales, autosomic y X 
  
  tabla_temp_autosomic <- diversity_divergence %>% filter(window_type=="Autosomic") %>% filter (population==pop)
  tabla_temp_X <- diversity_divergence %>% filter(XvsTOTAL=="1") %>% filter (population==pop)
  
  # Calculo el bootstrap de tW para la media y el ratio y lo añado a una tabla global 
    
  boot_mean_autosomic_tW_per_pop <- boot( data = tabla_temp_autosomic$tW_per_site, boot_mean, R = ITERATION)
  summary_boot_mean_autosomic_tW_per_pop <- cbind(tidy(boot_mean_autosomic_tW_per_pop), pop) %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tW")
  
  summary_boot_mean_autosomic_tW <- rbind(summary_boot_mean_autosomic_tW, summary_boot_mean_autosomic_tW_per_pop)
  
  boot_mean_X_tW_per_pop <- boot( data = tabla_temp_X$tW_per_site, boot_mean, R = ITERATION)
  summary_boot_mean_X_tW_per_pop <- cbind(tidy(boot_mean_X_tW_per_pop), pop) %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tW")
  
  summary_boot_mean_X_tW <- rbind(summary_boot_mean_X_tW, summary_boot_mean_X_tW_per_pop)
  
  # Para el ratio, hago el bootstrap de los corregidos!
  
  boot_mean_autosomic_tW_corrected_per_pop <- boot( data = tabla_temp_autosomic$tW_per_site_corrected, boot_mean, R = ITERATION)
  boot_mean_X_tW_corrected_per_pop <- boot( data = tabla_temp_X$tW_per_site_corrected, boot_mean, R = ITERATION)

  boot_ratio_X_autosomic_X_tW_per_pop <- as.data.frame(cbind(boot_mean_X_tW_corrected_per_pop$t, boot_mean_autosomic_tW_corrected_per_pop$t)) %>% mutate (ratio_X_autosomic=V1/V2) %>%  select (ratio_X_autosomic)
  summary_boot_ratio_X_autosomic_tW_per_pop <- cbind(tidy (boot_ratio_X_autosomic_X_tW_per_pop), pop) %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tW")
  
  summary_boot_ratio_X_autosomic_tW <- rbind(summary_boot_ratio_X_autosomic_tW, summary_boot_ratio_X_autosomic_tW_per_pop)

  
 # Calculo el bootstrap de tP para la media y el ratio y lo añado a una tabla global 
 
  boot_mean_autosomic_tP_per_pop <- boot( data = tabla_temp_autosomic$tP_per_site, boot_mean, R = ITERATION)
  summary_boot_mean_autosomic_tP_per_pop <- cbind(tidy(boot_mean_autosomic_tP_per_pop), pop)  %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tP")
  
  summary_boot_mean_autosomic_tP <- rbind(summary_boot_mean_autosomic_tP, summary_boot_mean_autosomic_tP_per_pop)
  
  boot_mean_X_tP_per_pop <- boot( data = tabla_temp_X$tP_per_site, boot_mean, R = ITERATION)
  summary_boot_mean_X_tP_per_pop <- cbind(tidy(boot_mean_X_tP_per_pop), pop) %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tP")
  
  summary_boot_mean_X_tP <- rbind(summary_boot_mean_X_tP, summary_boot_mean_X_tP_per_pop)
  
  
  # Hago el bootstrap de los corregidos por la divergencia. 
  
  boot_mean_autosomic_tP_corrected_per_pop <- boot( data = tabla_temp_autosomic$tP_per_site_corrected, boot_mean, R = ITERATION)
  boot_mean_X_tP_corrected_per_pop <- boot( data = tabla_temp_X$tP_per_site_corrected, boot_mean, R = ITERATION)

  
  boot_ratio_X_autosomic_X_tP_per_pop <- as.data.frame(cbind(boot_mean_X_tP_corrected_per_pop$t, boot_mean_autosomic_tP_corrected_per_pop$t)) %>% mutate (ratio_X_autosomic=V1/V2) %>%  select (ratio_X_autosomic)
  summary_boot_ratio_X_autosomic_tP_per_pop <- cbind(tidy (boot_ratio_X_autosomic_X_tP_per_pop), pop) %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="tP")
  summary_boot_ratio_X_autosomic_tP <- rbind(summary_boot_ratio_X_autosomic_tP, summary_boot_ratio_X_autosomic_tP_per_pop)

  
  
  # Remove intermediate files
  
    rm (tabla_temp_autosomic,tabla_temp_X,boot_mean_autosomic_tW_per_pop, summary_boot_mean_autosomic_tW_per_pop, boot_mean_X_tW_per_pop, summary_boot_mean_X_tW_per_pop, boot_ratio_X_autosomic_X_tW_per_pop, summary_boot_ratio_X_autosomic_tW_per_pop, boot_mean_autosomic_tP_per_pop, summary_boot_mean_autosomic_tP_per_pop, boot_mean_X_tP_per_pop, summary_boot_mean_X_tP_per_pop, boot_ratio_X_autosomic_X_tP_per_pop, summary_boot_ratio_X_autosomic_tP_per_pop)

  
}


# Para Tajima's

summary_boot_mean_autosomic_tajima <- data.frame(mean=numeric())

for(pop in unique(diversity_divergence$population)) {
  
  print (pop)
  
  # Hago dos tablas temporales, autosomic y X 
  
  tabla_temp_autosomic <- diversity_divergence %>% filter(window_type=="Autosomic") %>% filter (population==pop)
  
boot_mean_autosomic_tajima_per_pop <- boot( data = tabla_temp_autosomic$tajima_per_site, boot_mean, R = ITERATION)
  summary_boot_mean_autosomic_tajima_per_pop <- cbind(tidy(boot_mean_autosomic_tajima_per_pop), pop)  %>% mutate (., Populations =  ifelse (pop == "c_ll_po_n008", "Bialowieza",
                           ifelse (pop == "c_ll_ur_n006", "Urals", 
                           ifelse (pop == "c_ll_ki_n013", "Kirov", 
                           ifelse (pop == "c_ll_la_n006", "Latvia", 
                           ifelse (pop == "c_ll_no_n008", "Norway", 
                           ifelse (pop == "x_ll_ba_n003", "Balkans" ,
                           ifelse (pop == "c_ll_to_n002", "Töv",
                           ifelse (pop == "c_ll_ka_n004", "Khentii", 
                           ifelse (pop == "c_ll_og_n002", "Ömnögovi", 
                           ifelse (pop == "c_ll_cr_n006","Carpathians", 
                           ifelse (pop == "c_ll_to-c_ll_ka-c_ll_og_n008", "Mongolia", 
                           ifelse (pop == "c_ll_tu_n006", "Tuva", 
                           ifelse (pop == "c_ll_vl_n008", "Vladivostok",  
                           ifelse (pop == "c_ll_ya_n008", "Yakutia", NA))))))))))))))) %>% mutate(index="Tajima")
  
  summary_boot_mean_autosomic_tajima <- rbind(summary_boot_mean_autosomic_tajima, summary_boot_mean_autosomic_tajima_per_pop)
 
}
  ##


summary_boot_mean_autosomic <- rbind (summary_boot_mean_autosomic_tW, summary_boot_mean_autosomic_tP) 
rm (summary_boot_mean_autosomic_tW, summary_boot_mean_autosomic_tP)
summary_boot_mean_autosomic$Populations <- factor (summary_boot_mean_autosomic$Populations, levels=c("Norway", "Balkans", "Carpathians", "Bialowieza", "Latvia", "Kirov", "Urals", "Tuva", "Khentii", "Töv", "Ömnögovi", "Mongolia",  "Yakutia", "Vladivostok"))


summary_boot_mean_X <- rbind (summary_boot_mean_X_tW, summary_boot_mean_X_tP)
rm (summary_boot_mean_X_tW, summary_boot_mean_X_tP)
summary_boot_mean_X$Populations <- factor (summary_boot_mean_X$Populations, levels=c("Norway", "Balkans", "Carpathians", "Bialowieza", "Latvia", "Kirov", "Urals", "Tuva", "Khentii", "Töv", "Ömnögovi", "Mongolia",  "Yakutia", "Vladivostok"))


summary_boot_ratio_X_autosomic <- rbind (summary_boot_ratio_X_autosomic_tW, summary_boot_ratio_X_autosomic_tP)
rm  (summary_boot_ratio_X_autosomic_tW, summary_boot_ratio_X_autosomic_tP)
summary_boot_ratio_X_autosomic$Populations <- factor (summary_boot_ratio_X_autosomic$Populations, levels=c("Norway", "Balkans", "Carpathians", "Bialowieza", "Latvia", "Kirov", "Urals", "Tuva", "Khentii", "Töv", "Ömnögovi", "Mongolia",  "Yakutia", "Vladivostok"))

write.table(summary_boot_mean_autosomic, "/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/diversity_per_pop_windows_mean_stderror_autosomic.tsv", row.names = F, quote = F)
write.table(summary_boot_ratio_X_autosomic, "/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/ratio_per_pop_windows_mean_stderror_autosomic_Xchr.tsv", row.names = F, quote = F)

```



# Plot articulo

```{r}
library (dplyr)
library (ggplot2)
library (tidyr)
library (boot)
library (broom)
library (RColorBrewer)
library (viridis)

# Read tables
summary_boot_mean_autosomic <- read.csv ("/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/diversity_per_pop_windows_mean_stderror_autosomic.tsv", sep=" ", dec=".")
summary_boot_ratio_X_autosomic <- read.csv( "/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/ratio_per_pop_windows_mean_stderror_autosomic_Xchr.tsv", sep=" ", dec=".")

# Rename factors
summary_boot_mean_autosomic$Populations <- plyr::revalue(summary_boot_mean_autosomic$Populations, c("Khentii"="Mongolia", "Töv"="Mongolia", "Ömnögovi"="Mongolia", "Bialowieza"="NE-Poland"))
summary_boot_ratio_X_autosomic$Populations <- plyr::revalue(summary_boot_ratio_X_autosomic$Populations, c("Khentii"="Mongolia", "Töv"="Mongolia", "Ömnögovi"="Mongolia", "Bialowieza"="NE-Poland"))

# Order factors
summary_boot_mean_autosomic$Populations <- factor (summary_boot_mean_autosomic$Populations, levels=c("Norway", "Balkans", "Carpathians", "NE-Poland", "Latvia", "Kirov", "Urals", "Tuva", "Mongolia",  "Yakutia", "Vladivostok"))
summary_boot_ratio_X_autosomic$Populations <- factor (summary_boot_ratio_X_autosomic$Populations, levels=c("Norway", "Balkans", "Carpathians", "NE-Poland", "Latvia", "Kirov", "Urals", "Tuva", "Mongolia",  "Yakutia", "Vladivostok"))



cols <- c("NE-Poland"=viridis_pal()(5)[3], 
          "Urals"="#0F4909", 
          "Balkans"="#A035AF", 
          "Carpathians"=brewer.pal(12,"Paired")[9], 
          "Kirov"=viridis_pal()(5)[1], 
          "Latvia"=brewer.pal(12,"Paired")[3], 
       "Norway"=viridis_pal()(5)[2], 
       "Töv"=brewer.pal(12,"Paired")[12], 
          "Tuva"=brewer.pal(12,"Paired")[8], 
          "Khentii"=brewer.pal(12,"Paired")[7], 
          "Ömnögovi"=brewer.pal(12,"Paired")[11], 
          "Vladivostok"=brewer.pal(12,"Paired")[5], 
          "Yakutia"=brewer.pal(12,"Paired")[6],
          "Sierra Morena"=brewer.pal(8, "Greys") [5],
          "Doñana"=brewer.pal(8, "Greys") [8],
          "Mongolia"=brewer.pal(12,"Paired")[12])

 

# Ploteo este estadístico. 

# Autosómico

for (INDICE in unique(summary_boot_mean_autosomic$index))
{
 ggplot (filter(summary_boot_mean_autosomic, index==INDICE), aes(y=statistic, x=Populations, fill=Populations, colour=Populations)) + 
   geom_errorbar(aes(fill=Populations, colour=Populations, ymin=statistic-std.error, ymax=statistic+std.error)) +
   geom_point(aes(fill=Populations, colour=Populations), size=1) + 
   scale_fill_manual(values = cols) + 
   scale_colour_manual(values = cols) + 
   # scale_y_continuous(labels = function(x) format(x, scientific = TRUE)) +
   ylab (paste(INDICE, sep="")) +  
   theme_minimal() +
   theme (legend.position = "none", axis.title = element_text(size = 12), axis.text = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1)) 
ggsave(paste("/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/autosomic_mean_",INDICE,"_boot2.pdf", sep=""),  width = 10, height = 10, units = "cm",  useDingbats=FALSE) 

 }


# Ratio

for (INDICE in unique(summary_boot_ratio_X_autosomic$index))
{
 ggplot (filter(summary_boot_ratio_X_autosomic, index==INDICE), aes(x=Populations, y=mean, fill=Populations, colour=Populations)) + 
   geom_point(size=2, shape=20) + 
   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
   scale_fill_manual(values = cols) + 
   scale_colour_manual(values = cols) + 
   ylab (paste(INDICE, sep="")) + 
   theme_minimal() +
   theme (legend.position = "none", axis.title = element_text(size = 12), axis.text = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1)) 
ggsave(paste("/Users/marialucenaperez/Owncloud/publico/PhD/Eurasian_lynx_phylogeography/ANGSD_analysis/analysis_sfs/ratio_X_autosomic_",INDICE,"_boot2.pdf", sep=""),   width = 10, height = 10, units = "cm",  useDingbats=FALSE) 

 }


```

# Population density vs diversity

```{r}
library(dplyr)
library(ggplot2)
library (tidyr)
library(RColorBrewer)

wd <- "/Users/marialucenaperez/Owncloud/publico/Eurasian_lynx_phylogeography/population_density/"

pop_vs_diversity <- read.table(paste (wd, "population_vs_diversity.txt", sep=""), sep="\t", dec=",", header=T, na.strings = "-nan")


ggplot(pop_vs_diversity, aes(x=sqrtkm, y=pi, label=population)) +
  geom_point( ) +
  geom_smooth(method='lm') +
  geom_text(aes(label=population),hjust=0, vjust=0)




ggplot(pop_vs_diversity, aes(x=sqrtkm, y=watterson, label=population)) +
  geom_point( ) +
  geom_smooth(method='lm') +
  geom_text(aes(label=population),hjust=0, vjust=0)

```
  
  
  