---
title: "lynx canadensis ancestry"
output: html_document
---



#Adding lynx canadensis

# PCA 5X

Tenemos dudas sobre si lynx canadensis es verdaderamente un lynx canadensis o es un lynx lynx. Por tanto, vamos a correr un PCA y un NGSAdmix para ver su ancestría. 

```{bash}
 mkdir ancestry_lc_0001
 cd ancestry_lc_0001/


nano c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081.bamlist


/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_lc_zz_0001_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0075_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0076_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0077_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0078_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0079_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0080_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0081_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_no_0082_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ba_0216_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/h_ll_ba_0214_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/h_ll_ba_0215_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0205_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0206_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0207_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0208_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0209_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_cr_0212_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0001_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0002_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0003_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0011_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0014_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0019_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0105_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_po_0106_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0044_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0047_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0048_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0052_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0053_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_la_0054_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0090_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0091_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0092_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0093_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0094_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0095_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0096_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0097_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0098_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0099_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0100_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0101_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ki_0102_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0194_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0195_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0196_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0199_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0200_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ur_0203_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0153_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0157_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0158_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0159_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0165_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_tu_0166_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_to_0190_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_to_0191_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ka_0184_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ka_0186_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ka_0188_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ka_0189_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_og_0181_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_og_0187_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0138_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0139_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0140_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0142_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0143_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0145_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0146_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_ya_0147_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0107_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0108_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0109_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0110_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0112_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0113_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0128_recal_round-1.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/c_ll_vl_0132_recal_round-1.bam

POP=c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081

ANGSD=/opt/angsd/angsd
NGSTOOLS="/opt/ngsTools"
THREADS=20                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2) 
REGIONFILE="/home/mlucena/ANGSD_analysis/depth_calculus/no_genes_Lypa_10000longest_center_final_slop20_dot.rf"




# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
# echo $maxDepth
# echo $minDepth


### Calling genotype likelihoods: 
 
$ANGSD/angsd -P $THREADS -b $POP.bamlist -ref $REF -out $POP.rf \
$FILTER1 $FILTER2 \
-rf $REGIONFILE \
-GL 1 -doMajorMinor 1 -doMaf 1 -skipTriallelic 1 \
-doGeno 32 -doPost 1


# Unzip the results (but you cannot open it since it is in binary format)

gunzip $POP.rf.geno.gz

### ngsCovar

# We are going to use ngsCovar, which estimates the covariance matrix between individuals based on genotype probabilities. 
# Then this matrix will be decomposed into principal components which will be investigated for population structure analyses. 
# Note that although ngsCovar can account for SNPs uncertanity, we find that it is faster to perform a light SNP filtering first 
# (as we did using -SNP_pval 1e-3) than using all sites.

# If you type:

# $NGSTOOLS/ngsPopGen/ngsCovar

# you will see a list of possible options. 
# For instance, we need to define how many sites we have. To retrieve such value, we can inspect the file with allele frequencies:

#less -S $POP.mafs.gz
N_SITES=`zcat $POP.rf.mafs.gz | tail -n+2 | wc -l`
echo $N_SITES 

# 1647567


# Now we can perform a PCA by estimating the covariance matrix first:

$NGSTOOLS/ngsPopGen/ngsCovar -probfile $POP.*geno -outfile $POP.covar -nind $N_IND  -nsites $N_SITES -call 0 -norm 0 &> /dev/null

#-minmaf FLOAT: ignore sites below this threshold of minor allele frequency

# with the options -call 0 meaning that we do not perform genotype calling 
# and -norm 0 that we are not normalising by allele frequency. 
# The latter may give more weight to low frequency variants which are harder to estimate.

# Look at the output file:

#less -S $POP.covar

# which represents a matrix of NxN with N individuals giving the covariance. Note that this matrix is symmetric.

# Finally, we perform an eigenvector decomposition and plot the resulting map:

# create a cluster-like file defining the labelling (population) for each sample

#Example: Rscript -e 'write.table(cbind(seq(1,60),rep(1,60),c(rep("LWK",20),rep("TSI",20),rep("PEL",20))), row.names=F, sep=" ", col.names=c("FID","IID","CLUSTER"), file="Results/ALL.clst", quote=F)'

echo "FID" > $POP.ALL1.clst.borrar
seq $N_IND >> $POP.ALL1.clst.borrar
echo "IID" > $POP.ALL2.clst.borrar
# yes "1" | head -n $N_IND >> $POP.ALL2.clst.borrar
cat $POP.bamlist | rev | cut -d "/"  -f 1 | rev | colrm 13  >> $POP.ALL2.clst.borrar
echo "CLUSTER" > $POP.ALL3.clst.borrar
cat $POP.bamlist | rev | cut -d "/"  -f 1 | rev | colrm 8  >> $POP.ALL3.clst.borrar
paste $POP.ALL1.clst.borrar $POP.ALL2.clst.borrar $POP.ALL3.clst.borrar> $POP.ALL.clst

rm $POP.*clst.borrar

# run and plot

# Rscript $NGSTOOLS/Scripts/plotPCA.R -i Results/ALL.covar -c 1-2 -a Results/ALL.clst -o Results/ALL.pca.pdf
# Rscript $NGSTOOLS/Scripts/plotPCA.R -i $POP.covar -c 1-2 -a $POP.ALL.clst -o $POP.ALL.pca.pdf #--> No corre porque no puedo instalar librerías así que lo hago en mi propio ordenador
# evince Results/ALL.pca.pdf

# where the parameter -c 1-2 specifies that we are plotting only the first and second component. On the screen, you will see a series of numbers. These are the percentage of explained variance for each component.

```

## Copying to Dropbox folder


```{bash}
scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/ancestry_lc_0001/*.ALL.clst  /Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001

scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/ancestry_lc_0001/*.covar /Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001 

```


## R representation

```{r "setup", include=FALSE}

library(methods)
library(optparse)
library(ggplot2)
library(dplyr)
library(threejs)
library(RColorBrewer)
require("knitr")

# knitr::opts_knit$set(root.dir = normalizePath("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/analysis_PCA/test_intergenic_elena"))
# getwd()
# opts_knit$set(root.dir = normalizePath("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/analysis_PCA/test_intergenic_elena"))

WORKING_DIR="/Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001/"

pop=c("c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081")


 
###########################################################################################
#for (pop in pops) 
#{

# Annotation file is in plink cluster format

#################################################################################

# Read input file
covar <- read.table(paste(WORKING_DIR,pop, ".covar", sep=''), stringsAsFact=F);

# Read annot file
annot <- read.table(paste(WORKING_DIR,pop, ".ALL.clst", sep=""), header=TRUE) 

# Parse components to analyze
# comp <- as.numeric(strsplit("component1-component2", "-", fixed=TRUE)[[1]])
# comp <- as.numeric(strsplit(opt$comp, "-", fixed=TRUE)[[1]])

comp <- c(1,2)
# comp <- c(2,3)
# comp <- c(1,3)


# Eigenvalues
eig <- eigen(covar, symm=TRUE);
eig$val <- eig$val/sum(eig$val);
cat(signif(eig$val, digits=3)*100,"\n");

# Plot
PC <- as.data.frame(eig$vectors)
colnames(PC) <- gsub("V", "PC", colnames(PC))
PC$Pop <- factor(annot$CLUSTER)
PC$Ind <- factor(annot$IID)
# PC <- PC %>% filter(Ind == "UA")



PC <- PC %>%  mutate(., 
         Subspecies = ifelse (Pop == "c_ll_po" | Pop == "c_ll_ki" | Pop == "c_ll_la" | Pop == "c_ll_no"  | Pop == "c_ll_ur", "L.l.lynx",        
                          ifelse(Pop == "c_ll_ba" | Pop == "h_ll_ba", "L.l.balcanicus" ,
                          ifelse(Pop == "c_ll_cr","L.l.carpathicus",
                          ifelse(Pop == "c_ll_to" | Pop == "c_ll_tu" | Pop == "c_ll_ka", "L.l.kozlovi", 
                          ifelse(Pop == "c_ll_og", "L.l.isabellinus", 
                          ifelse(Pop == "c_ll_vl", "L.l.stroganovi", 
                          ifelse(Pop == "c_ll_ya", "L.l.wrangeli",
                                 ifelse(Pop == "c_lc_zz", "Lynx_canadensis",
                                 NA))))))))) %>% 
  mutate (., Populations =  ifelse (Pop == "c_ll_po", "Bialowieza",
                          ifelse (Pop == "c_ll_ur", "Urals",
                          ifelse (Pop == "c_ll_ki", "Kirov",
                          ifelse (Pop == "c_ll_la", "Latvia",
                          ifelse (Pop == "c_ll_no", "Norway",
                          ifelse (Pop == "c_ll_ba" | Pop == "h_ll_ba", "Balkans" ,
                          ifelse (Pop == "c_ll_cr","Carpathians",
                          ifelse (Pop == "c_ll_to", "Töv",
                          ifelse (Pop == "c_ll_tu", "Tuva",
                          ifelse (Pop == "c_ll_ka", "Khentii", 
                          ifelse (Pop == "c_ll_og", "Ömnögovi", 
                          ifelse (Pop == "c_ll_vl", "Vladivostok", 
                          ifelse (Pop == "c_ll_ya", "Yakutia",
                          ifelse (Pop == "c_lc_zz", "Lynx_canadensis",NA))))))))))))))) %>% 
    mutate (., color =  ifelse (Populations == "Bialowieza", brewer.pal(12,"Paired")[4], 
                      ifelse (Populations == "Urals", brewer.pal(9,"RdPu")[6], 
                      ifelse (Populations == "Balkans", brewer.pal(12,"Paired")[10], 
                      ifelse (Populations == "Carpathians", brewer.pal(12,"Paired")[9], 
                      ifelse (Populations == "Kirov", brewer.pal(12,"Paired")[1], 
                      ifelse (Populations == "Latvia", brewer.pal(12,"Paired")[3], 
                      ifelse (Populations == "Norway", brewer.pal(12,"Paired")[2], 
                      ifelse (Populations == "Töv", brewer.pal(12,"Paired")[12], 
                      ifelse (Populations == "Tuva", brewer.pal(12,"Paired")[8], 
                      ifelse (Populations == "Khentii", brewer.pal(12,"Paired")[7], 
                      ifelse (Populations == "Ömnögovi", brewer.pal(12,"Paired")[11], 
                      ifelse (Populations == "Vladivostok", brewer.pal(12,"Paired")[5], 
                      ifelse (Populations == "Yakutia", brewer.pal(12,"Paired")[6],
                      ifelse (Populations == "Lynx_canadensis", brewer.pal(8, "Greys") [5], NA))))))))))))))) %>% 
  mutate (., seq = ifelse (Ind == "c_ll_vl_0112" | Ind == "c_ll_vl_0112" | Ind == "c_ll_ya_0146" | Ind == "c_ll_cr_0212" | Ind == "c_ll_ki_0090" , "MACROGEN", "CNAG"))

                           
                           
cols <- c("Bialowieza"=brewer.pal(12,"Paired")[4], 
          "Urals"=brewer.pal(9,"RdPu")[6], 
          "Balkans"=brewer.pal(12,"Paired")[10], 
          "Carpathians"=brewer.pal(12,"Paired")[9], 
          "Kirov"=brewer.pal(12,"Paired")[1], 
          "Latvia"=brewer.pal(12,"Paired")[3], 
          "Norway"=brewer.pal(12,"Paired")[2], 
          "Töv"=brewer.pal(12,"Paired")[12], 
          "Tuva"=brewer.pal(12,"Paired")[8], 
          "Khentii"=brewer.pal(12,"Paired")[7], 
          "Ömnögovi"=brewer.pal(12,"Paired")[11], 
          "Vladivostok"=brewer.pal(12,"Paired")[5], 
          "Yakutia"=brewer.pal(12,"Paired")[6],
          "Sierra Morena"=brewer.pal(8, "Greys") [5],
          "Doñana"=brewer.pal(8, "Greys") [8], 
          "Lynx_canadensis"=brewer.pal(8, "Greys") [5])

title <- paste("PC",comp[1]," (",signif(eig$val[comp[1]], digits=3)*100,"%)"," / PC",comp[2]," (",signif(eig$val[comp[2]], digits=3)*100,"%)",sep="",collapse="")

x_axis = paste("PC",comp[1],sep="")
y_axis = paste("PC",comp[2],sep="")

# ggplot() + 
#   geom_point(data=PC, aes_string(x=x_axis, y=y_axis, color="Pop_label", shape="Pop"), size = 3) + 
#   scale_shape_manual(values=1:15) + ggtitle(title)
# # ggsave(paste(WORKING_DIR,pop, "_pca.TEST.eps", sep=""),units="mm",  width=270, height=190)
# # unlink("Rplots.pdf", force=TRUE)
# ggplot() + 
#   geom_point(data=PC, aes_string(x=x_axis, y=y_axis, color="Pop", shape="Pop"), size = 3) + 
#     scale_colour_manual(values = cols)+
#   scale_shape_manual(values=1:15) + ggtitle(title)
# 
# ggplot() + 
#   geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Pop", shape="subspecies"), size = 3) + 
#   scale_colour_manual(values = cols)+
#   scale_shape_manual(values=1:7)

ggplot() + 
  geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Populations"), size = 5) + 
  scale_colour_manual(values = cols)+ ggtitle(title) + theme_classic()
 ggsave(paste(WORKING_DIR,pop, "_pca_pop_PC1_PC2.pdf", sep=""),units="mm",  width=270, height=190)

#ggplot() + 
#  geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Populations", shape="seq"), size = 5) + 
#  scale_colour_manual(values = cols)+ ggtitle(title) + theme_classic()

# ggplot() + 
#   geom_point(data=PC, aes_string(x=x_axis, y=y_axis, shape="subspecies"), size = 3) + 
#   scale_shape_manual(values=1:7)

#ggplot() + 
#  geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Subspecies"), size = 5)  +
#  ggtitle(title) + theme_classic()
#  ggsave(paste(WORKING_DIR,pop, "_pca_subspecies_PC1_PC2.pdf", sep=""),units="mm",  width=270, height=190)

# ggplot() + 
#   geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="subspecies"), size = 3)  +
#   scale_shape_manual(values=1:7)
# 
# ggsave(paste(WORKING_DIR,pop, "_pca.TEST.eps", sep=""),plot=a, units="mm",  width=270, height=190)
# 
# ggsave(paste(WORKING_DIR,pop, "_pca2.TEST.eps", sep=""),plot=b, units="mm",  width=270, height=190)
  

###########################################################################################
# 3D plot
###########################################################################################

scatterplot3js(PC$PC1, PC$PC2, PC$PC3, 
                col=PC$color,  labels=PC$Ind,
               size=0.3, grid=FALSE, axis =T)


, renderer = "canvas"
```


# NGSAdmix 5x



```{bash}

POP="c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081"  # <--CHANGE POP HERE


THREADS=15                    # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!


ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)
REGIONFILE="/home/mlucena/ANGSD_analysis/depth_calculus/no_genes_Lypa_10000longest_center_final_slop20_dot.rf"



# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
# echo $maxDepth
# echo $minDepth
# echo $SNP_PVAL

# Running GL:

$ANGSD/angsd -nThreads $THREADS -bam "$POP".bamlist -ref $REF  \
-out "$POP".genolike \
$FILTER1 \
$FILTER2 \
-rf $REGIONFILE \
-GL 1  -doGlf 2 \
-doMajorMinor 1 -doMaf 1 -skipTriallelic 1 


zcat c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081.genolike.mafs.gz | tail -n+2 | wc -l

# 1647567

```


## Running admixture

Corro 10 veces el comando para cada K. 

```{r, engine=bash, eval=FALSE}


for i in `seq 1 10`;
do
echo "running RUN" $j "k" $i
/home/mlucena/software/NGSadmix -likes "$POP".genolike.beagle.gz -K "$i" -P 20 -o "$POP"_K"$i"
done  



```




### Plotting admix results

First we need a file with our population information:

```{bash}

cat $POP*.bamlist | rev | cut -d "/"  -f 1 | cut -d "_" -f 4,5,6 | rev  > $POP.column1.borrar
cat $POP*.bamlist | rev | cut -d "/"  -f 1 | cut -d "_" -f 3,4,5,6 | rev  > $POP.column2.borrar

paste $POP.column1.borrar $POP.column2.borrar > $POP.info

rm *borrar


```

### Creating a plot

Guardo este script para lanzarlo en secuencial para cada una de las ks:

```{bash}
nano Admix_plot.R
```


scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/ancestry_lc_0001/*.qopt /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/ancestry_lc_001/

scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/ancestry_lc_0001/*info /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/ancestry_lc_001/



```{bash}
for i in `seq 1 10`;
do
echo $i
Rscript Admix_plot_fancy.R "$POP"_K"$i".qopt "$POP".info $i $POP
done
```


cd /Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/ancestry_lc_001/

nano Admix_plot_fancy.R

```{r}


library(RColorBrewer)
library(dplyr)
library(reshape2)
library(ggplot2)
library(lemon)
library(grid)
library(data.table) # 1.8.10

args <- commandArgs(TRUE)
print(args)
admix<-t(as.matrix(read.table(args[1])))
pop<-read.table(args[2],as.is=T)

admix<-t(as.matrix(read.table("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/ancestry_lc_001/c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081_K10.qopt")))


pop<-read.table("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/ancestry_lc_001/c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081.info")
  
  
names(pop) <- c("pop", "ind")
pop$pop <- as.factor (pop$pop)
pop$pop <- factor (pop$pop, levels=c("c_ll_no", "h_ll_ba", "c_ll_ba","c_ll_cr",  "c_ll_po", "c_ll_la", "c_ll_ki", "c_ll_ur", "c_ll_tu", "c_ll_og", "c_ll_ka", "c_ll_to", "c_ll_ya", "c_ll_vl", "c_lc_zz"))

admix<-admix[,order(pop[,1])]
pop <-pop[order(pop[,1]),]


pop$Var2 <- seq.int(nrow(pop)) 

pop <- pop %>% mutate (Population =  
						  ifelse (pop == "c_lc_zz", "Lynx canadensis",
                          ifelse (pop == "c_ll_po", "Bialowieza",
                          ifelse (pop == "c_ll_ur", "Urals",
                          ifelse (pop == "c_ll_ki", "Kirov",
                          ifelse (pop == "c_ll_la", "Latvia",
                          ifelse (pop == "c_ll_no", "Norway",
                          ifelse (pop == "c_ll_ba" | pop == "h_ll_ba", "Balkans" ,
                          ifelse (pop == "c_ll_cr","Carpathians",
                          ifelse (pop == "c_ll_to", "Töv",
                          ifelse (pop == "c_ll_tu", "Tuva",
                          ifelse (pop == "c_ll_ka", "Khentii", 
                          ifelse (pop == "c_ll_og", "Ömnögovi", 
                          ifelse (pop == "c_ll_vl", "Vladivostok", 
                          ifelse (pop == "c_ll_ya", "Yakutia",
                          ifelse (pop == "c_ll_zz", "Zoo",NA))))))))))))))))  %>% rename(Pop=pop) 
tmp1 <- melt(admix)  

tmp <- full_join(tmp1, pop, by="Var2")

tmp$ind <- factor (tmp$ind, levels=c("c_ll_no_0075", "c_ll_no_0076", "c_ll_no_0077", "c_ll_no_0078", "c_ll_no_0079", "c_ll_no_0080", "c_ll_no_0081", "c_ll_no_0082", "c_ll_ba_0216", "h_ll_ba_0214", "h_ll_ba_0215", "c_ll_cr_0205", "c_ll_cr_0206", "c_ll_cr_0207", "c_ll_cr_0208", "c_ll_cr_0209", "c_ll_cr_0212", "c_ll_po_0001", "c_ll_po_0002", "c_ll_po_0003", "c_ll_po_0011", "c_ll_po_0014", "c_ll_po_0019", "c_ll_po_0105", "c_ll_po_0106", "c_ll_la_0044", "c_ll_la_0047", "c_ll_la_0048", "c_ll_la_0052", "c_ll_la_0053", "c_ll_la_0054", "c_ll_ki_0090", "c_ll_ki_0091", "c_ll_ki_0092", "c_ll_ki_0093", "c_ll_ki_0094", "c_ll_ki_0095", "c_ll_ki_0096", "c_ll_ki_0097", "c_ll_ki_0098", "c_ll_ki_0099", "c_ll_ki_0100", "c_ll_ki_0101", "c_ll_ki_0102", "c_ll_ur_0194", "c_ll_ur_0195", "c_ll_ur_0196", "c_ll_ur_0199", "c_ll_ur_0200", "c_ll_ur_0203", "c_ll_tu_0153", "c_ll_tu_0157", "c_ll_tu_0158", "c_ll_tu_0159", "c_ll_tu_0165", "c_ll_tu_0166", "c_ll_to_0190", "c_ll_to_0191", "c_ll_ka_0184", "c_ll_ka_0186", "c_ll_ka_0188", "c_ll_ka_0189", "c_ll_og_0181", "c_ll_og_0187", "c_ll_ya_0138", "c_ll_ya_0139", "c_ll_ya_0140", "c_ll_ya_0142", "c_ll_ya_0143", "c_ll_ya_0145", "c_ll_ya_0146", "c_ll_ya_0147", "c_ll_vl_0107", "c_ll_vl_0108", "c_ll_vl_0109", "c_ll_vl_0110", "c_ll_vl_0112", "c_ll_vl_0113", "c_ll_vl_0128", "c_ll_vl_0132", "c_lc_zz_0001"))


# n <- 24 # Esto servía porque aquí venían las k y así hacia una paleta adecuada a la k que tenía, en mi caso, ahora no hace falta. 
# qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',] Antes era así, lo que significa que coje aquellas paletas que son cualitativas. como yo quiero que coja paired la primera para que sea como en el PCA, pongo como condición que tenga maxcolors de 12. 
twelve_col_pals = brewer.pal.info[brewer.pal.info$maxcolors == '12',] 
col_vector = unlist(mapply(brewer.pal, twelve_col_pals$maxcolors, rownames(twelve_col_pals)))
# pie(rep(1,n), col=sample(col_vector, n))
my_paired_colours <- as.vector(col_vector)

x_breaks <-tapply(1:nrow(pop),pop[,4],mean)
x_breaks_to_dataframe <- data.frame(Population=names(x_breaks), breaks=x_breaks) %>% mutate ()
x_breaks_to_dataframe$Population <- as.character(x_breaks_to_dataframe$Population) 


# Here I try to get when the pop changes to add vertical lines. 

count_changes_in_pop <- pop %>% select (Population) %>% mutate (Population2 = Population)
end <- nrow(count_changes_in_pop)
values_to_change <- data.table:::uniqlist(count_changes_in_pop[, 1:2])  # De esta manera me quedo con los numeros donde hay un cambio. 
values_to_change <- append(values_to_change , end)

###



#setEPS()
# postscript(paste0(args[4],"_K",args[3],".eps"))

# pdf(paste0("K_8.pdf"), height=10, width = 30)

admix_plot <- ggplot() + 
   geom_bar (data = tmp, (aes(x=ind, y=value, fill=Var1)), stat = "identity", position = "fill",  width=1) +
   scale_fill_manual(values=my_paired_colours) +
   scale_y_continuous(breaks = seq(0, 1, by = 0.2)) +
   theme_classic() +
   labs(y="Admixture proportion") +
   coord_cartesian(ylim = c(0, 1.2), xlim = c(0, nrow(pop)+10)) +
   theme(plot.margin=unit(c(1,1,1,1),"cm"), # top, right, bottom, and left margins
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.line = element_blank(),
        # axis.title.y = element_blank(),
        legend.position="none",
        axis.text.y=element_text(size=18),
        axis.title.y =  element_text(size=15,color="grey30", hjust = 0.4)) 

admix_plot <- admix_plot + annotation_custom(
      grob = linesGrob(y = c(0, 1),x = c(0,0), gp = gpar(col = "black", lwd = 2)),
      ymin = 0,      # Vertical position of the textGrob
      ymax = 1)

for (i in 1:length(x_breaks_to_dataframe$breaks)) {
admix_plot <- admix_plot + annotation_custom(
      grob = textGrob(label = x_breaks_to_dataframe$Population[i], rot =50, vjust = 1.05, hjust=0.1,
                      gp = gpar(col = "grey30",fontsize = 15, face="plain")),
      xmin = x_breaks_to_dataframe$breaks[i],         
      xmax = x_breaks_to_dataframe$breaks[i],
      ymin = 1,      # Vertical position of the textGrob
      ymax = 1.15)

admix_plot <- admix_plot + annotation_custom(
      grob = grid.lines(y = c(0,1), gp = gpar(col = "grey30", lwd = 1)),
      ymin = 0,      # Vertical position of the textGrob
      ymax = 1.02,
      xmin = values_to_change[i]-0.5,      # Vertical position of the textGrob
      xmax = values_to_change[i]-0.5)
}
   
ggsave (paste0(args[4],"_K",args[3],".pdf"), admix_plot, width = 25, height = 15, units = "cm", device="pdf")
 
```
As the plot looks perfect, I will run both NGSAdmix and PCA over the 6 samples, 25x. 


# PCA lynx 25X

```{bash}

nano samples_MACROGEN.bamlist


/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_lc_zz_0001_recal_round-1_25x.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_ll_cr_0212_recal_round-1_25x.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_ll_ki_0090_recal_round-1_25x.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_ll_vl_0112_recal_round-1_25x.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_ll_ya_0146_recal_round-1_25x.bam
/home/mlucena/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x/c_lr_zz_0001_recal_round-1_25x.bam





POP="c_lc_zz-c_ll_cr-c_ll_ki-c_ll_vl-c_ll_ya-c_lr_zz_n006"

ANGSD=/opt/angsd/angsd
NGSTOOLS="/opt/ngsTools"
THREADS=20                     # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2) 
REGIONFILE="/home/mlucena/ANGSD_analysis/depth_calculus/no_genes_Lypa_10000longest_center_final_slop20_dot.rf"




# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
# echo $maxDepth
# echo $minDepth


### Calling genotype likelihoods: 
 
$ANGSD/angsd -P $THREADS -b $POP.bamlist -ref $REF -out $POP.rf \
$FILTER1 $FILTER2 \
-rf $REGIONFILE \
-GL 1 -doMajorMinor 1 -doMaf 1 -skipTriallelic 1 \
-doGeno 32 -doPost 1




gunzip $POP.rf.geno.gz

### ngsCovar

# We are going to use ngsCovar, which estimates the covariance matrix between individuals based on genotype probabilities. 
# Then this matrix will be decomposed into principal components which will be investigated for population structure analyses. 
# Note that although ngsCovar can account for SNPs uncertanity, we find that it is faster to perform a light SNP filtering first 
# (as we did using -SNP_pval 1e-3) than using all sites.

# If you type:

# $NGSTOOLS/ngsPopGen/ngsCovar

# you will see a list of possible options. 
# For instance, we need to define how many sites we have. To retrieve such value, we can inspect the file with allele frequencies:

#less -S $POP.mafs.gz
N_SITES=`zcat $POP.rf.mafs.gz | tail -n+2 | wc -l`
echo $N_SITES 

# 1647013


# Now we can perform a PCA by estimating the covariance matrix first:

$NGSTOOLS/ngsPopGen/ngsCovar -probfile $POP.*geno -outfile $POP.covar -nind $N_IND  -nsites $N_SITES -call 0 -norm 0 &> /dev/null

#-minmaf FLOAT: ignore sites below this threshold of minor allele frequency

# with the options -call 0 meaning that we do not perform genotype calling 
# and -norm 0 that we are not normalising by allele frequency. 
# The latter may give more weight to low frequency variants which are harder to estimate.

# Look at the output file:

#less -S $POP.covar

# which represents a matrix of NxN with N individuals giving the covariance. Note that this matrix is symmetric.

# Finally, we perform an eigenvector decomposition and plot the resulting map:

# create a cluster-like file defining the labelling (population) for each sample

#Example: Rscript -e 'write.table(cbind(seq(1,60),rep(1,60),c(rep("LWK",20),rep("TSI",20),rep("PEL",20))), row.names=F, sep=" ", col.names=c("FID","IID","CLUSTER"), file="Results/ALL.clst", quote=F)'

echo "FID" > $POP.ALL1.clst.borrar
seq $N_IND >> $POP.ALL1.clst.borrar
echo "IID" > $POP.ALL2.clst.borrar
# yes "1" | head -n $N_IND >> $POP.ALL2.clst.borrar
cat $POP.bamlist | rev | cut -d "/"  -f 1 | rev | colrm 13  >> $POP.ALL2.clst.borrar
echo "CLUSTER" > $POP.ALL3.clst.borrar
cat $POP.bamlist | rev | cut -d "/"  -f 1 | rev | colrm 8  >> $POP.ALL3.clst.borrar
paste $POP.ALL1.clst.borrar $POP.ALL2.clst.borrar $POP.ALL3.clst.borrar> $POP.ALL.clst

rm $POP.*clst.borrar

# run and plot

# Rscript $NGSTOOLS/Scripts/plotPCA.R -i Results/ALL.covar -c 1-2 -a Results/ALL.clst -o Results/ALL.pca.pdf
# Rscript $NGSTOOLS/Scripts/plotPCA.R -i $POP.covar -c 1-2 -a $POP.ALL.clst -o $POP.ALL.pca.pdf #--> No corre porque no puedo instalar librerías así que lo hago en mi propio ordenador
# evince Results/ALL.pca.pdf

# where the parameter -c 1-2 specifies that we are plotting only the first and second component. On the screen, you will see a series of numbers. These are the percentage of explained variance for each component.

```

## Copying to Dropbox folder


```{bash}
scp mlucena@genomics-b.ebd.csic.es://home/mlucena/ANGSD_analysis/ancestry_lc_0001/*.ALL.clst  /Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001

scp mlucena@genomics-b.ebd.csic.es:/home/mlucena/ANGSD_analysis/ancestry_lc_0001/*.covar /Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001 

```


## R representation

```{r "setup", include=FALSE}

library(methods)
library(optparse)
library(ggplot2)
library(dplyr)
library(threejs)
library(RColorBrewer)
require("knitr")

# knitr::opts_knit$set(root.dir = normalizePath("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/analysis_PCA/test_intergenic_elena"))
# getwd()
# opts_knit$set(root.dir = normalizePath("/Users/marialucenaperez/Dropbox/PhD/contemporary/ANGSD/analysis_PCA/test_intergenic_elena"))

WORKING_DIR="/Users/marialucenaperez/Dropbox/PhD/contemporary/ancestry_lc_001/"

pop=c("c_lc_zz-c_ll_no-c_ll_ba-h_ll_ba-c_ll_cr-c_ll_po-c_ll_la-c_ll_ki-c_ll_ur-c_ll_tu-c_ll_to-c_ll_ka-c_ll_og-c_ll_ya-c_ll_vl_n081")
pop=c("c_lc_zz-c_ll_cr-c_ll_ki-c_ll_vl-c_ll_ya-c_lr_zz_n006")

 
###########################################################################################
#for (pop in pops) 
#{

# Annotation file is in plink cluster format

#################################################################################

# Read input file
covar <- read.table(paste(WORKING_DIR,pop, ".covar", sep=''), stringsAsFact=F);

# Read annot file
annot <- read.table(paste(WORKING_DIR,pop, ".ALL.clst", sep=""), header=TRUE) 

# Parse components to analyze
# comp <- as.numeric(strsplit("component1-component2", "-", fixed=TRUE)[[1]])
# comp <- as.numeric(strsplit(opt$comp, "-", fixed=TRUE)[[1]])

comp <- c(1,2)
# comp <- c(2,3)
# comp <- c(1,3)


# Eigenvalues
eig <- eigen(covar, symm=TRUE);
eig$val <- eig$val/sum(eig$val);
cat(signif(eig$val, digits=3)*100,"\n");

# Plot
PC <- as.data.frame(eig$vectors)
colnames(PC) <- gsub("V", "PC", colnames(PC))
PC$Pop <- factor(annot$CLUSTER)
PC$Ind <- factor(annot$IID)
# PC <- PC %>% filter(Ind == "UA")



PC <- PC %>%  mutate(., Populations =  ifelse (Pop == "c_lr_zz", "Lynx_rufus",
                          ifelse (Pop == "c_ll_ki", "Kirov",
                          ifelse (Pop == "c_ll_cr","Carpathians",
                          ifelse (Pop == "c_ll_vl", "Vladivostok", 
                          ifelse (Pop == "c_ll_ya", "Yakutia",
                          ifelse (Pop == "c_lc_zz", "Lynx_canadensis",NA))))))) %>% 
    mutate (., color =  ifelse (Populations == "Lynx_rufus", brewer.pal(12,"Paired")[4], 
                      ifelse (Populations == "Carpathians", brewer.pal(12,"Paired")[9], 
                      ifelse (Populations == "Kirov", brewer.pal(12,"Paired")[1], 
                      ifelse (Populations == "Vladivostok", brewer.pal(12,"Paired")[5], 
                      ifelse (Populations == "Yakutia", brewer.pal(12,"Paired")[6],
                      ifelse (Populations == "Lynx_canadensis", brewer.pal(8, "Greys") [5], NA))))))) 

                           
                           
cols <- c("Lynx_rufus"=brewer.pal(12,"Paired")[4], 
          "Carpathians"=brewer.pal(12,"Paired")[9], 
          "Kirov"=brewer.pal(12,"Paired")[1], 
          "Vladivostok"=brewer.pal(12,"Paired")[5], 
          "Yakutia"=brewer.pal(12,"Paired")[6],
          "Lynx_canadensis"=brewer.pal(8, "Greys") [5])

title <- paste("PC",comp[1]," (",signif(eig$val[comp[1]], digits=3)*100,"%)"," / PC",comp[2]," (",signif(eig$val[comp[2]], digits=3)*100,"%)",sep="",collapse="")

x_axis = paste("PC",comp[1],sep="")
y_axis = paste("PC",comp[2],sep="")



ggplot() + 
  geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Populations"), size = 5) + 
  scale_colour_manual(values = cols)+ ggtitle(title) + theme_classic()
 ggsave(paste(WORKING_DIR,pop, "_pca_pop_PC1_PC2.pdf", sep=""),units="mm",  width=270, height=190)



 ggplot() + 
  geom_point(data=PC, aes_string(x=x_axis, y=y_axis, colour="Populations"), size = 5) + 
  scale_colour_manual(values = cols)+ ggtitle(title) + theme_classic()
 ggsave(paste(WORKING_DIR,pop, "_pca_pop_PC2_PC3.pdf", sep=""),units="mm",  width=270, height=190)

 
 
 
###########################################################################################
# 3D plot
###########################################################################################

scatterplot3js(PC$PC1, PC$PC2, PC$PC3, 
                col=PC$color,  labels=PC$Ind,
               size=0.3, grid=FALSE, axis =T)


, renderer = "canvas"
```





# NGSAdmix 25X



```{bash}

POP="c_lc_zz-c_ll_cr-c_ll_ki-c_ll_vl-c_ll_ya-c_lr_zz_n006"  # <--CHANGE POP HERE


THREADS=15                    # no. of computer cores used by bwa and samtools. 20 = OK, >20 = ask people first!


ANGSD="/opt/angsd/angsd"
NGSTOOLS="/opt/angsd/angsd/misc"
REF="/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23_without_repetitive_transposable_low_complexity.fa"
ANC="/home/GRUPOS/grupolince/reference_genomes/lynx_rufus_genome/c_lr_zz_0001_recal1.fa"
FILTER1=" -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -baq 1 -C 50 "
FILTER2=" -minMapQ 30 -minQ 20 -doCounts 1 "
N_IND=$(echo ${POP: -3} )
MIN_IND=$(expr $N_IND / 2)
REGIONFILE="/home/mlucena/ANGSD_analysis/depth_calculus/no_genes_Lypa_10000longest_center_final_slop20_dot.rf"



# Sanity checks:

echo $POP
echo $N_IND
echo $MIN_IND
# echo $maxDepth
# echo $minDepth
# echo $SNP_PVAL

# Running GL:

$ANGSD/angsd -nThreads $THREADS -bam "$POP".bamlist -ref $REF  \
-out "$POP".genolike \
$FILTER1 \
$FILTER2 \
-rf $REGIONFILE \
-GL 1  -doGlf 2 \
-doMajorMinor 1 -doMaf 1 -skipTriallelic 1 

```



